<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data Analysis with `augmentedRCBD` • augmentedRCBD</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Data Analysis with `augmentedRCBD`">
<meta property="og:description" content="augmentedRCBD">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-116716530-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-116716530-2');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">augmentedRCBD</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.1.4.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
    Home
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fas fa-code"></span>
     
    Reference
  </a>
</li>
<li>
  <a href="../articles/Data_Analysis_with_augmentedRCBD.html">
    <span class="far fa-file"></span>
     
    Tutorial
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="far fa-newspaper"></span>
     
    News
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/aravind-j/augmentedRCBD">
    <span class="fab fa-github-alt"></span>
     
    Github
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Data_Analysis_with_augmentedRCBD_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data Analysis with <code>augmentedRCBD</code>
</h1>
                        <h4 class="author">Aravind, J.<sup>1</sup>, Mukesh Sankar, S.<sup>2</sup>, Wankhede, D. P.<sup>3</sup>, and Kaur, V.<sup>4</sup>
</h4>
            
            <h4 class="date">2021-05-20</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/aravind-j/augmentedRCBD/blob/master/vignettes/Data_Analysis_with_augmentedRCBD.Rmd"><code>vignettes/Data_Analysis_with_augmentedRCBD.Rmd</code></a></small>
      <div class="hidden name"><code>Data_Analysis_with_augmentedRCBD.Rmd</code></div>

    </div>

    
    
<p>
</p>
<ol style="list-style-type: decimal">
<li>Division of Germplasm Conservation, ICAR-National Bureau of Plant Genetic Resources, New Delhi.
</li>
</ol>
<p>
</p>
<ol start="2" style="list-style-type: decimal">
<li>Division of Genetics, ICAR-Indian Agricultural Research Institute, New Delhi.
</li>
</ol>
<p>
</p>
<ol start="3" style="list-style-type: decimal">
<li>Division of Genomic Resources, ICAR-National Bureau of Plant Genetic Resources, New Delhi.
</li>
</ol>
<p>
</p>
<ol start="4" style="list-style-type: decimal">
<li>Division of Germplasm Evaluation, ICAR-National Bureau of Plant Genetic Resources, New Delhi.
</li>
</ol>
<p><img src="https://raw.githubusercontent.com/aravind-j/augmentedRCBD/master/inst/extdata/augmentedRCBD.png" align="right" alt="logo" width="173" height="200" style="padding: 10px; border: none; float: right;"></p>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>1 Overview</h1>
<p>The software <code>augmentedRCBD</code> is built on the <a href="https://en.wikipedia.org/wiki/R_(programming_language)"><code>R</code> statistical programming language</a> as an add-on (or ‘package’ in the <code>R</code> <em>lingua franca</em>). It performs the analysis of data generated from experiments in augmented randomised complete block design according to Federer, W.T. <span class="citation">(<a href="#ref-federer_augmented_1956" role="doc-biblioref">1956</a><a href="#ref-federer_augmented_1956" role="doc-biblioref">a</a>, <a href="#ref-federer_augmented_1956-1" role="doc-biblioref">1956</a><a href="#ref-federer_augmented_1956-1" role="doc-biblioref">b</a>, <a href="#ref-federer_augmented_1961" role="doc-biblioref">1961</a>; <a href="#ref-federerModelConsiderationsVariance1976" role="doc-biblioref">1976</a>)</span>. It also computes analysis of variance, adjusted means, descriptive statistics, genetic variability statistics etc. and includes options for data visualization and report generation.</p>
<p>This tutorial aims to educate the users in utilising this package for performing such analysis. Utilising <code>augmentedRCBD</code> for data analysis requires a basic knowledge of <code>R</code> programming language. However, as many of the intended end-users may not be familiar with <code>R</code>, <a href="#rsoft">sections 2 to 4</a> give a ‘gentle’ introduction to <code>R</code>, especially those aspects which are necessary to get <code>augmentedRCBD</code> up and running for performing data analysis in a Windows environment. Users already familiar with <code>R</code> can feel free to skip to <a href="#install">section 5</a>.</p>

<p><img src="https://www.r-project.org/logo/Rlogo.png" align="right" alt="logo" width="173" height="134" style="padding: 10px; border: none; float: right;"></p>
</div>
<div id="rsoft" class="section level1">
<h1 class="hasAnchor">
<a href="#rsoft" class="anchor"></a>2 <code>R</code> software</h1>
<p>It is a free software environment for statistical computing and graphics. It is free and open source, platform independent (works on Linux, Windows or MacOS), very flexible, comprehensive with robust interfaces for all the popular programming languages as well as databases. It is strengthened by its diverse library of add-on packages extending its ability as well as the incredible community support. It is one of the most popular tools being used in academia today <span class="citation">(Tippmann, <a href="#ref-tippmann_programming_2015" role="doc-biblioref">2015</a>)</span>.</p>

</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>3 Getting Started</h1>
<p>This section details the steps required to set up the <code>R</code> programming environment under a third-party interface called <code>RStudio</code> in Windows.</p>
<div id="installing-r" class="section level2">
<h2 class="hasAnchor">
<a href="#installing-r" class="anchor"></a>3.1 Installing <code>R</code>
</h2>
<p>Download and install <code>R</code> for Windows from <a href="http://cran.r-project.org/bin/windows/base/" class="uri">http://cran.r-project.org/bin/windows/base/</a>.</p>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/rbase.png" align="center" alt="The `R` download location."></p>
<p style="text-align: left;">
<strong>Fig. 1</strong>: The <code>R</code> download location.
</p>
</div>
<div id="installing-rstudio" class="section level2">
<h2 class="hasAnchor">
<a href="#installing-rstudio" class="anchor"></a>3.2 Installing <code>RStudio</code>
</h2>
<p>The basic <a href="https://en.wikipedia.org/wiki/Command_line_interface">command line interface</a> in native <code>R</code> is rather limiting. There are several interfaces which enhance it’s functionality and ease of use, <a href="https://www.rstudio.com/"><code>RStudio</code></a> being one of the most popular among <code>R</code> programmers.</p>
<p>Download and install <code>RStudio</code> for Windows from <a href="https://www.rstudio.com/products/rstudio/download/#download" class="uri">https://www.rstudio.com/products/rstudio/download/#download</a></p>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/rstudio.png" align="center" alt="The `RStudio` download location."></p>
<p style="text-align: left;">
<strong>Fig. 2</strong>: The <code>RStudio</code> download location.
</p>
</div>
<div id="the-rstudio-interface" class="section level2">
<h2 class="hasAnchor">
<a href="#the-rstudio-interface" class="anchor"></a>3.3 The <code>RStudio</code> Interface</h2>
<p>On opening <code>RStudio</code>, the default interface with four panes/windows is visible as follows. Few panes have different tabs.</p>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/rstudio%20panes.png" align="center" alt="The default `RStudio` interface with the four panes."></p>
<p style="text-align: left;">
<strong>Fig. 3</strong>: The default <code>RStudio</code> interface with the four panes.
</p>
<div id="console" class="section level3">
<h3 class="hasAnchor">
<a href="#console" class="anchor"></a>3.3.1 Console</h3>
<p>This is where the action happens. Here any authentic <code>R</code> code typed after the ‘<code><a href="https://rdrr.io/r/base/Comparison.html">&gt;</a></code>’ prompt will be executed after pressing ‘Enter’ to generate the output.</p>
<p>For example, type <code>1+1</code> in the console and press ‘Enter’.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fl">1</span><span class="op">+</span><span class="fl">1</span></code></pre></div>
<pre><code>[1] 2</code></pre>
</div>
<div id="source" class="section level3">
<h3 class="hasAnchor">
<a href="#source" class="anchor"></a>3.3.2 Source</h3>
<p>This is where <code>R</code> Scripts (collection of code) can be created and edited. <code>R</code> scripts are text files with a <code>.R</code> extension. <code>R</code> Code for analysis can be typed and saved in such <code>R</code> scripts. New scripts can be opened by clicking ‘File|New File’ and selecting ‘R Script’. Code can be selected from <code>R</code> Scripts and sent to console for evaluation by clicking ‘Run’ on the ‘Source’ pane or by pressing ‘Ctrl + Enter’.</p>
</div>
<div id="environmenthistoryconnections" class="section level3">
<h3 class="hasAnchor">
<a href="#environmenthistoryconnections" class="anchor"></a>3.3.3 Environment|History|Connections</h3>
<p>The ‘Environment’ tab shows the list of all the ‘objects’ (see <a href="#ObjFun">section 4.3</a>) defined in the current <code>R</code> session. It has also some buttons up top to open, save and clear the environment as well as few options for import of data under <code>Import Dataset</code>.</p>
<p>The ‘History’ tab shows a history of all the code that was previously evaluated. This is useful, if you want to go back to some code.</p>
<p>The ‘Connections’ tab helps to establish and manage connections with different databases and data sources.</p>
</div>
<div id="filesplotspackageshelpviewer" class="section level3">
<h3 class="hasAnchor">
<a href="#filesplotspackageshelpviewer" class="anchor"></a>3.3.4 Files|Plots|Packages|Help|Viewer</h3>
<p>The ‘Files’ tab shows a sleek file browser to access the file directory in the computer with options to manage the working directory (see <a href="#wdir">section 4.1</a>) under the More button.</p>
<p>The ‘Plots’ tab shows all the plots generated in <code>R</code> with buttons to delete unnecessary ones and export useful ones as a pdf file or as an image file.</p>
<p>The ‘Packages’ tab shows a list of all the <code>R</code> add-on packages installed. The check box on the left shows whether they are loaded or not. There are also buttons to install and update <code>R</code> packages.</p>
<p>The ‘Viewer’ tab shows any web content output generated by an <code>R</code> code.</p>
</div>
</div>
</div>
<div id="some-basics" class="section level1">
<h1 class="hasAnchor">
<a href="#some-basics" class="anchor"></a>4 Some Basics</h1>
<p>This section describes some basics to enable the users to have a working knowledge in <code>R</code> in order to use <code>augmentedRCBD</code>.</p>
<div id="wdir" class="section level2">
<h2 class="hasAnchor">
<a href="#wdir" class="anchor"></a>4.1 Working Directory</h2>
<p>It is a file path to a folder on the computer which is recognised by <code>R</code> as the default location to read files from or write files to. The code <code><a href="https://rdrr.io/r/base/getwd.html">getwd()</a></code> shows the current working directory, while <code><a href="https://rdrr.io/r/base/getwd.html">setwd()</a></code> can be used to change the existing working directory.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Print current working directory</span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "C:/Users/Computer/Documents"</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Set new working directory</span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"C:/Data Analysis/"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "C:/Data Analysis/"</code></pre>
<p>One key detail is that file paths in <code>R</code> uses forward slashes (<code><a href="https://rdrr.io/r/base/Arithmetic.html">/</a></code>) as in MacOS or Linux, unlike backward slashes (<code>\</code>) in Windows. This needs to be considered while copying paths from default Windows file explorer.</p>
</div>
<div id="expression-and-assignment" class="section level2">
<h2 class="hasAnchor">
<a href="#expression-and-assignment" class="anchor"></a>4.2 Expression and Assignment</h2>
<p>Expressions are instructions in the form of code to be entered after the <code><a href="https://rdrr.io/r/base/Comparison.html">&gt;</a></code> prompt in the console. Expressions can be a constant, an arithmetic or a condition. A more advanced and most useful expression is a function call (see <a href="#ObjFun">section 4.3</a>).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Constant</span>
<span class="fl">123</span></code></pre></div>
<pre><code>[1] 123</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Arithmetic (add two numbers)</span>
<span class="fl">1</span> <span class="op">+</span> <span class="fl">2</span></code></pre></div>
<pre><code>[1] 3</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Condition</span>
<span class="fl">34</span> <span class="op">&gt;</span> <span class="fl">25</span></code></pre></div>
<pre><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fl">1</span> <span class="op">==</span> <span class="fl">2</span></code></pre></div>
<pre><code>[1] FALSE</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Function call (mean of a series of numbers)</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">25</span>,<span class="fl">56</span>,<span class="fl">89</span>,<span class="fl">35</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 51.25</code></pre>
<p>Information from an expression can be stored as an ‘object’ (see <a href="#ObjFun">section 4.3</a>) by assigning a name using the operator ‘<code><a href="https://rdrr.io/r/base/assignOps.html">&lt;-</a></code>’.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Assign the result of the expression 1 + 2 to an object 'a'</span>
<span class="va">a</span> <span class="op">&lt;-</span> <span class="fl">1</span> <span class="op">+</span> <span class="fl">2</span>
<span class="va">a</span></code></pre></div>
<pre><code>[1] 3</code></pre>
<p>It is recommended to add comments to explain the code by using the ‘<code>#</code>’ sign. Any code after the ‘<code>#</code>’ sign will be ignored by <code>R</code>.</p>
</div>
<div id="ObjFun" class="section level2">
<h2 class="hasAnchor">
<a href="#ObjFun" class="anchor"></a>4.3 Objects and Functions</h2>
<p><code>R</code> is an object-oriented programming language (OOP). Any kind or construct created in <code>R</code> is an ‘object’. Each object has a ‘class’ (shown using the <code><a href="https://rdrr.io/r/base/class.html">class()</a></code> function) and different ‘attributes’ which defines what operations can be done on that object. There are different types of data structure objects in <code>R</code> such as vectors, matrices, factors, data frames, and lists. A ‘function’ is also an object, which defines a procedure or a sequence of expressions.</p>
<div id="vector" class="section level3">
<h3 class="hasAnchor">
<a href="#vector" class="anchor"></a>4.3.1 Vector</h3>
<p>A vector is a collection of elements of a single type (or ‘mode’). The common vector modes are ‘numeric’, ‘integer’, ‘character’ and ‘logical’. The <code><a href="https://rdrr.io/r/base/c.html">c()</a></code> function is used to create vectors. The functions <code><a href="https://rdrr.io/r/base/class.html">class()</a></code>, <code><a href="https://rdrr.io/r/utils/str.html">str()</a></code> and <code><a href="https://rdrr.io/r/base/length.html">length()</a></code> show the attributes of vectors.</p>
<p>Vector modes ‘numeric’ stores real numbers, while ‘integer’ stores integers, which can be enforced by suffixing elements with ‘<code>L</code>’.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># A numeric vector</span>
<span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3.3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "numeric"</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></code></pre></div>
<pre><code> num [1:3] 1 2 3.3</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 3</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># An integer vector</span>
<span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1L</span>, <span class="fl">2L</span>, <span class="fl">3L</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "integer"</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></code></pre></div>
<pre><code> int [1:3] 1 2 3</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 3</code></pre>
<p>The vector mode ‘character’ store text.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># A character vector</span>
<span class="va">c</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"one"</span>,<span class="st">"two"</span>,<span class="st">"three"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "character"</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></code></pre></div>
<pre><code> chr [1:3] "one" "two" "three"</code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 3</code></pre>
<p>The vector mode ‘logical’ stores ‘<code>TRUE</code>’ OR ‘<code>FALSE</code>’ logical data.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#logical vector</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="cn">TRUE</span>,<span class="cn">TRUE</span>,<span class="cn">TRUE</span>,<span class="cn">FALSE</span>,<span class="cn">TRUE</span>,<span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "logical"</code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></code></pre></div>
<pre><code> logi [1:6] TRUE TRUE TRUE FALSE TRUE FALSE</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 6</code></pre>
</div>
<div id="factor" class="section level3">
<h3 class="hasAnchor">
<a href="#factor" class="anchor"></a>4.3.2 Factor</h3>
<p>A ‘factor’ in <code>R</code> stores data from categorical data in variables as different levels.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">catg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"male"</span>,<span class="st">"female"</span>,<span class="st">"female"</span>,<span class="st">"male"</span>,<span class="st">"male"</span><span class="op">)</span>
<span class="va">catg</span></code></pre></div>
<pre><code>[1] "male"   "female" "female" "male"   "male"  </code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/factor.html">is.factor</a></span><span class="op">(</span><span class="va">catg</span><span class="op">)</span></code></pre></div>
<pre><code>[1] FALSE</code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Apply the factor function</span>
<span class="va">factor_catg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">catg</span><span class="op">)</span>

<span class="va">factor_catg</span></code></pre></div>
<pre><code>[1] male   female female male   male  
Levels: female male</code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/factor.html">is.factor</a></span><span class="op">(</span><span class="va">factor_catg</span><span class="op">)</span></code></pre></div>
<pre><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">factor_catg</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "factor"</code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">factor_catg</span><span class="op">)</span></code></pre></div>
<pre><code> Factor w/ 2 levels "female","male": 2 1 1 2 2</code></pre>
<p>A character, numeric or integer vector can be transformed to a factor by using the <code><a href="https://rdrr.io/r/base/factor.html">as.factor()</a></code> function.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Conversion of numeric to factor</span>
<span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3.3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "numeric"</code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></code></pre></div>
<pre><code> num [1:3] 1 2 3.3</code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fac_a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">fac_a</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "factor"</code></pre>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">fac_a</span><span class="op">)</span></code></pre></div>
<pre><code> Factor w/ 3 levels "1","2","3.3": 1 2 3</code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Conversion of integer to factor</span>
<span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1L</span>, <span class="fl">2L</span>, <span class="fl">3L</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "integer"</code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></code></pre></div>
<pre><code> int [1:3] 1 2 3</code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fac_b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">fac_b</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "factor"</code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">fac_b</span><span class="op">)</span></code></pre></div>
<pre><code> Factor w/ 3 levels "1","2","3": 1 2 3</code></pre>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Conversion of character to factor</span>
<span class="va">c</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"one"</span>,<span class="st">"two"</span>,<span class="st">"three"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "character"</code></pre>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></code></pre></div>
<pre><code> chr [1:3] "one" "two" "three"</code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fac_c</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">fac_c</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "factor"</code></pre>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">fac_c</span><span class="op">)</span></code></pre></div>
<pre><code> Factor w/ 3 levels "one","three",..: 1 3 2</code></pre>
</div>
<div id="matrix" class="section level3">
<h3 class="hasAnchor">
<a href="#matrix" class="anchor"></a>4.3.3 Matrix</h3>
<p>A ‘matrix’ in <code>R</code> is a vector with the attributes ‘<code>nrow</code>’ and ‘<code>ncol</code>’.</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Generate 5 * 4 numeric matrix</span>
<span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span>, nrow <span class="op">=</span> <span class="fl">5</span>, ncol <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>
<span class="va">m</span></code></pre></div>
<pre><code>     [,1] [,2] [,3] [,4]
[1,]    1    6   11   16
[2,]    2    7   12   17
[3,]    3    8   13   18
[4,]    4    9   14   19
[5,]    5   10   15   20</code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "matrix" "array" </code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/typeof.html">typeof</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "integer"</code></pre>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Dimensions of m</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span> </code></pre></div>
<pre><code>[1] 5 4</code></pre>
</div>
<div id="list" class="section level3">
<h3 class="hasAnchor">
<a href="#list" class="anchor"></a>4.3.4 List</h3>
<p>A ‘list’ is a container containing different objects. The contents of list need not be of the same type or mode. A list can encompass a mixture of data types such as vectors, matrices, data frames, other lists or any other data structure.</p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">a</span>, <span class="va">m</span>, <span class="va">d</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">b</span>, <span class="va">c</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">w</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "list"</code></pre>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">w</span><span class="op">)</span></code></pre></div>
<pre><code>List of 4
 $ : num [1:3] 1 2 3.3
 $ : int [1:5, 1:4] 1 2 3 4 5 6 7 8 9 10 ...
 $ : logi [1:6] TRUE TRUE TRUE FALSE TRUE FALSE
 $ :List of 2
  ..$ : int [1:3] 1 2 3
  ..$ : chr [1:3] "one" "two" "three"</code></pre>
</div>
<div id="dataframe" class="section level3">
<h3 class="hasAnchor">
<a href="#dataframe" class="anchor"></a>4.3.5 Data Frame</h3>
<p>A ‘data frame’ in <code>R</code> is a special kind of list with every element having equal length. It is very important for handling tabular data in <code>R</code>. It is a array like structure with rows and columns. Each column needs to be of a single data type, however data type can vary between columns.</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">L</span> <span class="op">&lt;-</span> <span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="va">y</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span>
<span class="va">z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"This"</span>, <span class="st">"is"</span>, <span class="st">"a"</span>, <span class="st">"data frame"</span><span class="op">)</span>
<span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">L</span>, x <span class="op">=</span> <span class="fl">1</span>, <span class="va">y</span>, <span class="va">z</span><span class="op">)</span>
<span class="va">df</span></code></pre></div>
<pre><code>  L x y          z
1 A 1 1       This
2 B 1 2         is
3 C 1 3          a
4 D 1 4 data frame</code></pre>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></code></pre></div>
<pre><code>'data.frame':   4 obs. of  4 variables:
 $ L: chr  "A" "B" "C" "D"
 $ x: num  1 1 1 1
 $ y: int  1 2 3 4
 $ z: chr  "This" "is" "a" "data frame"</code></pre>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/attributes.html">attributes</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></code></pre></div>
<pre><code>$names
[1] "L" "x" "y" "z"

$class
[1] "data.frame"

$row.names
[1] 1 2 3 4</code></pre>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "1" "2" "3" "4"</code></pre>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "L" "x" "y" "z"</code></pre>
</div>
<div id="functions" class="section level3">
<h3 class="hasAnchor">
<a href="#functions" class="anchor"></a>4.3.6 Functions</h3>
<p>All of the work in <code>R</code> is done by functions. It is an object defining a procedure which takes one or more objects as input (or ‘arguments’), performs some action on them and finally gives a new object as output (or ‘return’). <code><a href="https://rdrr.io/r/base/class.html">class()</a></code>, <code><a href="https://rdrr.io/r/base/mean.html">mean()</a></code>, <code><a href="https://rdrr.io/r/base/getwd.html">getwd()</a></code>, <code><a href="https://rdrr.io/r/base/Arithmetic.html">+</a></code>, etc. are all functions.</p>
<p>For example the function <code><a href="https://rdrr.io/r/base/mean.html">mean()</a></code> takes a numeric vector as argument and returns the mean as a numeric vector.</p>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3.3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 2.1</code></pre>
<p>The user can also create custom functions. For example the function <code>foo</code> adds two numbers and gives the result.</p>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">foo</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">n1</span>, <span class="va">n2</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">out</span> <span class="op">&lt;-</span> <span class="va">n1</span> <span class="op">+</span> <span class="va">n2</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu">foo</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span></code></pre></div>
<pre><code>[1] 5</code></pre>
</div>
</div>
<div id="special-elements" class="section level2">
<h2 class="hasAnchor">
<a href="#special-elements" class="anchor"></a>4.4 Special Elements</h2>
<p>In addition to numbers and text, there are some special elements which can be included in different data objects.</p>
<p><code>NA</code> (not available) indicates missing data.</p>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2.5</span>, <span class="cn">NA</span>, <span class="fl">8.6</span><span class="op">)</span>
<span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">FALSE</span>, <span class="cn">NA</span><span class="op">)</span>
<span class="va">z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"k"</span>, <span class="cn">NA</span>, <span class="st">"m"</span>, <span class="st">"n"</span>, <span class="st">"o"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<pre><code>[1] FALSE  TRUE FALSE</code></pre>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></code></pre></div>
<pre><code>[1] FALSE  TRUE FALSE FALSE FALSE</code></pre>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/NA.html">anyNA</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<pre><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">a</span></code></pre></div>
<pre><code>[1] 1.0 2.0 3.3</code></pre>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></code></pre></div>
<pre><code>[1] FALSE FALSE FALSE</code></pre>
<p><code>Inf</code> indicates infinity.</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fl">1</span><span class="op">/</span><span class="fl">0</span></code></pre></div>
<pre><code>[1] Inf</code></pre>
<p><code>NaN</code> (Not a Number) indicates any undefined value.</p>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fl">0</span><span class="op">/</span><span class="fl">0</span></code></pre></div>
<pre><code>[1] NaN</code></pre>
</div>
<div id="indexing" class="section level2">
<h2 class="hasAnchor">
<a href="#indexing" class="anchor"></a>4.5 Indexing</h2>
<p>The <code><a href="https://rdrr.io/r/base/Extract.html">[</a></code> function is used to extract elements of an object by indexing (numeric or logical). Named elements in lists and data frames can be extracted by using the <code><a href="https://rdrr.io/r/base/Extract.html">$</a></code> operator.</p>
<p>Consider a vector <code>a</code>.</p>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3.3</span>, <span class="fl">2.8</span>, <span class="fl">6.7</span><span class="op">)</span>
<span class="co"># Numeric indexing</span>
<span class="co"># Extract first element</span>
<span class="va">a</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></code></pre></div>
<pre><code>[1] 1</code></pre>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Extract elements 2:3</span>
<span class="va">a</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></code></pre></div>
<pre><code>[1] 2.0 3.3</code></pre>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Logical indexing</span>
<span class="va">a</span><span class="op">[</span><span class="va">a</span> <span class="op">&gt;</span> <span class="fl">3</span><span class="op">]</span></code></pre></div>
<pre><code>[1] 3.3 6.7</code></pre>
<p>Consider a matrix <code>m</code>.</p>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">9</span>, nrow <span class="op">=</span> <span class="fl">3</span>, ncol <span class="op">=</span> <span class="fl">3</span>, byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'a'</span>, <span class="st">'b'</span>, <span class="st">'c'</span><span class="op">)</span>
<span class="va">m</span></code></pre></div>
<pre><code>     a b c
[1,] 1 2 3
[2,] 4 5 6
[3,] 7 8 9</code></pre>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Extract elements</span>
<span class="va">m</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="co"># 2nd column of matrix</span></code></pre></div>
<pre><code>[1] 2 5 8</code></pre>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span><span class="op">[</span><span class="fl">3</span>,<span class="op">]</span> <span class="co"># 3rd row of matrix</span></code></pre></div>
<pre><code>a b c 
7 8 9 </code></pre>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="co"># rows 2,3 of columns 1,2,3 </span></code></pre></div>
<pre><code>     a b c
[1,] 4 5 6
[2,] 7 8 9</code></pre>
<div class="sourceCode" id="cb133"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">]</span> <span class="co"># Element in 2nd column of 2nd row</span></code></pre></div>
<pre><code>b 
5 </code></pre>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span><span class="op">[</span>, <span class="st">'b'</span><span class="op">]</span> <span class="co"># Column 'b'</span></code></pre></div>
<pre><code>[1] 2 5 8</code></pre>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'a'</span>, <span class="st">'c'</span><span class="op">)</span><span class="op">]</span> <span class="co"># Column 'a' and 'c'</span></code></pre></div>
<pre><code>     a c
[1,] 1 3
[2,] 4 6
[3,] 7 9</code></pre>
<p>Consider a list <code>w</code>.</p>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>vec <span class="op">=</span> <span class="va">a</span>, mat <span class="op">=</span> <span class="va">m</span>, data <span class="op">=</span> <span class="va">df</span>, alist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">b</span>, <span class="va">c</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Indexing by number</span>
<span class="va">w</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="co"># As list structure</span></code></pre></div>
<pre><code>$mat
     a b c
[1,] 1 2 3
[2,] 4 5 6
[3,] 7 8 9</code></pre>
<div class="sourceCode" id="cb141"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">w</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="co"># Without list structure</span></code></pre></div>
<pre><code>     a b c
[1,] 1 2 3
[2,] 4 5 6
[3,] 7 8 9</code></pre>
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Indexing by name</span>
<span class="va">w</span><span class="op">$</span><span class="va">vec</span></code></pre></div>
<pre><code>[1] 1.0 2.0 3.3 2.8 6.7</code></pre>
<div class="sourceCode" id="cb145"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">w</span><span class="op">$</span><span class="va">data</span></code></pre></div>
<pre><code>  L x y          z
1 A 1 1       This
2 B 1 2         is
3 C 1 3          a
4 D 1 4 data frame</code></pre>
<p>Consider a data frame <code>df</code>.</p>
<div class="sourceCode" id="cb147"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span></code></pre></div>
<pre><code>  L x y          z
1 A 1 1       This
2 B 1 2         is
3 C 1 3          a
4 D 1 4 data frame</code></pre>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Indexing by number</span>
<span class="va">df</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="co"># 2nd column of data frame</span></code></pre></div>
<pre><code>[1] 1 1 1 1</code></pre>
<div class="sourceCode" id="cb151"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="co"># 2nd column of data frame</span></code></pre></div>
<pre><code>  x
1 1
2 1
3 1
4 1</code></pre>
<div class="sourceCode" id="cb153"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span><span class="op">[</span><span class="fl">3</span>,<span class="op">]</span> <span class="co"># 3rd row of data frame</span></code></pre></div>
<pre><code>  L x y z
3 C 1 3 a</code></pre>
<div class="sourceCode" id="cb155"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="co"># rows 2,3 of columns 1,2,3 </span></code></pre></div>
<pre><code>  L x y
2 B 1 2
3 C 1 3</code></pre>
<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">]</span> <span class="co"># Element in 2nd column of 2nd row</span></code></pre></div>
<pre><code>[1] 1</code></pre>
<div class="sourceCode" id="cb159"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Indexing by name</span>
<span class="va">df</span><span class="op">$</span><span class="va">L</span></code></pre></div>
<pre><code>[1] "A" "B" "C" "D"</code></pre>
<div class="sourceCode" id="cb161"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span><span class="op">$</span><span class="va">z</span></code></pre></div>
<pre><code>[1] "This"       "is"         "a"          "data frame"</code></pre>
</div>
<div id="help-documentation" class="section level2">
<h2 class="hasAnchor">
<a href="#help-documentation" class="anchor"></a>4.6 Help Documentation</h2>
<p>The help documentation regarding any function can be viewed using the <code><a href="https://rdrr.io/r/utils/Question.html">?</a></code> or <code>help()</code> function. The help documentation shows the default usage of the function including, the arguments that are taken by the function and the type of output object returned (‘Value’).</p>
<div class="sourceCode" id="cb163"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="va">ls</span>
<span class="fu"><a href="https://rdrr.io/r/utils/help.html">help</a></span><span class="op">(</span><span class="va">ls</span><span class="op">)</span>

<span class="op">?</span><span class="va">mean</span>

<span class="op">?</span><span class="va">setwd</span></code></pre></div>
</div>
<div id="pack" class="section level2">
<h2 class="hasAnchor">
<a href="#pack" class="anchor"></a>4.7 Packages</h2>
<p>Packages in <code>R</code> are collections of <code>R</code> functions, data, and compiled code in a well-defined format. They are add-ons which extend the functionality of <code>R</code> and at present, there are <a href="https://cran.r-project.org/web/packages/available_packages_by_name.html">17632</a> packages available for deployment and use at the official repository, the Comprehensive R Archive Network (CRAN).</p>
<p>Valid packages from CRAN can be installed by using the <code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages()</a></code> command.</p>
<div class="sourceCode" id="cb164"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Install the package 'readxl' for importing data from excel</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="va">readxl</span><span class="op">)</span></code></pre></div>
<p>Installed packages can be loaded using the function <code><a href="https://rdrr.io/r/base/library.html">library()</a></code>.</p>
<div class="sourceCode" id="cb165"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Install the package 'readxl' for importing data from excel</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">readxl</span><span class="op">)</span></code></pre></div>
</div>
<div id="impexp" class="section level2">
<h2 class="hasAnchor">
<a href="#impexp" class="anchor"></a>4.8 Importing and Exporting Tabular Data</h2>
<p>Tabular data from a spreadsheet can be imported into <code>R</code> in different ways. Consider some data such as in Table 1. Copy this data in to a spreadsheet editor such as MS Excel and save it as <code>augdata.csv</code>, a comma-separated-value file and <code>augdata.xlsx</code>, an Excel file in the working directory (<code><a href="https://rdrr.io/r/base/getwd.html">getwd()</a></code>).</p>
<p style="text-align: left;">
<strong>Table 1</strong>: Example data from an experiment in augmented RCBD design.
</p>
<table class="table">
<thead><tr class="header">
<th align="left">blk</th>
<th align="right">trt</th>
<th align="right">y1</th>
<th align="right">y2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">I</td>
<td align="right">1</td>
<td align="right">92</td>
<td align="right">258</td>
</tr>
<tr class="even">
<td align="left">I</td>
<td align="right">2</td>
<td align="right">79</td>
<td align="right">224</td>
</tr>
<tr class="odd">
<td align="left">I</td>
<td align="right">3</td>
<td align="right">87</td>
<td align="right">238</td>
</tr>
<tr class="even">
<td align="left">I</td>
<td align="right">4</td>
<td align="right">81</td>
<td align="right">278</td>
</tr>
<tr class="odd">
<td align="left">I</td>
<td align="right">7</td>
<td align="right">96</td>
<td align="right">347</td>
</tr>
<tr class="even">
<td align="left">I</td>
<td align="right">11</td>
<td align="right">89</td>
<td align="right">300</td>
</tr>
<tr class="odd">
<td align="left">I</td>
<td align="right">12</td>
<td align="right">82</td>
<td align="right">289</td>
</tr>
<tr class="even">
<td align="left">II</td>
<td align="right">1</td>
<td align="right">79</td>
<td align="right">260</td>
</tr>
<tr class="odd">
<td align="left">II</td>
<td align="right">2</td>
<td align="right">81</td>
<td align="right">220</td>
</tr>
<tr class="even">
<td align="left">II</td>
<td align="right">3</td>
<td align="right">81</td>
<td align="right">237</td>
</tr>
<tr class="odd">
<td align="left">II</td>
<td align="right">4</td>
<td align="right">91</td>
<td align="right">227</td>
</tr>
<tr class="even">
<td align="left">II</td>
<td align="right">5</td>
<td align="right">79</td>
<td align="right">281</td>
</tr>
<tr class="odd">
<td align="left">II</td>
<td align="right">9</td>
<td align="right">78</td>
<td align="right">311</td>
</tr>
<tr class="even">
<td align="left">III</td>
<td align="right">1</td>
<td align="right">83</td>
<td align="right">250</td>
</tr>
<tr class="odd">
<td align="left">III</td>
<td align="right">2</td>
<td align="right">77</td>
<td align="right">240</td>
</tr>
<tr class="even">
<td align="left">III</td>
<td align="right">3</td>
<td align="right">78</td>
<td align="right">268</td>
</tr>
<tr class="odd">
<td align="left">III</td>
<td align="right">4</td>
<td align="right">78</td>
<td align="right">287</td>
</tr>
<tr class="even">
<td align="left">III</td>
<td align="right">8</td>
<td align="right">70</td>
<td align="right">226</td>
</tr>
<tr class="odd">
<td align="left">III</td>
<td align="right">6</td>
<td align="right">75</td>
<td align="right">395</td>
</tr>
<tr class="even">
<td align="left">III</td>
<td align="right">10</td>
<td align="right">74</td>
<td align="right">450</td>
</tr>
</tbody>
</table>
<p>The <code>augdata.csv</code> file can be imported into <code>R</code> using the <code><a href="https://rdrr.io/r/utils/read.table.html">read.csv()</a></code> function or the <a href="https://readr.tidyverse.org/reference/read_delim.html"><code>read_csv()</code></a> function in the <code>readr</code> package.</p>
<div class="sourceCode" id="cb166"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"augdata.csv"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></code></pre></div>
<pre><code>'data.frame':   20 obs. of  4 variables:
 $ blk: Factor w/ 3 levels "I","II","III": 1 1 1 1 1 1 1 2 2 2 ...
 $ trt: num  1 2 3 4 7 11 12 1 2 3 ...
 $ y1 : num  92 79 87 81 96 89 82 79 81 81 ...
 $ y2 : num  258 224 238 278 347 300 289 260 220 237 ...</code></pre>
<p>The argument <code>stringsAsFactors = FALSE</code> reads the text columns as of type <code>character</code> instead of the default <code>factor</code>.</p>
<div class="sourceCode" id="cb168"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"augdata.csv"</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></code></pre></div>
<pre><code>'data.frame':   20 obs. of  4 variables:
 $ blk: chr  "I" "I" "I" "I" ...
 $ trt: num  1 2 3 4 7 11 12 1 2 3 ...
 $ y1 : num  92 79 87 81 96 89 82 79 81 81 ...
 $ y2 : num  258 224 238 278 347 300 289 260 220 237 ...</code></pre>
<p>The <code>augdata.xlsx</code> file can be imported into <code>R</code> using the <a href="https://readxl.tidyverse.org/reference/read_excel.html"><code>read_excel()</code></a> function in the <code>readxl</code> package.</p>
<div class="sourceCode" id="cb170"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">readxl</span><span class="op">)</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu">read_excel</span><span class="op">(</span>path <span class="op">=</span> <span class="st">"augdata.xlsx"</span><span class="op">)</span></code></pre></div>
<pre><code>'data.frame':   20 obs. of  4 variables:
 $ blk: chr  "I" "I" "I" "I" ...
 $ trt: num  1 2 3 4 7 11 12 1 2 3 ...
 $ y1 : num  92 79 87 81 96 89 82 79 81 81 ...
 $ y2 : num  258 224 238 278 347 300 289 260 220 237 ...</code></pre>
<p>The tabular data can be exported from <code>R</code> to a <code>.csv</code> (comma-separated-value) file by the <a href="https://www.rdocumentation.org/packages/utils/versions/3.5.1/topics/write.table"><code>write.csv()</code></a> function.</p>
<div class="sourceCode" id="cb172"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.csv</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">data</span>, file <span class="op">=</span> <span class="st">"augdata.csv"</span><span class="op">)</span></code></pre></div>
</div>
<div id="additional-resources" class="section level2">
<h2 class="hasAnchor">
<a href="#additional-resources" class="anchor"></a>4.9 Additional Resources</h2>
<p>To learn more about <code>R</code>, there are umpteen number of online tutorials as well as free courses available. Queries about various aspects can be put to the active and vibrant `R community online.</p>
<ul>
<li>Online tutorials
<ul>
<li><a href="http://www.cran.r-project.org/other-docs.html" class="uri">http://www.cran.r-project.org/other-docs.html</a></li>
<li><a href="https://bookdown.org/ndphillips/YaRrr/" class="uri">https://bookdown.org/ndphillips/YaRrr/</a></li>
</ul>
</li>
<li>Free online courses
<ul>
<li><a href="http://tryr.codeschool.com/" class="uri">http://tryr.codeschool.com/</a></li>
<li><a href="https://www.datacamp.com/courses/free-introduction-to-r" class="uri">https://www.datacamp.com/courses/free-introduction-to-r</a></li>
</ul>
</li>
<li>
<code>R</code> community support
<ul>
<li><a href="http://stackoverflow.com/" class="uri">http://stackoverflow.com/</a></li>
<li>
<code>R</code> help mailing lists : <a href="http://www.r-project.org/mail.html" class="uri">http://www.r-project.org/mail.html</a>
</li>
</ul>
</li>
</ul>
</div>
</div>
<div id="install" class="section level1">
<h1 class="hasAnchor">
<a href="#install" class="anchor"></a>5 Installation of <code>augmentedRCBD</code>
</h1>
<p>The package <code>augmentedRCBD</code> can be installed using the following functions.</p>
<div class="sourceCode" id="cb173"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Install from CRAN</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">'augmentedRCBD'</span>, dependencies<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># Install development version from Github</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st">'devtools'</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">'devtools'</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">devtools</span><span class="op">)</span>
<span class="fu">install_github</span><span class="op">(</span><span class="st">"aravind-j/augmentedRCBD"</span><span class="op">)</span></code></pre></div>
<p>The stable release is hosted in <a href="https://CRAN.R-project.org/package=augmentedRCBD">CRAN</a> <a href="#pack">(see section 4.7)</a>, while the under-development version is hosted as a <a href="https://github.com/aravind-j/augmentedRCBD">Github</a> repository. To install from github, you need to use the <a href="https://devtools.r-lib.org/reference/reexports.html"><code>install_github()</code></a> function from <a href="https://devtools.r-lib.org/">`devtools</a> package.</p>
<p>Then the package can be loaded using the function</p>
<div class="sourceCode" id="cb174"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/aravind-j/augmentedRCBD">augmentedRCBD</a></span><span class="op">)</span></code></pre></div>
<pre><code>
--------------------------------------------------------------------------------
Welcome to augmentedRCBD version 0.1.4.9000


# To know how to use this package type:
  browseVignettes(package = 'augmentedRCBD')
  for the package vignette.

# To know whats new in this version type:
  news(package='augmentedRCBD')
  for the NEWS file.

# To cite the methods in the package type:
  citation(package='augmentedRCBD')

# To suppress this message use:
  suppressPackageStartupMessages(library(augmentedRCBD))
--------------------------------------------------------------------------------</code></pre>
<p>The current version of the package is 0.1.4. The previous versions are as follows.</p>
<p><strong>Table 2.</strong> Version history of <code>augmentedRCBD</code> <code>R</code> package.</p>
<pre><code>[1] "Packages 'RCurl', 'httr' and 'XML' are required to generate this table"</code></pre>
<p>To know detailed history of changes use <code><a href="https://rdrr.io/r/utils/news.html">news(package='augmentedRCBD')</a></code>.</p>
</div>
<div id="data-format" class="section level1">
<h1 class="hasAnchor">
<a href="#data-format" class="anchor"></a>6 Data Format</h1>
<p>Certain details need to be considered for arranging experimental data for analysis using the <code>augmentedRCBD</code> package.</p>
<p>The data should be in long/vertical form, where each row has the data from one genotype per block. For example, consider the following data (Table 3) recorded for a trait from an experiment laid out in an augmented block design with 3 blocks and 12 genotypes(or treatment) with 6 to 7 genotypes/block. 8 genotypes (Test, G 5 to G 12) are not replicated, while 4 genotypes (Check, G 1 to G 4) are replicated.</p>
<p style="text-align: left;">
<strong>Table 3</strong>: Data from an experiment in augmented RCBD design.
</p>
<table class="table"><tbody>
<tr class="odd">
<td align="left"><strong>Block I</strong></td>
<td align="left">G12</td>
<td align="left"><strong>G4</strong></td>
<td align="left">G11</td>
<td align="left"><strong>G2</strong></td>
<td align="left"><strong>G1</strong></td>
<td align="left">G7</td>
<td align="left"><strong>G3</strong></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">82</td>
<td align="left">81</td>
<td align="left">89</td>
<td align="left">79</td>
<td align="left">92</td>
<td align="left">96</td>
<td align="left">87</td>
</tr>
<tr class="odd">
<td align="left"><strong>Block II</strong></td>
<td align="left">G5</td>
<td align="left">G9</td>
<td align="left">–</td>
<td align="left"><strong>G3</strong></td>
<td align="left"><strong>G1</strong></td>
<td align="left"><strong>G2</strong></td>
<td align="left"><strong>G4</strong></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">79</td>
<td align="left">78</td>
<td align="left">–</td>
<td align="left">81</td>
<td align="left">79</td>
<td align="left">81</td>
<td align="left">91</td>
</tr>
<tr class="odd">
<td align="left"><strong>Block III</strong></td>
<td align="left"><strong>G4</strong></td>
<td align="left"><strong>G2</strong></td>
<td align="left"><strong>G1</strong></td>
<td align="left">G6</td>
<td align="left">G10</td>
<td align="left"><strong>G3</strong></td>
<td align="left">G8</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">78</td>
<td align="left">77</td>
<td align="left">83</td>
<td align="left">75</td>
<td align="left">74</td>
<td align="left">78</td>
<td align="left">70</td>
</tr>
</tbody></table>
<p>This data needs to be arranged with columns showing block, genotype (or treatment) and the data of the trait for each genotype per block (Table 4).</p>
<p style="text-align: left;">
<strong>Table 4</strong>: Data from an experiment in augmented RCBD design arranged in long-form.
</p>
<table class="table">
<thead><tr class="header">
<th align="left">Block</th>
<th align="left">Treatment</th>
<th align="right">Trait</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 1</td>
<td align="right">92</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 2</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 3</td>
<td align="right">87</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 4</td>
<td align="right">81</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 7</td>
<td align="right">96</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 11</td>
<td align="right">89</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 12</td>
<td align="right">82</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 1</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 2</td>
<td align="right">81</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 3</td>
<td align="right">81</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 4</td>
<td align="right">91</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 5</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 9</td>
<td align="right">78</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 1</td>
<td align="right">83</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 2</td>
<td align="right">77</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 3</td>
<td align="right">78</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 4</td>
<td align="right">78</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 8</td>
<td align="right">70</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 6</td>
<td align="right">75</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 10</td>
<td align="right">74</td>
</tr>
</tbody>
</table>
<p>The data for block and genotype (or treatment) can also be depicted as numbers (Table 5).</p>
<p style="text-align: left;">
<strong>Table 5</strong>: Data from an experiment in augmented RCBD design arranged in long-form (Block and Treatment as numbers).
</p>
<table class="table">
<thead><tr class="header">
<th align="right">Block</th>
<th align="right">Treatment</th>
<th align="right">Trait</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="right">1</td>
<td align="right">92</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">2</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">3</td>
<td align="right">87</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">4</td>
<td align="right">81</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">7</td>
<td align="right">96</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">11</td>
<td align="right">89</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">12</td>
<td align="right">82</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">2</td>
<td align="right">81</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">3</td>
<td align="right">81</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">4</td>
<td align="right">91</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">5</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">9</td>
<td align="right">78</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">1</td>
<td align="right">83</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">2</td>
<td align="right">77</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">3</td>
<td align="right">78</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">4</td>
<td align="right">78</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">8</td>
<td align="right">70</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">6</td>
<td align="right">75</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">10</td>
<td align="right">74</td>
</tr>
</tbody>
</table>
<p>Multiple traits can be added as additional columns (Table 6).</p>
<p style="text-align: left;">
<strong>Table 6</strong>: Data from an experiment in augmented RCBD design arranged in long-form (Multiple traits).
</p>
<table class="table">
<thead><tr class="header">
<th align="left">Block</th>
<th align="left">Treatment</th>
<th align="right">Trait1</th>
<th align="right">Trait2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 1</td>
<td align="right">92</td>
<td align="right">258</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 2</td>
<td align="right">79</td>
<td align="right">224</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 3</td>
<td align="right">87</td>
<td align="right">238</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 4</td>
<td align="right">81</td>
<td align="right">278</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 7</td>
<td align="right">96</td>
<td align="right">347</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 11</td>
<td align="right">89</td>
<td align="right">300</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 12</td>
<td align="right">82</td>
<td align="right">289</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 1</td>
<td align="right">79</td>
<td align="right">260</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 2</td>
<td align="right">81</td>
<td align="right">220</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 3</td>
<td align="right">81</td>
<td align="right">237</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 4</td>
<td align="right">91</td>
<td align="right">227</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 5</td>
<td align="right">79</td>
<td align="right">281</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 9</td>
<td align="right">78</td>
<td align="right">311</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 1</td>
<td align="right">83</td>
<td align="right">250</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 2</td>
<td align="right">77</td>
<td align="right">240</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 3</td>
<td align="right">78</td>
<td align="right">268</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 4</td>
<td align="right">78</td>
<td align="right">287</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 8</td>
<td align="right">70</td>
<td align="right">226</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 6</td>
<td align="right">75</td>
<td align="right">395</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 10</td>
<td align="right">74</td>
<td align="right">450</td>
</tr>
</tbody>
</table>
<p>Data should preferably be balanced i.e. all the check genotypes should be present in all the blocks. If not, a warning is issued. The number of test genotypes can vary within a block. There should not be any missing values. Rows of genotypes with missing values for one or more traits should be removed.</p>
<p>Such a tabular data should be imported (<a href="#impexp">see section 7.8</a>) into <code>R</code> as a data frame object (<a href="#dataframe">see section 4.3.5</a>). The columns with the block and treatment categorical data should of the type factor (<a href="#factor">see section 4.3.2</a>), while the column(s) with the trait data should be of the type integer or numeric (<a href="#vector">see section 4.3.1</a>).</p>
</div>
<div id="data-analysis-for-a-single-trait" class="section level1">
<h1 class="hasAnchor">
<a href="#data-analysis-for-a-single-trait" class="anchor"></a>7 Data Analysis for a Single Trait</h1>
<p>Analysis of data for a single trait can be performed by using <code>augmentedRCBD</code> function. It generates an object of class <code>augmentedRCBD</code>. Such an object can then be taken as input by the several functions to print the results to console (<code>print.augmentedRCBD</code>), generate descriptive statistics from adjusted means (<code>describe.augmentedRCBD</code>), plot frequency distribution (<code>freqdist.augmentedRCBD</code>) and computed genetic variability statistics (gva.augmentedRCBD). All these outputs can also be exported as a MS Word report using the <code>report.augmentedRCBD</code> function.</p>
<pre><code>[1] "package 'diagram' is required to generate this figure"</code></pre>
<p><strong>Fig. 4</strong>. Workflow for analysis of single traits with <code>augmentedRCBD</code>.</p>
<div id="augmentedrcbd" class="section level2">
<h2 class="hasAnchor">
<a href="#augmentedrcbd" class="anchor"></a>7.1 <code>augmentedRCBD()</code>
</h2>
<p>Consider the data in <a href="#impexp">Table 1</a>. The data can be imported into <code>R</code> as <a href="#vector">vectors</a> as follows.</p>
<div class="sourceCode" id="cb178"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">blk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span>
<span class="va">trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">11</span>, <span class="fl">12</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">9</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">8</span>, <span class="fl">6</span>, <span class="fl">10</span><span class="op">)</span>
<span class="va">y1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">92</span>, <span class="fl">79</span>, <span class="fl">87</span>, <span class="fl">81</span>, <span class="fl">96</span>, <span class="fl">89</span>, <span class="fl">82</span>, <span class="fl">79</span>, <span class="fl">81</span>, <span class="fl">81</span>, <span class="fl">91</span>, <span class="fl">79</span>, <span class="fl">78</span>, <span class="fl">83</span>, <span class="fl">77</span>, <span class="fl">78</span>, <span class="fl">78</span>,
        <span class="fl">70</span>, <span class="fl">75</span>, <span class="fl">74</span><span class="op">)</span>
<span class="va">y2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">258</span>, <span class="fl">224</span>, <span class="fl">238</span>, <span class="fl">278</span>, <span class="fl">347</span>, <span class="fl">300</span>, <span class="fl">289</span>, <span class="fl">260</span>, <span class="fl">220</span>, <span class="fl">237</span>, <span class="fl">227</span>, <span class="fl">281</span>, <span class="fl">311</span>, <span class="fl">250</span>,
        <span class="fl">240</span>, <span class="fl">268</span>, <span class="fl">287</span>, <span class="fl">226</span>, <span class="fl">395</span>, <span class="fl">450</span><span class="op">)</span></code></pre></div>
<p>The <code>blk</code> and <code>trt</code> vectors with the block and treatment data need to be converted into factors as follows before analysis.</p>
<div class="sourceCode" id="cb179"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Convert block and treatment to factors</span>
<span class="va">blk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">blk</span><span class="op">)</span>
<span class="va">trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">trt</span><span class="op">)</span></code></pre></div>
<p>With the data in appropriate format, the analysis can be performed as follows for the trait <code>y1</code> as follows.</p>
<div class="sourceCode" id="cb180"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">out1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">blk</span>, <span class="va">trt</span>, <span class="va">y1</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,
                      alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"

ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value Pr(&gt;F)  
Block (ignoring Treatments)           2  360.1  180.04   6.675 0.0298 *
Treatment (eliminating Blocks)       11  285.1   25.92   0.961 0.5499  
  Treatment: Check                    3   52.9   17.64   0.654 0.6092  
  Treatment: Test and Test vs. Check  8  232.2   29.02   1.076 0.4779  
Residuals                             6  161.8   26.97                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value Pr(&gt;F)
Treatment (ignoring Blocks)    11  575.7   52.33   1.940  0.215
  Treatment: Check              3   52.9   17.64   0.654  0.609
  Treatment: Test               7  505.9   72.27   2.679  0.125
  Treatment: Test vs. Check     1   16.9   16.87   0.626  0.459
Block (eliminating Treatments)  2   69.5   34.75   1.288  0.342
Residuals                       6  161.8   26.97               

Treatment Means
===============
   Treatment Block    Means       SE r Min Max Adjusted Means
1          1       84.66667 3.844188 3  79  92       84.66667
2         10     3 74.00000       NA 1  74  74       77.25000
3         11     1 89.00000       NA 1  89  89       86.50000
4         12     1 82.00000       NA 1  82  82       79.50000
5          2       79.00000 1.154701 3  77  81       79.00000
6          3       82.00000 2.645751 3  78  87       82.00000
7          4       83.33333 3.929942 3  78  91       83.33333
8          5     2 79.00000       NA 1  79  79       78.25000
9          6     3 75.00000       NA 1  75  75       78.25000
10         7     1 96.00000       NA 1  96  96       93.50000
11         8     3 70.00000       NA 1  70  70       73.25000
12         9     2 78.00000       NA 1  78  78       77.25000

Coefficient of Variation
========================
6.372367

Overall Adjusted Mean
=====================
81.0625

Standard Errors
===================
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             4.240458 10.37603
Two Test Treatments (Same Block)                    7.344688 17.97180
Two Test Treatments (Different Blocks)              8.211611 20.09309
A Test Treatment and a Control Treatment            6.704752 16.40594

Treatment Groups
==================

Method : lsd

   Treatment Adjusted Means       SE df lower.CL  upper.CL Group
8          8       73.25000 5.609598  6 59.52381  86.97619    1 
9          9       77.25000 5.609598  6 63.52381  90.97619    12
10        10       77.25000 5.609598  6 63.52381  90.97619    12
5          5       78.25000 5.609598  6 64.52381  91.97619    12
6          6       78.25000 5.609598  6 64.52381  91.97619    12
2          2       79.00000 2.998456  6 71.66304  86.33696    12
12        12       79.50000 5.609598  6 65.77381  93.22619    12
3          3       82.00000 2.998456  6 74.66304  89.33696    12
4          4       83.33333 2.998456  6 75.99637  90.67029    12
1          1       84.66667 2.998456  6 77.32971  92.00363    12
11        11       86.50000 5.609598  6 72.77381 100.22619    12
7          7       93.50000 5.609598  6 79.77381 107.22619     2</code></pre>
<div class="sourceCode" id="cb182"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">out1</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "augmentedRCBD"</code></pre>
<p>Similarly the analysis for the trait <code>y2</code> can be computed as follows.</p>
<div class="sourceCode" id="cb184"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">out2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">blk</span>, <span class="va">trt</span>, <span class="va">y2</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,
                      alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"

ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Block (ignoring Treatments)           2   7019    3510  12.261 0.007597 ** 
Treatment (eliminating Blocks)       11  58965    5360  18.727 0.000920 ***
  Treatment: Check                    3   2150     717   2.504 0.156116    
  Treatment: Test and Test vs. Check  8  56815    7102  24.810 0.000473 ***
Residuals                             6   1717     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Treatment (ignoring Blocks)    11  64708    5883  20.550 0.000707 ***
  Treatment: Check              3   2150     717   2.504 0.156116    
  Treatment: Test               7  34863    4980  17.399 0.001366 ** 
  Treatment: Test vs. Check     1  27694   27694  96.749 6.36e-05 ***
Block (eliminating Treatments)  2   1277     639   2.231 0.188645    
Residuals                       6   1718     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Treatment Means
===============
   Treatment Block    Means        SE r Min Max Adjusted Means
1          1       256.0000  3.055050 3 250 260       256.0000
2         10     3 450.0000        NA 1 450 450       437.6667
3         11     1 300.0000        NA 1 300 300       299.4167
4         12     1 289.0000        NA 1 289 289       288.4167
5          2       228.0000  6.110101 3 220 240       228.0000
6          3       247.6667 10.170764 3 237 268       247.6667
7          4       264.0000 18.681542 3 227 287       264.0000
8          5     2 281.0000        NA 1 281 281       293.9167
9          6     3 395.0000        NA 1 395 395       382.6667
10         7     1 347.0000        NA 1 347 347       346.4167
11         8     3 226.0000        NA 1 226 226       213.6667
12         9     2 311.0000        NA 1 311 311       323.9167

Coefficient of Variation
========================
6.057617

Overall Adjusted Mean
=====================
298.4792

Standard Errors
===================
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             13.81424 33.80224
Two Test Treatments (Same Block)                    23.92697 58.54719
Two Test Treatments (Different Blocks)              26.75117 65.45775
A Test Treatment and a Control Treatment            21.84224 53.44603

Treatment Groups
==================

Method : lsd

   Treatment Adjusted Means        SE df lower.CL upper.CL    Group
8          8       213.6667 18.274527  6 168.9505 258.3828  12     
2          2       228.0000  9.768146  6 204.0982 251.9018  1      
3          3       247.6667  9.768146  6 223.7649 271.5685  123    
1          1       256.0000  9.768146  6 232.0982 279.9018  1234   
4          4       264.0000  9.768146  6 240.0982 287.9018   234   
12        12       288.4167 18.274527  6 243.7005 333.1328    345  
5          5       293.9167 18.274527  6 249.2005 338.6328    345  
11        11       299.4167 18.274527  6 254.7005 344.1328     45  
9          9       323.9167 18.274527  6 279.2005 368.6328      56 
7          7       346.4167 18.274527  6 301.7005 391.1328      56 
6          6       382.6667 18.274527  6 337.9505 427.3828       67
10        10       437.6667 18.274527  6 392.9505 482.3828        7</code></pre>
<div class="sourceCode" id="cb186"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">out2</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "augmentedRCBD"</code></pre>
<p>The data can also be imported as a <a href="#dataframe">data frame</a> and then used for analysis. Consider the data frame <code>data</code> imported from <a href="#impexp">Table 1</a> according to the instructions in <a href="#impexp">section 4.8</a>.</p>
<div class="sourceCode" id="cb188"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></code></pre></div>
<pre><code>'data.frame':   20 obs. of  4 variables:
 $ blk: Factor w/ 3 levels "1","2","3": 1 1 1 1 1 1 1 2 2 2 ...
 $ trt: Factor w/ 12 levels "1","2","3","4",..: 1 2 3 4 7 11 12 1 2 3 ...
 $ y1 : num  92 79 87 81 96 89 82 79 81 81 ...
 $ y2 : num  258 224 238 278 347 300 289 260 220 237 ...</code></pre>
<div class="sourceCode" id="cb190"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Convert block and treatment to factors</span>
<span class="va">data</span><span class="op">$</span><span class="va">blk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span><span class="op">)</span>
<span class="va">data</span><span class="op">$</span><span class="va">trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">trt</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb191"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Results for variable y1</span>
<span class="va">out1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span>, <span class="va">data</span><span class="op">$</span><span class="va">trt</span>, <span class="va">data</span><span class="op">$</span><span class="va">y1</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,
                      alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"

ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value Pr(&gt;F)  
Block (ignoring Treatments)           2  360.1  180.04   6.675 0.0298 *
Treatment (eliminating Blocks)       11  285.1   25.92   0.961 0.5499  
  Treatment: Check                    3   52.9   17.64   0.654 0.6092  
  Treatment: Test and Test vs. Check  8  232.2   29.02   1.076 0.4779  
Residuals                             6  161.8   26.97                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value Pr(&gt;F)
Treatment (ignoring Blocks)    11  575.7   52.33   1.940  0.215
  Treatment: Check              3   52.9   17.64   0.654  0.609
  Treatment: Test               7  505.9   72.27   2.679  0.125
  Treatment: Test vs. Check     1   16.9   16.87   0.626  0.459
Block (eliminating Treatments)  2   69.5   34.75   1.288  0.342
Residuals                       6  161.8   26.97               

Treatment Means
===============
   Treatment Block    Means       SE r Min Max Adjusted Means
1          1       84.66667 3.844188 3  79  92       84.66667
2         10     3 74.00000       NA 1  74  74       77.25000
3         11     1 89.00000       NA 1  89  89       86.50000
4         12     1 82.00000       NA 1  82  82       79.50000
5          2       79.00000 1.154701 3  77  81       79.00000
6          3       82.00000 2.645751 3  78  87       82.00000
7          4       83.33333 3.929942 3  78  91       83.33333
8          5     2 79.00000       NA 1  79  79       78.25000
9          6     3 75.00000       NA 1  75  75       78.25000
10         7     1 96.00000       NA 1  96  96       93.50000
11         8     3 70.00000       NA 1  70  70       73.25000
12         9     2 78.00000       NA 1  78  78       77.25000

Coefficient of Variation
========================
6.372367

Overall Adjusted Mean
=====================
81.0625

Standard Errors
===================
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             4.240458 10.37603
Two Test Treatments (Same Block)                    7.344688 17.97180
Two Test Treatments (Different Blocks)              8.211611 20.09309
A Test Treatment and a Control Treatment            6.704752 16.40594

Treatment Groups
==================

Method : lsd

   Treatment Adjusted Means       SE df lower.CL  upper.CL Group
8          8       73.25000 5.609598  6 59.52381  86.97619    1 
9          9       77.25000 5.609598  6 63.52381  90.97619    12
10        10       77.25000 5.609598  6 63.52381  90.97619    12
5          5       78.25000 5.609598  6 64.52381  91.97619    12
6          6       78.25000 5.609598  6 64.52381  91.97619    12
2          2       79.00000 2.998456  6 71.66304  86.33696    12
12        12       79.50000 5.609598  6 65.77381  93.22619    12
3          3       82.00000 2.998456  6 74.66304  89.33696    12
4          4       83.33333 2.998456  6 75.99637  90.67029    12
1          1       84.66667 2.998456  6 77.32971  92.00363    12
11        11       86.50000 5.609598  6 72.77381 100.22619    12
7          7       93.50000 5.609598  6 79.77381 107.22619     2</code></pre>
<div class="sourceCode" id="cb193"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">out1</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "augmentedRCBD"</code></pre>
<div class="sourceCode" id="cb195"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Results for variable y2</span>
<span class="va">out2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span>, <span class="va">data</span><span class="op">$</span><span class="va">trt</span>, <span class="va">data</span><span class="op">$</span><span class="va">y2</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,
                     alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"

ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Block (ignoring Treatments)           2   7019    3510  12.261 0.007597 ** 
Treatment (eliminating Blocks)       11  58965    5360  18.727 0.000920 ***
  Treatment: Check                    3   2150     717   2.504 0.156116    
  Treatment: Test and Test vs. Check  8  56815    7102  24.810 0.000473 ***
Residuals                             6   1717     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Treatment (ignoring Blocks)    11  64708    5883  20.550 0.000707 ***
  Treatment: Check              3   2150     717   2.504 0.156116    
  Treatment: Test               7  34863    4980  17.399 0.001366 ** 
  Treatment: Test vs. Check     1  27694   27694  96.749 6.36e-05 ***
Block (eliminating Treatments)  2   1277     639   2.231 0.188645    
Residuals                       6   1718     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Treatment Means
===============
   Treatment Block    Means        SE r Min Max Adjusted Means
1          1       256.0000  3.055050 3 250 260       256.0000
2         10     3 450.0000        NA 1 450 450       437.6667
3         11     1 300.0000        NA 1 300 300       299.4167
4         12     1 289.0000        NA 1 289 289       288.4167
5          2       228.0000  6.110101 3 220 240       228.0000
6          3       247.6667 10.170764 3 237 268       247.6667
7          4       264.0000 18.681542 3 227 287       264.0000
8          5     2 281.0000        NA 1 281 281       293.9167
9          6     3 395.0000        NA 1 395 395       382.6667
10         7     1 347.0000        NA 1 347 347       346.4167
11         8     3 226.0000        NA 1 226 226       213.6667
12         9     2 311.0000        NA 1 311 311       323.9167

Coefficient of Variation
========================
6.057617

Overall Adjusted Mean
=====================
298.4792

Standard Errors
===================
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             13.81424 33.80224
Two Test Treatments (Same Block)                    23.92697 58.54719
Two Test Treatments (Different Blocks)              26.75117 65.45775
A Test Treatment and a Control Treatment            21.84224 53.44603

Treatment Groups
==================

Method : lsd

   Treatment Adjusted Means        SE df lower.CL upper.CL    Group
8          8       213.6667 18.274527  6 168.9505 258.3828  12     
2          2       228.0000  9.768146  6 204.0982 251.9018  1      
3          3       247.6667  9.768146  6 223.7649 271.5685  123    
1          1       256.0000  9.768146  6 232.0982 279.9018  1234   
4          4       264.0000  9.768146  6 240.0982 287.9018   234   
12        12       288.4167 18.274527  6 243.7005 333.1328    345  
5          5       293.9167 18.274527  6 249.2005 338.6328    345  
11        11       299.4167 18.274527  6 254.7005 344.1328     45  
9          9       323.9167 18.274527  6 279.2005 368.6328      56 
7          7       346.4167 18.274527  6 301.7005 391.1328      56 
6          6       382.6667 18.274527  6 337.9505 427.3828       67
10        10       437.6667 18.274527  6 392.9505 482.3828        7</code></pre>
<div class="sourceCode" id="cb197"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">out2</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "augmentedRCBD"</code></pre>
<p>Check genotypes are inferred by default on the basis of number of replications. However, if some test genotypes are also replicated, they may also be falsely detected as checks. To avoid this, the checks can be specified by the <code>checks</code> argument.</p>
<div class="sourceCode" id="cb199"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Results for variable y1 (checks specified)</span>
<span class="va">out1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span>, <span class="va">data</span><span class="op">$</span><span class="va">trt</span>, <span class="va">data</span><span class="op">$</span><span class="va">y1</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,
                      alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span>,
                      checks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"

ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value Pr(&gt;F)  
Block (ignoring Treatments)           2  360.1  180.04   6.675 0.0298 *
Treatment (eliminating Blocks)       11  285.1   25.92   0.961 0.5499  
  Treatment: Check                    3   52.9   17.64   0.654 0.6092  
  Treatment: Test and Test vs. Check  8  232.2   29.02   1.076 0.4779  
Residuals                             6  161.8   26.97                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value Pr(&gt;F)
Treatment (ignoring Blocks)    11  575.7   52.33   1.940  0.215
  Treatment: Check              3   52.9   17.64   0.654  0.609
  Treatment: Test               7  505.9   72.27   2.679  0.125
  Treatment: Test vs. Check     1   16.9   16.87   0.626  0.459
Block (eliminating Treatments)  2   69.5   34.75   1.288  0.342
Residuals                       6  161.8   26.97               

Treatment Means
===============
   Treatment Block    Means       SE r Min Max Adjusted Means
1          1       84.66667 3.844188 3  79  92       84.66667
2         10     3 74.00000       NA 1  74  74       77.25000
3         11     1 89.00000       NA 1  89  89       86.50000
4         12     1 82.00000       NA 1  82  82       79.50000
5          2       79.00000 1.154701 3  77  81       79.00000
6          3       82.00000 2.645751 3  78  87       82.00000
7          4       83.33333 3.929942 3  78  91       83.33333
8          5     2 79.00000       NA 1  79  79       78.25000
9          6     3 75.00000       NA 1  75  75       78.25000
10         7     1 96.00000       NA 1  96  96       93.50000
11         8     3 70.00000       NA 1  70  70       73.25000
12         9     2 78.00000       NA 1  78  78       77.25000

Coefficient of Variation
========================
6.372367

Overall Adjusted Mean
=====================
81.0625

Standard Errors
===================
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             4.240458 10.37603
Two Test Treatments (Same Block)                    7.344688 17.97180
Two Test Treatments (Different Blocks)              8.211611 20.09309
A Test Treatment and a Control Treatment            6.704752 16.40594

Treatment Groups
==================

Method : lsd

   Treatment Adjusted Means       SE df lower.CL  upper.CL Group
8          8       73.25000 5.609598  6 59.52381  86.97619    1 
9          9       77.25000 5.609598  6 63.52381  90.97619    12
10        10       77.25000 5.609598  6 63.52381  90.97619    12
5          5       78.25000 5.609598  6 64.52381  91.97619    12
6          6       78.25000 5.609598  6 64.52381  91.97619    12
2          2       79.00000 2.998456  6 71.66304  86.33696    12
12        12       79.50000 5.609598  6 65.77381  93.22619    12
3          3       82.00000 2.998456  6 74.66304  89.33696    12
4          4       83.33333 2.998456  6 75.99637  90.67029    12
1          1       84.66667 2.998456  6 77.32971  92.00363    12
11        11       86.50000 5.609598  6 72.77381 100.22619    12
7          7       93.50000 5.609598  6 79.77381 107.22619     2</code></pre>
<div class="sourceCode" id="cb201"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Results for variable y2 (checks specified)</span>
<span class="va">out2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span>, <span class="va">data</span><span class="op">$</span><span class="va">trt</span>, <span class="va">data</span><span class="op">$</span><span class="va">y2</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,
                      alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span>,
                      checks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"

ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Block (ignoring Treatments)           2   7019    3510  12.261 0.007597 ** 
Treatment (eliminating Blocks)       11  58965    5360  18.727 0.000920 ***
  Treatment: Check                    3   2150     717   2.504 0.156116    
  Treatment: Test and Test vs. Check  8  56815    7102  24.810 0.000473 ***
Residuals                             6   1717     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Treatment (ignoring Blocks)    11  64708    5883  20.550 0.000707 ***
  Treatment: Check              3   2150     717   2.504 0.156116    
  Treatment: Test               7  34863    4980  17.399 0.001366 ** 
  Treatment: Test vs. Check     1  27694   27694  96.749 6.36e-05 ***
Block (eliminating Treatments)  2   1277     639   2.231 0.188645    
Residuals                       6   1718     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Treatment Means
===============
   Treatment Block    Means        SE r Min Max Adjusted Means
1          1       256.0000  3.055050 3 250 260       256.0000
2         10     3 450.0000        NA 1 450 450       437.6667
3         11     1 300.0000        NA 1 300 300       299.4167
4         12     1 289.0000        NA 1 289 289       288.4167
5          2       228.0000  6.110101 3 220 240       228.0000
6          3       247.6667 10.170764 3 237 268       247.6667
7          4       264.0000 18.681542 3 227 287       264.0000
8          5     2 281.0000        NA 1 281 281       293.9167
9          6     3 395.0000        NA 1 395 395       382.6667
10         7     1 347.0000        NA 1 347 347       346.4167
11         8     3 226.0000        NA 1 226 226       213.6667
12         9     2 311.0000        NA 1 311 311       323.9167

Coefficient of Variation
========================
6.057617

Overall Adjusted Mean
=====================
298.4792

Standard Errors
===================
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             13.81424 33.80224
Two Test Treatments (Same Block)                    23.92697 58.54719
Two Test Treatments (Different Blocks)              26.75117 65.45775
A Test Treatment and a Control Treatment            21.84224 53.44603

Treatment Groups
==================

Method : lsd

   Treatment Adjusted Means        SE df lower.CL upper.CL    Group
8          8       213.6667 18.274527  6 168.9505 258.3828  12     
2          2       228.0000  9.768146  6 204.0982 251.9018  1      
3          3       247.6667  9.768146  6 223.7649 271.5685  123    
1          1       256.0000  9.768146  6 232.0982 279.9018  1234   
4          4       264.0000  9.768146  6 240.0982 287.9018   234   
12        12       288.4167 18.274527  6 243.7005 333.1328    345  
5          5       293.9167 18.274527  6 249.2005 338.6328    345  
11        11       299.4167 18.274527  6 254.7005 344.1328     45  
9          9       323.9167 18.274527  6 279.2005 368.6328      56 
7          7       346.4167 18.274527  6 301.7005 391.1328      56 
6          6       382.6667 18.274527  6 337.9505 427.3828       67
10        10       437.6667 18.274527  6 392.9505 482.3828        7</code></pre>
<p>In case the large number of treatments or genotypes, it is advisable to avoid treatment comparisons with the <code>group = FALSE</code> argument as it will be memory and processor intensive. Further it is advised to simplify output with <code>simplify = TRUE</code> in order to reduce output object size.</p>
<p>If <code>truncate.means = TRUE</code>, then any negative adjusted means will be truncated to zero with a warning.</p>
</div>
<div id="print-augmentedrcbd" class="section level2">
<h2 class="hasAnchor">
<a href="#print-augmentedrcbd" class="anchor"></a>7.2 <code>print.augmentedRCBD()</code>
</h2>
<p>The results of analysis in an object of class <code>augmentedRCBD</code> can be printed to the console as follows.</p>
<div class="sourceCode" id="cb203"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Print results for variable y1</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">out1</span><span class="op">)</span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"

ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value Pr(&gt;F)  
Block (ignoring Treatments)           2  360.1  180.04   6.675 0.0298 *
Treatment (eliminating Blocks)       11  285.1   25.92   0.961 0.5499  
  Treatment: Check                    3   52.9   17.64   0.654 0.6092  
  Treatment: Test and Test vs. Check  8  232.2   29.02   1.076 0.4779  
Residuals                             6  161.8   26.97                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value Pr(&gt;F)
Treatment (ignoring Blocks)    11  575.7   52.33   1.940  0.215
  Treatment: Check              3   52.9   17.64   0.654  0.609
  Treatment: Test               7  505.9   72.27   2.679  0.125
  Treatment: Test vs. Check     1   16.9   16.87   0.626  0.459
Block (eliminating Treatments)  2   69.5   34.75   1.288  0.342
Residuals                       6  161.8   26.97               

Treatment Means
===============
   Treatment Block    Means       SE r Min Max Adjusted Means
1          1       84.66667 3.844188 3  79  92       84.66667
2         10     3 74.00000       NA 1  74  74       77.25000
3         11     1 89.00000       NA 1  89  89       86.50000
4         12     1 82.00000       NA 1  82  82       79.50000
5          2       79.00000 1.154701 3  77  81       79.00000
6          3       82.00000 2.645751 3  78  87       82.00000
7          4       83.33333 3.929942 3  78  91       83.33333
8          5     2 79.00000       NA 1  79  79       78.25000
9          6     3 75.00000       NA 1  75  75       78.25000
10         7     1 96.00000       NA 1  96  96       93.50000
11         8     3 70.00000       NA 1  70  70       73.25000
12         9     2 78.00000       NA 1  78  78       77.25000

Coefficient of Variation
========================
6.372367

Overall Adjusted Mean
=====================
81.0625

Standard Errors
===================
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             4.240458 10.37603
Two Test Treatments (Same Block)                    7.344688 17.97180
Two Test Treatments (Different Blocks)              8.211611 20.09309
A Test Treatment and a Control Treatment            6.704752 16.40594

Treatment Groups
==================

Method : lsd

   Treatment Adjusted Means       SE df lower.CL  upper.CL Group
8          8       73.25000 5.609598  6 59.52381  86.97619    1 
9          9       77.25000 5.609598  6 63.52381  90.97619    12
10        10       77.25000 5.609598  6 63.52381  90.97619    12
5          5       78.25000 5.609598  6 64.52381  91.97619    12
6          6       78.25000 5.609598  6 64.52381  91.97619    12
2          2       79.00000 2.998456  6 71.66304  86.33696    12
12        12       79.50000 5.609598  6 65.77381  93.22619    12
3          3       82.00000 2.998456  6 74.66304  89.33696    12
4          4       83.33333 2.998456  6 75.99637  90.67029    12
1          1       84.66667 2.998456  6 77.32971  92.00363    12
11        11       86.50000 5.609598  6 72.77381 100.22619    12
7          7       93.50000 5.609598  6 79.77381 107.22619     2</code></pre>
<div class="sourceCode" id="cb205"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Print results for variable y2</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">out2</span><span class="op">)</span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"

ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Block (ignoring Treatments)           2   7019    3510  12.261 0.007597 ** 
Treatment (eliminating Blocks)       11  58965    5360  18.727 0.000920 ***
  Treatment: Check                    3   2150     717   2.504 0.156116    
  Treatment: Test and Test vs. Check  8  56815    7102  24.810 0.000473 ***
Residuals                             6   1717     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Treatment (ignoring Blocks)    11  64708    5883  20.550 0.000707 ***
  Treatment: Check              3   2150     717   2.504 0.156116    
  Treatment: Test               7  34863    4980  17.399 0.001366 ** 
  Treatment: Test vs. Check     1  27694   27694  96.749 6.36e-05 ***
Block (eliminating Treatments)  2   1277     639   2.231 0.188645    
Residuals                       6   1718     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Treatment Means
===============
   Treatment Block    Means        SE r Min Max Adjusted Means
1          1       256.0000  3.055050 3 250 260       256.0000
2         10     3 450.0000        NA 1 450 450       437.6667
3         11     1 300.0000        NA 1 300 300       299.4167
4         12     1 289.0000        NA 1 289 289       288.4167
5          2       228.0000  6.110101 3 220 240       228.0000
6          3       247.6667 10.170764 3 237 268       247.6667
7          4       264.0000 18.681542 3 227 287       264.0000
8          5     2 281.0000        NA 1 281 281       293.9167
9          6     3 395.0000        NA 1 395 395       382.6667
10         7     1 347.0000        NA 1 347 347       346.4167
11         8     3 226.0000        NA 1 226 226       213.6667
12         9     2 311.0000        NA 1 311 311       323.9167

Coefficient of Variation
========================
6.057617

Overall Adjusted Mean
=====================
298.4792

Standard Errors
===================
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             13.81424 33.80224
Two Test Treatments (Same Block)                    23.92697 58.54719
Two Test Treatments (Different Blocks)              26.75117 65.45775
A Test Treatment and a Control Treatment            21.84224 53.44603

Treatment Groups
==================

Method : lsd

   Treatment Adjusted Means        SE df lower.CL upper.CL    Group
8          8       213.6667 18.274527  6 168.9505 258.3828  12     
2          2       228.0000  9.768146  6 204.0982 251.9018  1      
3          3       247.6667  9.768146  6 223.7649 271.5685  123    
1          1       256.0000  9.768146  6 232.0982 279.9018  1234   
4          4       264.0000  9.768146  6 240.0982 287.9018   234   
12        12       288.4167 18.274527  6 243.7005 333.1328    345  
5          5       293.9167 18.274527  6 249.2005 338.6328    345  
11        11       299.4167 18.274527  6 254.7005 344.1328     45  
9          9       323.9167 18.274527  6 279.2005 368.6328      56 
7          7       346.4167 18.274527  6 301.7005 391.1328      56 
6          6       382.6667 18.274527  6 337.9505 427.3828       67
10        10       437.6667 18.274527  6 392.9505 482.3828        7</code></pre>
</div>
<div id="describe-augmentedrcbd" class="section level2">
<h2 class="hasAnchor">
<a href="#describe-augmentedrcbd" class="anchor"></a>7.3 <code>describe.augmentedRCBD()</code>
</h2>
<p>The descriptive statistics such as count, mean, standard error, minimum, maximum, skewness ( with p-value from D’Agostino test of skewness (<span class="citation">D’Agostino (<a href="#ref-dagostino_transformation_1970" role="doc-biblioref">1970</a>)</span>)) and kurtosis (with p-value from Anscombe-Glynn test of kurtosis (<span class="citation">Anscombe and Glynn (<a href="#ref-anscombe_distribution_1983" role="doc-biblioref">1983</a>)</span>)) for the adjusted means from the results in an object of class <code>augmentedRCBD</code> can be computed as follows.</p>
<div class="sourceCode" id="cb207"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Descriptive statistics for variable y1</span>
<span class="fu"><a href="../reference/describe.augmentedRCBD.html">describe.augmentedRCBD</a></span><span class="op">(</span><span class="va">out1</span><span class="op">)</span></code></pre></div>
<pre><code>$Count
[1] 12

$Mean
[1] 81.0625

$Std.Error
[1] 1.547002

$Std.Deviation
[1] 5.358973

$Min
[1] 73.25

$Max
[1] 93.5

$`Skewness(statistic)`
     skew         z 
0.9250344 1.6745760 

$`Skewness(p.value)`
[1] 0.09401746

$`Kurtosis(statistic)`
    kurt        z 
3.522807 1.282305 

$`Kurtosis(p.value)`
[1] 0.1997357</code></pre>
<div class="sourceCode" id="cb209"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Descriptive statistics for variable y2</span>
<span class="fu"><a href="../reference/describe.augmentedRCBD.html">describe.augmentedRCBD</a></span><span class="op">(</span><span class="va">out2</span><span class="op">)</span></code></pre></div>
<pre><code>$Count
[1] 12

$Mean
[1] 298.4792

$Std.Error
[1] 18.92257

$Std.Deviation
[1] 65.5497

$Min
[1] 213.6667

$Max
[1] 437.6667

$`Skewness(statistic)`
     skew         z 
0.7449405 1.3680211 

$`Skewness(p.value)`
[1] 0.1713055

$`Kurtosis(statistic)`
    kurt        z 
2.787997 0.536812 

$`Kurtosis(p.value)`
[1] 0.5913975</code></pre>
</div>
<div id="freqdist-augmentedrcbd" class="section level2">
<h2 class="hasAnchor">
<a href="#freqdist-augmentedrcbd" class="anchor"></a>7.4 <code>freqdist.augmentedRCBD()</code>
</h2>
<p>The frequency distribution of the adjusted means from the results in an object of class <code>augmentedRCBD</code> can be plotted as follows.</p>
<div class="sourceCode" id="cb211"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Frequency distribution for variable y1</span>
<span class="va">freq1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out1</span>, xlab <span class="op">=</span> <span class="st">"Trait 1"</span><span class="op">)</span></code></pre></div>
<pre><code>Warning: Removed 2 rows containing missing values (geom_bar).</code></pre>
<div class="sourceCode" id="cb213"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">freq1</span><span class="op">)</span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-70-1.png" width="700"></p>
<div class="sourceCode" id="cb214"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Frequency distribution for variable y2</span>
<span class="va">freq2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out2</span>, xlab <span class="op">=</span> <span class="st">"Trait 2"</span><span class="op">)</span></code></pre></div>
<pre><code>Warning: Removed 2 rows containing missing values (geom_bar).</code></pre>
<div class="sourceCode" id="cb216"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">freq2</span><span class="op">)</span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-70-2.png" width="700"></p>
<p>The colours for the check values may be specified using the argument <code>check.col</code>.</p>
<div class="sourceCode" id="cb217"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">colset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red3"</span>, <span class="st">"green4"</span>, <span class="st">"purple3"</span>, <span class="st">"darkorange3"</span><span class="op">)</span>

<span class="co"># Frequency distribution for variable y1</span>
<span class="va">freq1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out1</span>, xlab <span class="op">=</span> <span class="st">"Trait 1"</span>, check.col <span class="op">=</span> <span class="va">colset</span><span class="op">)</span></code></pre></div>
<pre><code>Warning: Removed 2 rows containing missing values (geom_bar).</code></pre>
<div class="sourceCode" id="cb219"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">freq1</span><span class="op">)</span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-71-1.png" width="700"></p>
<div class="sourceCode" id="cb220"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Frequency distribution for variable y2</span>
<span class="va">freq2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out2</span>, xlab <span class="op">=</span> <span class="st">"Trait 2"</span>, check.col <span class="op">=</span> <span class="va">colset</span><span class="op">)</span></code></pre></div>
<pre><code>Warning: Removed 2 rows containing missing values (geom_bar).</code></pre>
<div class="sourceCode" id="cb222"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">freq2</span><span class="op">)</span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-71-2.png" width="700"></p>
<p>The default the check highlighting can be avoided using the argument <code>highlight.check = FALSE</code>.</p>
<div class="sourceCode" id="cb223"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Frequency distribution for variable y1</span>
<span class="va">freq1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out1</span>, xlab <span class="op">=</span> <span class="st">"Trait 1"</span>,
                                highlight.check <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>Warning: Removed 2 rows containing missing values (geom_bar).</code></pre>
<div class="sourceCode" id="cb225"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">freq1</span><span class="op">)</span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-72-1.png" width="700"></p>
<div class="sourceCode" id="cb226"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Frequency distribution for variable y2</span>
<span class="va">freq2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out2</span>, xlab <span class="op">=</span> <span class="st">"Trait 2"</span>,
                                highlight.check <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>Warning: Removed 2 rows containing missing values (geom_bar).</code></pre>
<div class="sourceCode" id="cb228"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">freq2</span><span class="op">)</span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-72-2.png" width="700"></p>
</div>
<div id="gva-augmentedrcbd" class="section level2">
<h2 class="hasAnchor">
<a href="#gva-augmentedrcbd" class="anchor"></a>7.5 <code>gva.augmentedRCBD()</code>
</h2>
<p>The genetic variability statistics such as mean, phenotypic, genotypic and environmental variation (<span class="citation">Federer and Searle (<a href="#ref-federerModelConsiderationsVariance1976" role="doc-biblioref">1976</a>)</span>), phenotypic, genotypic and environmental coefficient of variation (<span class="citation">Burton (<a href="#ref-burton_quantitative_1951" role="doc-biblioref">1951</a>)</span>, <span class="citation">Burton (<a href="#ref-burton_qualitative_1952" role="doc-biblioref">1952</a>)</span>), category of phenotypic and genotypic coefficient of variation according to <span class="citation">Sivasubramaniam and Madhavamenon (<a href="#ref-sivasubramaniam_genotypic_1973" role="doc-biblioref">1973</a>)</span>, broad-sense heritability (<em>H<sup>2</sup></em>) (<span class="citation">Lush (<a href="#ref-lush_intra-sire_1940" role="doc-biblioref">1940</a>)</span>), <em>H<sup>2</sup></em> category according to <span class="citation">Robinson (<a href="#ref-robinson_quantitative_1966" role="doc-biblioref">1966</a>)</span>, Genetic advance (GA), genetic advance as per cent of mean (GAM) and GAM category according to <span class="citation">Johnson et al. (<a href="#ref-johnson_estimates_1955" role="doc-biblioref">1955</a>)</span> are computed from an object of class <code>augmentedRCBD</code> as follows. Genetic variability analysis needs to be performed only if the sum of squares of “Treatment: Test” are significant.</p>
<div class="sourceCode" id="cb229"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Genetic variability statistics for variable y1</span>
<span class="fu"><a href="../reference/gva.augmentedRCBD.html">gva.augmentedRCBD</a></span><span class="op">(</span><span class="va">out1</span><span class="op">)</span></code></pre></div>
<pre><code>$Mean
[1] 81.0625

$PV
[1] 72.26786

$GV
[1] 45.29563

$EV
[1] 26.97222

$GCV
[1] 8.302487

$`GCV category`
[1] "Low"

$PCV
[1] 10.48703

$`PCV category`
[1] "Medium"

$ECV
[1] 6.406759

$hBS
[1] 62.67743

$`hBS category`
[1] "High"

$GA
[1] 10.99216

$GAM
[1] 13.5601

$`GAM category`
[1] "Medium"</code></pre>
<div class="sourceCode" id="cb231"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Genetic variability statistics for variable y2</span>
<span class="fu"><a href="../reference/gva.augmentedRCBD.html">gva.augmentedRCBD</a></span><span class="op">(</span><span class="va">out2</span><span class="op">)</span></code></pre></div>
<pre><code>Warning in gva.augmentedRCBD(out2): P-value for "Treatment: Test" is &lt; 0.05.</code></pre>
<pre><code>$Mean
[1] 298.4792

$PV
[1] 4980.411

$GV
[1] 4694.161

$EV
[1] 286.25

$GCV
[1] 22.95435

$`GCV category`
[1] "High"

$PCV
[1] 23.64387

$`PCV category`
[1] "High"

$ECV
[1] 5.668377

$hBS
[1] 94.25248

$`hBS category`
[1] "High"

$GA
[1] 137.2223

$GAM
[1] 45.97382

$`GAM category`
[1] "High"</code></pre>
<p>Negative estimates of variance components if computed are not abnormal. For information on how to deal with these, refer <span class="citation">Robinson et al. (<a href="#ref-robinson_genetic_1955" role="doc-biblioref">1955</a>)</span> and <span class="citation">Dudley and Moll (<a href="#ref-dudley_interpretation_1969" role="doc-biblioref">1969</a>)</span>.</p>
</div>
<div id="report-augmentedrcbd" class="section level2">
<h2 class="hasAnchor">
<a href="#report-augmentedrcbd" class="anchor"></a>7.5 <code>report.augmentedRCBD()</code>
</h2>
<p>The results generated by the analysis can be exported to a MS Word file as follows.</p>
<div class="sourceCode" id="cb234"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># MS word report for variable y1</span>
<span class="fu"><a href="../reference/report.augmentedRCBD.html">report.augmentedRCBD</a></span><span class="op">(</span>aug <span class="op">=</span> <span class="va">out1</span>,
                     target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"augmentedRCBD output - y1.docx"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># MS word report for variable y2</span>
<span class="fu"><a href="../reference/report.augmentedRCBD.html">report.augmentedRCBD</a></span><span class="op">(</span>aug <span class="op">=</span> <span class="va">out1</span>,
                     target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"augmentedRCBD output - y2.docx"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/augRCBDword.png" align="center" alt="The `R` download location."></p>
<p style="text-align: left;">
<strong>Fig. 6</strong>: MS Word report generated with <code>report.agumentedRCBD</code> function.
</p>
</div>
</div>
<div id="data-analysis-for-a-multiple-traits" class="section level1">
<h1 class="hasAnchor">
<a href="#data-analysis-for-a-multiple-traits" class="anchor"></a>8 Data Analysis for a Multiple Traits</h1>
<p>Analysis of data for a multiple traits simultaneously can be performed by using <code>augmentedRCBD.bulk</code> function. It generates an object of class <code>augmentedRCBD.bulk</code>. Such an object can then be taken as input by <code>print.augmentedRCBD.bulk</code> to print the results to console. The results can also be exported as a MS Word report using the <code>report.augmentedRCBD.bulk</code> function.</p>
<pre><code>[1] "package 'diagram' is required to generate this figure"</code></pre>
<p><strong>Fig. 7</strong>. Workflow for analysis of multiple traits with <code>augmentedRCBD</code>.</p>
<div id="augmentedrcbd-bulk" class="section level2">
<h2 class="hasAnchor">
<a href="#augmentedrcbd-bulk" class="anchor"></a>8.1 <code>augmentedRCBD.bulk()</code>
</h2>
<p>Consider the data frame <code>data</code> imported from <a href="#impexp">Table 1</a> according to the instructions in <a href="#impexp">section 4.8</a>.</p>
<div class="sourceCode" id="cb236"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></code></pre></div>
<pre><code>'data.frame':   20 obs. of  4 variables:
 $ blk: Factor w/ 3 levels "1","2","3": 1 1 1 1 1 1 1 2 2 2 ...
 $ trt: Factor w/ 12 levels "1","2","3","4",..: 1 2 3 4 7 11 12 1 2 3 ...
 $ y1 : num  92 79 87 81 96 89 82 79 81 81 ...
 $ y2 : num  258 224 238 278 347 300 289 260 220 237 ...</code></pre>
<div class="sourceCode" id="cb238"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Convert block and treatment to factors</span>
<span class="va">data</span><span class="op">$</span><span class="va">blk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span><span class="op">)</span>
<span class="va">data</span><span class="op">$</span><span class="va">trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">trt</span><span class="op">)</span></code></pre></div>
<p>Rather than performing the analysis individually for each variable/trait separately using <code>augmentedRCBD</code>, the analysis can be performed simultaneously for for both the traits using <code>augmentedRCBD.bulk</code> function. It is a wrapper around the <code>augmentedRCBD</code> core function and its associated helper functions.</p>
<p>However in this case treatment comparisons/grouping by least significant difference or Tukey’s honest significant difference method is not computed. Also the output object size is reduced using the <code>simplify = TRUE</code> argument in the <code>augmentedRCBD</code> function.</p>
<p>The logical arguments <code>describe</code>, <code>freqdist</code> and <code>gva</code> can be used to specify whether to generate the descriptive statistics, frequency distribution plots and genetic variability statistics respectively. If <code>gva = TRUE</code>, then plots to compare phenotypic and genotypic coefficient of variation, broad sense heritability and genetic advance over mean between traits are also generated.</p>
<div class="sourceCode" id="cb239"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bout</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.bulk.html">augmentedRCBD.bulk</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, block <span class="op">=</span> <span class="st">"blk"</span>,
                           treatment <span class="op">=</span> <span class="st">"trt"</span>, traits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"y1"</span>, <span class="st">"y2"</span><span class="op">)</span>,
                           checks <span class="op">=</span> <span class="cn">NULL</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, describe <span class="op">=</span> <span class="cn">TRUE</span>,
                           freqdist <span class="op">=</span> <span class="cn">TRUE</span>, gva <span class="op">=</span> <span class="cn">TRUE</span>,
                           check.col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"brown"</span>, <span class="st">"darkcyan"</span>,
                                         <span class="st">"forestgreen"</span>, <span class="st">"purple"</span><span class="op">)</span>,
                           console <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>
ANOVA for y1 computed (1/2)

ANOVA for y2 computed (2/2)

Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"
Number of traits           "2"         
Traits                     "y1, y2"    

ANOVA, Treatment Adjusted
=========================
                              Source Df         y1          y2
1        Block (ignoring Treatments)  2 180.04 *   3509.67 ** 
2     Treatment (eliminating Blocks) 11  25.92 ns  5360.49 ** 
3                   Treatment: Check  3  17.64 ns   716.75 ns 
4 Treatment: Test and Test vs. Check  8  29.02 ns  7101.89 ** 
5                          Residuals  6  26.97      286.25    

ANOVA, Block Adjusted
=====================
                          Source Df        y1           y2
1    Treatment (ignoring Blocks) 11 52.33 ns    5882.5 ** 
2               Treatment: Check  3 17.64 ns    716.75 ns 
3      Treatment: Test vs. Check  1 16.87 ns  27694.41 ** 
4                Treatment: Test  7 72.27 ns   4980.41 ** 
5 Block (eliminating Treatments)  2 34.75 ns    638.58 ns 
6                      Residuals  6 26.97       286.25    

Coefficient of Variation
========================
  Trait   CV
1    y1 6.37
2    y2 6.06


Overall Adjusted Mean
=====================
  Trait Overall.adjusted.mean
1    y1                 81.06
2    y2                298.48


Standard Errors
===================
                                Comparison   y1    y2
1 A Test Treatment and a Control Treatment  6.7 21.84
2                  Control Treatment Means 4.24 13.81
3   Two Test Treatments (Different Blocks) 8.21 26.75
4         Two Test Treatments (Same Block) 7.34 23.93


Critical Difference
===================
alpha = 0.05                                Comparison    y1    y2
1 A Test Treatment and a Control Treatment 16.41 53.45
2                  Control Treatment Means 10.38  33.8
3   Two Test Treatments (Different Blocks) 20.09 65.46
4         Two Test Treatments (Same Block) 17.97 58.55


Descriptive Statistics
===================
         Trait Count   Mean Std.Error Std.Deviation    Min    Max Skewness
skew...1    y1    12  81.06      1.55          5.36  73.25   93.5 0.93 ns 
skew...2    y2    12 298.48     18.92         65.55 213.67 437.67 0.74 ns 
         Kurtosis
skew...1 3.52 ns 
skew...2 2.79 ns 


Genetic Variability Analysis
===================
  Trait   Mean      PV      GV     EV   GCV GCV.category   PCV PCV.category
1    y1  81.06   72.27    45.3  26.97   8.3          Low 10.49       Medium
2    y2 298.48 4980.41 4694.16 286.25 22.95         High 23.64         High
   ECV   hBS hBS.category     GA   GAM GAM.category
1 6.41 62.68         High  10.99 13.56       Medium
2 5.67 94.25         High 137.22 45.97         High


Warning Messages
===================
y1
Removed 2 rows containing missing values (geom_bar).
y2
Removed 2 rows containing missing values (geom_bar).

Treatment Means
===============
   Treatment    y1     y2
1          1 84.67    256
2         10 77.25 437.67
3         11  86.5 299.42
4         12  79.5 288.42
5          2    79    228
6          3    82 247.67
7          4 83.33    264
8          5 78.25 293.92
9          6 78.25 382.67
10         7  93.5 346.42
11         8 73.25 213.67
12         9 77.25 323.92</code></pre>
</div>
<div id="print-augmentedrcbd-bulk" class="section level2">
<h2 class="hasAnchor">
<a href="#print-augmentedrcbd-bulk" class="anchor"></a>8.2 <code>print.augmentedRCBD.bulk()</code>
</h2>
<p>The results of analysis in an object of class <code>augmentedRCBD.bulk</code> can be printed to the console as follows.</p>
<div class="sourceCode" id="cb241"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Print results</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">bout</span><span class="op">)</span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"
Number of traits           "2"         
Traits                     "y1, y2"    

ANOVA, Treatment Adjusted
=========================
                              Source Df         y1          y2
1        Block (ignoring Treatments)  2 180.04 *   3509.67 ** 
2     Treatment (eliminating Blocks) 11  25.92 ns  5360.49 ** 
3                   Treatment: Check  3  17.64 ns   716.75 ns 
4 Treatment: Test and Test vs. Check  8  29.02 ns  7101.89 ** 
5                          Residuals  6  26.97      286.25    

ANOVA, Block Adjusted
=====================
                          Source Df        y1           y2
1    Treatment (ignoring Blocks) 11 52.33 ns    5882.5 ** 
2               Treatment: Check  3 17.64 ns    716.75 ns 
3      Treatment: Test vs. Check  1 16.87 ns  27694.41 ** 
4                Treatment: Test  7 72.27 ns   4980.41 ** 
5 Block (eliminating Treatments)  2 34.75 ns    638.58 ns 
6                      Residuals  6 26.97       286.25    

Coefficient of Variation
========================
  Trait   CV
1    y1 6.37
2    y2 6.06


Overall Adjusted Mean
=====================
  Trait Overall.adjusted.mean
1    y1                 81.06
2    y2                298.48


Standard Errors
===================
                                Comparison   y1    y2
1 A Test Treatment and a Control Treatment  6.7 21.84
2                  Control Treatment Means 4.24 13.81
3   Two Test Treatments (Different Blocks) 8.21 26.75
4         Two Test Treatments (Same Block) 7.34 23.93


Critical Difference
===================
alpha = 0.05                                Comparison    y1    y2
1 A Test Treatment and a Control Treatment 16.41 53.45
2                  Control Treatment Means 10.38  33.8
3   Two Test Treatments (Different Blocks) 20.09 65.46
4         Two Test Treatments (Same Block) 17.97 58.55


Descriptive Statistics
===================
         Trait Count   Mean Std.Error Std.Deviation    Min    Max Skewness
skew...1    y1    12  81.06      1.55          5.36  73.25   93.5 0.93 ns 
skew...2    y2    12 298.48     18.92         65.55 213.67 437.67 0.74 ns 
         Kurtosis
skew...1 3.52 ns 
skew...2 2.79 ns 


Genetic Variability Analysis
===================
  Trait   Mean      PV      GV     EV   GCV GCV.category   PCV PCV.category
1    y1  81.06   72.27    45.3  26.97   8.3          Low 10.49       Medium
2    y2 298.48 4980.41 4694.16 286.25 22.95         High 23.64         High
   ECV   hBS hBS.category     GA   GAM GAM.category
1 6.41 62.68         High  10.99 13.56       Medium
2 5.67 94.25         High 137.22 45.97         High


Warning Messages
===================
y1
Removed 2 rows containing missing values (geom_bar).
y2
Removed 2 rows containing missing values (geom_bar).

Treatment Means
===============
   Treatment    y1     y2
1          1 84.67    256
2         10 77.25 437.67
3         11  86.5 299.42
4         12  79.5 288.42
5          2    79    228
6          3    82 247.67
7          4 83.33    264
8          5 78.25 293.92
9          6 78.25 382.67
10         7  93.5 346.42
11         8 73.25 213.67
12         9 77.25 323.92</code></pre>
</div>
<div id="report-augmentedrcbd-bulk" class="section level2">
<h2 class="hasAnchor">
<a href="#report-augmentedrcbd-bulk" class="anchor"></a>8.3 <code>report.augmentedRCBD.bulk()</code>
</h2>
<p>The results generated by the analysis can be exported to a MS Word file as follows.</p>
<div class="sourceCode" id="cb243"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># MS word report</span>
<span class="fu"><a href="../reference/report.augmentedRCBD.bulk.html">report.augmentedRCBD.bulk</a></span><span class="op">(</span>aug.bulk <span class="op">=</span> <span class="va">bout</span>,
                          target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>,
                          <span class="st">"augmentedRCBD bulk output.docx"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/augRCBDbulkword.png" align="center" alt="The `R` download location."></p>
<p style="text-align: left;">
<strong>Fig. 8</strong>: MS Word report generated with <code>report.agumentedRCBD.bulk</code> function.
</p>
</div>
</div>
<div id="citing-augmentedrcbd" class="section level1">
<h1 class="hasAnchor">
<a href="#citing-augmentedrcbd" class="anchor"></a>9 Citing <code>augmentedRCBD</code>
</h1>
<pre><code>
To cite the R package 'augmentedRCBD' in publications use:

  Aravind, J., Mukesh Sankar, S., Wankhede, D. P., and Kaur, V. (2021).
  augmentedRCBD: Analysis of Augmented Randomised Complete Block
  Designs. R package version 0.1.4.9000,
  https://aravind-j.github.io/augmentedRCBD/https://cran.r-project.org/package=augmentedRCBD.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {augmentedRCBD: Analysis of Augmented Randomised Complete Block Designs},
    author = {J. Aravind and S. {Mukesh Sankar} and Dhammaprakash Pandhari Wankhede and Vikender Kaur},
    year = {2021},
    note = {R package version 0.1.4.9000},
    note = {https://aravind-j.github.io/augmentedRCBD/},
    note = {https://cran.r-project.org/package=augmentedRCBD},
  }

This free and open-source software implements academic research by the
authors and co-workers. If you use it, please support the project by
citing the package.</code></pre>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>10 Session Info</h1>
<div class="sourceCode" id="cb245"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] augmentedRCBD_0.1.4.9000

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6         mvtnorm_1.1-1      lattice_0.20-41    multcompView_0.1-8
 [5] zoo_1.8-9          rprojroot_2.0.2    digest_0.6.27      utf8_1.2.1        
 [9] plyr_1.8.6         R6_2.5.0           evaluate_0.14      httr_1.4.2        
[13] ggplot2_3.3.3      highr_0.9          pillar_1.6.1       Rdpack_2.1.1      
[17] gdtools_0.2.3      rlang_0.4.11       multcomp_1.4-17    uuid_0.1-4        
[21] data.table_1.14.0  jquerylib_0.1.4    Matrix_1.3-2       flextable_0.6.6   
[25] rmarkdown_2.8      mathjaxr_1.4-0     pkgdown_1.6.1      labeling_0.4.2    
[29] moments_0.14       splines_4.0.5      textshaping_0.3.1  desc_1.3.0        
[33] stringr_1.4.0      munsell_0.5.0      compiler_4.0.5     xfun_0.23         
[37] pkgconfig_2.0.3    systemfonts_1.0.2  base64enc_0.1-3    htmltools_0.5.1.1 
[41] tidyselect_1.1.1   tibble_3.1.2       codetools_0.2-18   fansi_0.4.2       
[45] crayon_1.4.1       dplyr_1.0.6        MASS_7.3-53.1      rbibutils_2.1.1   
[49] grid_4.0.5         jsonlite_1.7.2     xtable_1.8-4       gtable_0.3.0      
[53] lifecycle_1.0.0    magrittr_2.0.1     scales_1.1.1       zip_2.1.1         
[57] estimability_1.3   stringi_1.6.2      cachem_1.0.5       farver_2.1.0      
[61] reshape2_1.4.4     fs_1.5.0           xml2_1.3.2         bslib_0.2.5.1     
[65] ellipsis_0.3.2     ragg_1.1.1         generics_0.1.0     vctrs_0.3.8       
[69] sandwich_3.0-1     TH.data_1.0-10     tools_4.0.5        glue_1.4.2        
[73] officer_0.3.18     purrr_0.3.4        emmeans_1.6.0      fastmap_1.1.0     
[77] survival_3.2-10    yaml_2.2.1         colorspace_2.0-1   memoise_2.0.0     
[81] knitr_1.33         sass_0.4.0        </code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-anscombe_distribution_1983">
<p>Anscombe, F. J., and Glynn, W. J. (1983). Distribution of the kurtosis statistic <em>b</em><span class="math inline">\(_{{2}}\)</span> for normal samples. <em>Biometrika</em> 70, 227–234. doi:<a href="https://doi.org/10.1093/biomet/70.1.227">10.1093/biomet/70.1.227</a>.</p>
</div>
<div id="ref-burton_quantitative_1951">
<p>Burton, G. W. (1951). Quantitative inheritance in pearl millet (<em>Pennisetum glaucum</em>). <em>Agronomy Journal</em> 43, 409–417. doi:<a href="https://doi.org/10.2134/agronj1951.00021962004300090001x">10.2134/agronj1951.00021962004300090001x</a>.</p>
</div>
<div id="ref-burton_qualitative_1952">
<p>Burton, G. W. (1952). Qualitative inheritance in grasses. Vol. 1. in <em>Proceedings of the 6th International Grassland Congress, Pennsylvania State College</em>, 17–23.</p>
</div>
<div id="ref-dagostino_transformation_1970">
<p>D’Agostino, R. B. (1970). Transformation to normality of the null distribution of <em>g</em><span class="math inline">\(_{{1}}\)</span>. <em>Biometrika</em> 57, 679–681. doi:<a href="https://doi.org/10.1093/biomet/57.3.679">10.1093/biomet/57.3.679</a>.</p>
</div>
<div id="ref-dudley_interpretation_1969">
<p>Dudley, J. W., and Moll, R. H. (1969). Interpretation and use of estimates of heritability and genetic variances in plant breeding. <em>Crop Science</em> 9, 257–262. doi:<a href="https://doi.org/10.2135/cropsci1969.0011183X000900030001x">10.2135/cropsci1969.0011183X000900030001x</a>.</p>
</div>
<div id="ref-federer_augmented_1956">
<p>Federer, W. T. (1956a). Augmented (or Hoonuiaku) designs. <em>The Hawaiian Planters’ Record</em> LV(2), 191–208.</p>
</div>
<div id="ref-federer_augmented_1956-1">
<p>Federer, W. T. (1956b). Augmented (or Hoonuiaku) Designs. Cornell University, New York Available at: <a href="https://ecommons.cornell.edu/handle/1813/32841">https://ecommons.cornell.edu/handle/1813/32841</a>.</p>
</div>
<div id="ref-federer_augmented_1961">
<p>Federer, W. T. (1961). Augmented designs with one-way elimination of heterogeneity. <em>Biometrics</em> 17, 447–473. doi:<a href="https://doi.org/10.2307/2527837">10.2307/2527837</a>.</p>
</div>
<div id="ref-federerModelConsiderationsVariance1976">
<p>Federer, W. T., and Searle, S. R. (1976). Model Considerations and Variance Component Estimation in Augmented Completely Randomized and Randomized Complete Blocks Designs-Preliminary Version. Cornell University, New York Available at: <a href="https://hdl.handle.net/1813/32691">https://hdl.handle.net/1813/32691</a>.</p>
</div>
<div id="ref-johnson_estimates_1955">
<p>Johnson, H. W., Robinson, H. F., and Comstock, R. E. (1955). Estimates of genetic and environmental variability in soybeans. <em>Agronomy journal</em> 47, 314–318. doi:<a href="https://doi.org/10.2134/agronj1955.00021962004700070009x">10.2134/agronj1955.00021962004700070009x</a>.</p>
</div>
<div id="ref-lush_intra-sire_1940">
<p>Lush, J. L. (1940). Intra-sire correlations or regressions of offspring on dam as a method of estimating heritability of characteristics. <em>Proceedings of the American Society of Animal Nutrition</em> 1940, 293–301. doi:<a href="https://doi.org/10.2527/jas1940.19401293x">10.2527/jas1940.19401293x</a>.</p>
</div>
<div id="ref-robinson_quantitative_1966">
<p>Robinson, H. F. (1966). Quantitative genetics in relation to breeding on centennial of Mendelism. <em>Indian Journal of Genetics and Plant Breeding</em>, 171.</p>
</div>
<div id="ref-robinson_genetic_1955">
<p>Robinson, H. F., Comstock, R. E., and Harvey, P. H. (1955). Genetic variances in open pollinated varieties of corn. <em>Genetics</em> 40, 45–60.</p>
</div>
<div id="ref-sivasubramaniam_genotypic_1973">
<p>Sivasubramaniam, S., and Madhavamenon, P. (1973). Genotypic and phenotypic variability in rice. <em>The Madras Agricultural Journal</em> 60, 1093–1096.</p>
</div>
<div id="ref-tippmann_programming_2015">
<p>Tippmann, S. (2015). Programming tools: Adventures with R. <em>Nature News</em> 517, 109. doi:<a href="https://doi.org/10.1038/517109a">10.1038/517109a</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by J. Aravind, S. Mukesh Sankar, Dhammaprakash Pandhari Wankhede, Vikender Kaur.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'bdafb7ea4490644e670f470ee3ef537f',
    indexName: 'augmentedrcbd',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
