<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data Analysis with `augmentedRCBD` • augmentedRCBD</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Data Analysis with `augmentedRCBD`">
<meta property="og:description" content="augmentedRCBD">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-116716530-2"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-116716530-2');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">augmentedRCBD</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.1.5.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
    Home
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Reference
  </a>
</li>
<li>
  <a href="../articles/Data_Analysis_with_augmentedRCBD.html">
    <span class="far fa-file"></span>
     
    Tutorial
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="far fa-newspaper"></span>
     
    News
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/aravind-j/augmentedRCBD" class="external-link">
    <span class="fab fa-github-alt"></span>
     
    Github
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data Analysis with
<code>augmentedRCBD</code>
</h1>
                        <h4 data-toc-skip class="author">Aravind,
J.<sup>1</sup>, Mukesh Sankar, S.<sup>2</sup>, Wankhede, D.
P.<sup>3</sup>, and Kaur, V.<sup>4</sup>
</h4>
            
            <h4 data-toc-skip class="date">2023-05-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/aravind-j/augmentedRCBD/blob/HEAD/vignettes/Data_Analysis_with_augmentedRCBD.Rmd" class="external-link"><code>vignettes/Data_Analysis_with_augmentedRCBD.Rmd</code></a></small>
      <div class="hidden name"><code>Data_Analysis_with_augmentedRCBD.Rmd</code></div>

    </div>

    
    
<p>
</p>
<ol style="list-style-type: decimal">
<li>Division of Germplasm Conservation, ICAR-National Bureau of Plant
Genetic Resources, New Delhi.
</li>
</ol>
<p>
</p>
<ol start="2" style="list-style-type: decimal">
<li>Division of Genetics, ICAR-Indian Agricultural Research Institute,
New Delhi.
</li>
</ol>
<p>
</p>
<ol start="3" style="list-style-type: decimal">
<li>Division of Genomic Resources, ICAR-National Bureau of Plant Genetic
Resources, New Delhi.
</li>
</ol>
<p>
</p>
<ol start="4" style="list-style-type: decimal">
<li>Division of Germplasm Evaluation, ICAR-National Bureau of Plant
Genetic Resources, New Delhi.
</li>
</ol>
<p><img src="https://raw.githubusercontent.com/aravind-j/augmentedRCBD/master/inst/extdata/augmentedRCBD.png" align="right" alt="logo" width="173" height="200" style="padding: 10px; border: none; float: right;"></p>
<div class="section level2">
<h2 id="overview">1 Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The software <code>augmentedRCBD</code> is built on the <a href="https://en.wikipedia.org/wiki/R_(programming_language)" class="external-link"><code>R</code>
statistical programming language</a> as an add-on (or ‘package’ in the
<code>R</code> <em>lingua franca</em>). It performs the analysis of data
generated from experiments in augmented randomised complete block design
according to Federer, W.T. <span class="citation">(<a href="#ref-federer_augmented_1956" role="doc-biblioref">1956a</a>, <a href="#ref-federer_augmented_1956-1" role="doc-biblioref">1956b</a>, <a href="#ref-federer_augmented_1961" role="doc-biblioref">1961</a>; <a href="#ref-federerModelConsiderationsVariance1976" role="doc-biblioref">1976</a>)</span>. It also computes analysis of
variance, adjusted means, descriptive statistics, genetic variability
statistics etc. and includes options for data visualization and report
generation.</p>
<p>This tutorial aims to educate the users in utilising this package for
performing such analysis. Utilising <code>augmentedRCBD</code> for data
analysis requires a basic knowledge of <code>R</code> programming
language. However, as many of the intended end-users may not be familiar
with <code>R</code>, <a href="#rsoft">sections 2 to 4</a> give a
‘gentle’ introduction to <code>R</code>, especially those aspects which
are necessary to get <code>augmentedRCBD</code> up and running for
performing data analysis in a Windows environment. Users already
familiar with <code>R</code> can feel free to skip to <a href="#install">section 5</a>.</p>
<p><img src="https://www.r-project.org/logo/Rlogo.png" align="right" alt="logo" width="173" height="134" style="padding: 10px; border: none; float: right;"></p>
</div>
<div class="section level2">
<h2 id="rsoft">2 <code>R</code> software<a class="anchor" aria-label="anchor" href="#rsoft"></a>
</h2>
<p>It is a free software environment for statistical computing and
graphics. It is free and open source, platform independent (works on
Linux, Windows or MacOS), very flexible, comprehensive with robust
interfaces for all the popular programming languages as well as
databases. It is strengthened by its diverse library of add-on packages
extending its ability as well as the incredible community support. It is
one of the most popular tools being used in academia today <span class="citation">(<a href="#ref-tippmann_programming_2015" role="doc-biblioref">Tippmann, 2015</a>)</span>.</p>
</div>
<div class="section level2">
<h2 id="getting-started">3 Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>This section details the steps required to set up the <code>R</code>
programming environment under a third-party interface called
<code>RStudio</code> in Windows.</p>
<div class="section level3">
<h3 id="installing-r">3.1 Installing <code>R</code><a class="anchor" aria-label="anchor" href="#installing-r"></a>
</h3>
<p>Download and install <code>R</code> for Windows from <a href="http://cran.r-project.org/bin/windows/base/" class="external-link uri">http://cran.r-project.org/bin/windows/base/</a>.</p>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/rbase.png" align="center" alt="The `R` download location."></p>
<p style="text-align: left;">
<strong>Fig. 1</strong>: The <code>R</code> download location.
</p>
</div>
<div class="section level3">
<h3 id="installing-rstudio">3.2 Installing <code>RStudio</code><a class="anchor" aria-label="anchor" href="#installing-rstudio"></a>
</h3>
<p>The basic <a href="https://en.wikipedia.org/wiki/Command_line_interface" class="external-link">command line
interface</a> in native <code>R</code> is rather limiting. There are
several interfaces which enhance it’s functionality and ease of use, <a href="https://www.rstudio.com/" class="external-link"><code>RStudio</code></a> being one of
the most popular among <code>R</code> programmers.</p>
<p>Download and install <code>RStudio</code> for Windows from <a href="https://www.rstudio.com/products/rstudio/download/#download" class="external-link uri">https://www.rstudio.com/products/rstudio/download/#download</a></p>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/rstudio.png" align="center" alt="The `RStudio` download location."></p>
<p style="text-align: left;">
<strong>Fig. 2</strong>: The <code>RStudio</code> download location.
</p>
</div>
<div class="section level3">
<h3 id="the-rstudio-interface">3.3 The <code>RStudio</code> Interface<a class="anchor" aria-label="anchor" href="#the-rstudio-interface"></a>
</h3>
<p>On opening <code>RStudio</code>, the default interface with four
panes/windows is visible as follows. Few panes have different tabs.</p>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/rstudio%20panes.png" align="center" alt="The default `RStudio` interface with the four panes."></p>
<p style="text-align: left;">
<strong>Fig. 3</strong>: The default <code>RStudio</code> interface with
the four panes.
</p>
<div class="section level4">
<h4 id="console">3.3.1 Console<a class="anchor" aria-label="anchor" href="#console"></a>
</h4>
<p>This is where the action happens. Here any authentic <code>R</code>
code typed after the ‘<code>&gt;</code>’ prompt will be executed after
pressing ‘Enter’ to generate the output.</p>
<p>For example, type <code>1+1</code> in the console and press
‘Enter’.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fl">1</span><span class="op">+</span><span class="fl">1</span></span></code></pre></div>
<pre><code>[1] 2</code></pre>
</div>
<div class="section level4">
<h4 id="source">3.3.2 Source<a class="anchor" aria-label="anchor" href="#source"></a>
</h4>
<p>This is where <code>R</code> Scripts (collection of code) can be
created and edited. <code>R</code> scripts are text files with a
<code>.R</code> extension. <code>R</code> Code for analysis can be typed
and saved in such <code>R</code> scripts. New scripts can be opened by
clicking ‘File|New File’ and selecting ‘R Script’. Code can be selected
from <code>R</code> Scripts and sent to console for evaluation by
clicking ‘Run’ on the ‘Source’ pane or by pressing ‘Ctrl + Enter’.</p>
</div>
<div class="section level4">
<h4 id="environmenthistoryconnections">3.3.3 Environment|History|Connections<a class="anchor" aria-label="anchor" href="#environmenthistoryconnections"></a>
</h4>
<p>The ‘Environment’ tab shows the list of all the ‘objects’ (see <a href="#ObjFun">section 4.3</a>) defined in the current <code>R</code>
session. It has also some buttons up top to open, save and clear the
environment as well as few options for import of data under
<code>Import Dataset</code>.</p>
<p>The ‘History’ tab shows a history of all the code that was previously
evaluated. This is useful, if you want to go back to some code.</p>
<p>The ‘Connections’ tab helps to establish and manage connections with
different databases and data sources.</p>
</div>
<div class="section level4">
<h4 id="filesplotspackageshelpviewer">3.3.4 Files|Plots|Packages|Help|Viewer<a class="anchor" aria-label="anchor" href="#filesplotspackageshelpviewer"></a>
</h4>
<p>The ‘Files’ tab shows a sleek file browser to access the file
directory in the computer with options to manage the working directory
(see <a href="#wdir">section 4.1</a>) under the More button.</p>
<p>The ‘Plots’ tab shows all the plots generated in <code>R</code> with
buttons to delete unnecessary ones and export useful ones as a pdf file
or as an image file.</p>
<p>The ‘Packages’ tab shows a list of all the <code>R</code> add-on
packages installed. The check box on the left shows whether they are
loaded or not. There are also buttons to install and update
<code>R</code> packages.</p>
<p>The ‘Viewer’ tab shows any web content output generated by an
<code>R</code> code.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="some-basics">4 Some Basics<a class="anchor" aria-label="anchor" href="#some-basics"></a>
</h2>
<p>This section describes some basics to enable the users to have a
working knowledge in <code>R</code> in order to use
<code>augmentedRCBD</code>.</p>
<div class="section level3">
<h3 id="wdir">4.1 Working Directory<a class="anchor" aria-label="anchor" href="#wdir"></a>
</h3>
<p>It is a file path to a folder on the computer which is recognised by
<code>R</code> as the default location to read files from or write files
to. The code <code><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd()</a></code> shows the current working directory,
while <code><a href="https://rdrr.io/r/base/getwd.html" class="external-link">setwd()</a></code> can be used to change the existing working
directory.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Print current working directory</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "C:/Users/Computer/Documents"</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set new working directory</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">setwd</a></span><span class="op">(</span><span class="st">"C:/Data Analysis/"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "C:/Data Analysis/"</code></pre>
<p>One key detail is that file paths in <code>R</code> uses forward
slashes (<code>/</code>) as in MacOS or Linux, unlike backward slashes
(<code>\</code>) in Windows. This needs to be considered while copying
paths from default Windows file explorer.</p>
</div>
<div class="section level3">
<h3 id="expression-and-assignment">4.2 Expression and Assignment<a class="anchor" aria-label="anchor" href="#expression-and-assignment"></a>
</h3>
<p>Expressions are instructions in the form of code to be entered after
the <code>&gt;</code> prompt in the console. Expressions can be a
constant, an arithmetic or a condition. A more advanced and most useful
expression is a function call (see <a href="#ObjFun">section
4.3</a>).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Constant</span></span>
<span><span class="fl">123</span></span></code></pre></div>
<pre><code>[1] 123</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Arithmetic (add two numbers)</span></span>
<span><span class="fl">1</span> <span class="op">+</span> <span class="fl">2</span></span></code></pre></div>
<pre><code>[1] 3</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Condition</span></span>
<span><span class="fl">34</span> <span class="op">&gt;</span> <span class="fl">25</span></span></code></pre></div>
<pre><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fl">1</span> <span class="op">==</span> <span class="fl">2</span></span></code></pre></div>
<pre><code>[1] FALSE</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Function call (mean of a series of numbers)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">25</span>,<span class="fl">56</span>,<span class="fl">89</span>,<span class="fl">35</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 51.25</code></pre>
<p>Information from an expression can be stored as an ‘object’ (see <a href="#ObjFun">section 4.3</a>) by assigning a name using the operator
‘<code>&lt;-</code>’.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Assign the result of the expression 1 + 2 to an object 'a'</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fl">1</span> <span class="op">+</span> <span class="fl">2</span></span>
<span><span class="va">a</span></span></code></pre></div>
<pre><code>[1] 3</code></pre>
<p>It is recommended to add comments to explain the code by using the
‘<code>#</code>’ sign. Any code after the ‘<code>#</code>’ sign will be
ignored by <code>R</code>.</p>
</div>
<div class="section level3">
<h3 id="ObjFun">4.3 Objects and Functions<a class="anchor" aria-label="anchor" href="#ObjFun"></a>
</h3>
<p><code>R</code> is an object-oriented programming language (OOP). Any
kind or construct created in <code>R</code> is an ‘object’. Each object
has a ‘class’ (shown using the <code><a href="https://rdrr.io/r/base/class.html" class="external-link">class()</a></code> function) and
different ‘attributes’ which defines what operations can be done on that
object. There are different types of data structure objects in
<code>R</code> such as vectors, matrices, factors, data frames, and
lists. A ‘function’ is also an object, which defines a procedure or a
sequence of expressions.</p>
<div class="section level4">
<h4 id="vector">4.3.1 Vector<a class="anchor" aria-label="anchor" href="#vector"></a>
</h4>
<p>A vector is a collection of elements of a single type (or ‘mode’).
The common vector modes are ‘numeric’, ‘integer’, ‘character’ and
‘logical’. The <code><a href="https://rdrr.io/r/base/c.html" class="external-link">c()</a></code> function is used to create vectors. The
functions <code><a href="https://rdrr.io/r/base/class.html" class="external-link">class()</a></code>, <code><a href="https://rdrr.io/r/utils/str.html" class="external-link">str()</a></code> and
<code><a href="https://rdrr.io/r/base/length.html" class="external-link">length()</a></code> show the attributes of vectors.</p>
<p>Vector modes ‘numeric’ stores real numbers, while ‘integer’ stores
integers, which can be enforced by suffixing elements with
‘<code>L</code>’.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># A numeric vector</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3.3</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "numeric"</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></span></code></pre></div>
<pre><code> num [1:3] 1 2 3.3</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 3</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># An integer vector</span></span>
<span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1L</span>, <span class="fl">2L</span>, <span class="fl">3L</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "integer"</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<pre><code> int [1:3] 1 2 3</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 3</code></pre>
<p>The vector mode ‘character’ store text.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># A character vector</span></span>
<span><span class="va">c</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"one"</span>,<span class="st">"two"</span>,<span class="st">"three"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "character"</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></span></code></pre></div>
<pre><code> chr [1:3] "one" "two" "three"</code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 3</code></pre>
<p>The vector mode ‘logical’ stores ‘<code>TRUE</code>’ OR
‘<code>FALSE</code>’ logical data.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#logical vector</span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">TRUE</span>,<span class="cn">TRUE</span>,<span class="cn">TRUE</span>,<span class="cn">FALSE</span>,<span class="cn">TRUE</span>,<span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "logical"</code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span></code></pre></div>
<pre><code> logi [1:6] TRUE TRUE TRUE FALSE TRUE FALSE</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 6</code></pre>
</div>
<div class="section level4">
<h4 id="factor">4.3.2 Factor<a class="anchor" aria-label="anchor" href="#factor"></a>
</h4>
<p>A ‘factor’ in <code>R</code> stores data from categorical data in
variables as different levels.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">catg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"male"</span>,<span class="st">"female"</span>,<span class="st">"female"</span>,<span class="st">"male"</span>,<span class="st">"male"</span><span class="op">)</span></span>
<span><span class="va">catg</span></span></code></pre></div>
<pre><code>[1] "male"   "female" "female" "male"   "male"  </code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">is.factor</a></span><span class="op">(</span><span class="va">catg</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] FALSE</code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Apply the factor function</span></span>
<span><span class="va">factor_catg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">catg</span><span class="op">)</span></span>
<span></span>
<span><span class="va">factor_catg</span></span></code></pre></div>
<pre><code>[1] male   female female male   male  
Levels: female male</code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">is.factor</a></span><span class="op">(</span><span class="va">factor_catg</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">factor_catg</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "factor"</code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">factor_catg</span><span class="op">)</span></span></code></pre></div>
<pre><code> Factor w/ 2 levels "female","male": 2 1 1 2 2</code></pre>
<p>A character, numeric or integer vector can be transformed to a factor
by using the <code><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor()</a></code> function.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Conversion of numeric to factor</span></span>
<span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3.3</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "numeric"</code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></span></code></pre></div>
<pre><code> num [1:3] 1 2 3.3</code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fac_a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">fac_a</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "factor"</code></pre>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">fac_a</span><span class="op">)</span></span></code></pre></div>
<pre><code> Factor w/ 3 levels "1","2","3.3": 1 2 3</code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Conversion of integer to factor</span></span>
<span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1L</span>, <span class="fl">2L</span>, <span class="fl">3L</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "integer"</code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></span></code></pre></div>
<pre><code> int [1:3] 1 2 3</code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fac_b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">fac_b</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "factor"</code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">fac_b</span><span class="op">)</span></span></code></pre></div>
<pre><code> Factor w/ 3 levels "1","2","3": 1 2 3</code></pre>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Conversion of character to factor</span></span>
<span><span class="va">c</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"one"</span>,<span class="st">"two"</span>,<span class="st">"three"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "character"</code></pre>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></span></code></pre></div>
<pre><code> chr [1:3] "one" "two" "three"</code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fac_c</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">c</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">fac_c</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "factor"</code></pre>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">fac_c</span><span class="op">)</span></span></code></pre></div>
<pre><code> Factor w/ 3 levels "one","three",..: 1 3 2</code></pre>
</div>
<div class="section level4">
<h4 id="matrix">4.3.3 Matrix<a class="anchor" aria-label="anchor" href="#matrix"></a>
</h4>
<p>A ‘matrix’ in <code>R</code> is a vector with the attributes
‘<code>nrow</code>’ and ‘<code>ncol</code>’.</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Generate 5 * 4 numeric matrix</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span>, nrow <span class="op">=</span> <span class="fl">5</span>, ncol <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">m</span></span></code></pre></div>
<pre><code>     [,1] [,2] [,3] [,4]
[1,]    1    6   11   16
[2,]    2    7   12   17
[3,]    3    8   13   18
[4,]    4    9   14   19
[5,]    5   10   15   20</code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "matrix" "array" </code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/typeof.html" class="external-link">typeof</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "integer"</code></pre>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dimensions of m</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span> </span></code></pre></div>
<pre><code>[1] 5 4</code></pre>
</div>
<div class="section level4">
<h4 id="list">4.3.4 List<a class="anchor" aria-label="anchor" href="#list"></a>
</h4>
<p>A ‘list’ is a container containing different objects. The contents of
list need not be of the same type or mode. A list can encompass a
mixture of data types such as vectors, matrices, data frames, other
lists or any other data structure.</p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">a</span>, <span class="va">m</span>, <span class="va">d</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">b</span>, <span class="va">c</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">w</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "list"</code></pre>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">w</span><span class="op">)</span></span></code></pre></div>
<pre><code>List of 4
 $ : num [1:3] 1 2 3.3
 $ : int [1:5, 1:4] 1 2 3 4 5 6 7 8 9 10 ...
 $ : logi [1:6] TRUE TRUE TRUE FALSE TRUE FALSE
 $ :List of 2
  ..$ : int [1:3] 1 2 3
  ..$ : chr [1:3] "one" "two" "three"</code></pre>
</div>
<div class="section level4">
<h4 id="dataframe">4.3.5 Data Frame<a class="anchor" aria-label="anchor" href="#dataframe"></a>
</h4>
<p>A ‘data frame’ in <code>R</code> is a special kind of list with every
element having equal length. It is very important for handling tabular
data in <code>R</code>. It is a array like structure with rows and
columns. Each column needs to be of a single data type, however data
type can vary between columns.</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">L</span> <span class="op">&lt;-</span> <span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span></span>
<span><span class="va">z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"This"</span>, <span class="st">"is"</span>, <span class="st">"a"</span>, <span class="st">"data frame"</span><span class="op">)</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">L</span>, x <span class="op">=</span> <span class="fl">1</span>, <span class="va">y</span>, <span class="va">z</span><span class="op">)</span></span>
<span><span class="va">df</span></span></code></pre></div>
<pre><code>  L x y          z
1 A 1 1       This
2 B 1 2         is
3 C 1 3          a
4 D 1 4 data frame</code></pre>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre></div>
<pre><code>'data.frame':   4 obs. of  4 variables:
 $ L: chr  "A" "B" "C" "D"
 $ x: num  1 1 1 1
 $ y: int  1 2 3 4
 $ z: chr  "This" "is" "a" "data frame"</code></pre>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre></div>
<pre><code>$names
[1] "L" "x" "y" "z"

$class
[1] "data.frame"

$row.names
[1] 1 2 3 4</code></pre>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "1" "2" "3" "4"</code></pre>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "L" "x" "y" "z"</code></pre>
</div>
<div class="section level4">
<h4 id="functions">4.3.6 Functions<a class="anchor" aria-label="anchor" href="#functions"></a>
</h4>
<p>All of the work in <code>R</code> is done by functions. It is an
object defining a procedure which takes one or more objects as input (or
‘arguments’), performs some action on them and finally gives a new
object as output (or ‘return’). <code><a href="https://rdrr.io/r/base/class.html" class="external-link">class()</a></code>,
<code><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean()</a></code>, <code><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd()</a></code>, <code>+</code>, etc. are all
functions.</p>
<p>For example the function <code><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean()</a></code> takes a numeric vector
as argument and returns the mean as a numeric vector.</p>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3.3</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 2.1</code></pre>
<p>The user can also create custom functions. For example the function
<code>foo</code> adds two numbers and gives the result.</p>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">foo</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">n1</span>, <span class="va">n2</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">out</span> <span class="op">&lt;-</span> <span class="va">n1</span> <span class="op">+</span> <span class="va">n2</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">out</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">foo</span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] 5</code></pre>
</div>
</div>
<div class="section level3">
<h3 id="special-elements">4.4 Special Elements<a class="anchor" aria-label="anchor" href="#special-elements"></a>
</h3>
<p>In addition to numbers and text, there are some special elements
which can be included in different data objects.</p>
<p><code>NA</code> (not available) indicates missing data.</p>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2.5</span>, <span class="cn">NA</span>, <span class="fl">8.6</span><span class="op">)</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">FALSE</span>, <span class="cn">NA</span><span class="op">)</span></span>
<span><span class="va">z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"k"</span>, <span class="cn">NA</span>, <span class="st">"m"</span>, <span class="st">"n"</span>, <span class="st">"o"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] FALSE  TRUE FALSE</code></pre>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] FALSE  TRUE FALSE FALSE FALSE</code></pre>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">anyNA</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] TRUE</code></pre>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a</span></span></code></pre></div>
<pre><code>[1] 1.0 2.0 3.3</code></pre>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] FALSE FALSE FALSE</code></pre>
<p><code>Inf</code> indicates infinity.</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fl">1</span><span class="op">/</span><span class="fl">0</span></span></code></pre></div>
<pre><code>[1] Inf</code></pre>
<p><code>NaN</code> (Not a Number) indicates any undefined value.</p>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fl">0</span><span class="op">/</span><span class="fl">0</span></span></code></pre></div>
<pre><code>[1] NaN</code></pre>
</div>
<div class="section level3">
<h3 id="indexing">4.5 Indexing<a class="anchor" aria-label="anchor" href="#indexing"></a>
</h3>
<p>The <code>[</code> function is used to extract elements of an object
by indexing (numeric or logical). Named elements in lists and data
frames can be extracted by using the <code>$</code> operator.</p>
<p>Consider a vector <code>a</code>.</p>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3.3</span>, <span class="fl">2.8</span>, <span class="fl">6.7</span><span class="op">)</span></span>
<span><span class="co"># Numeric indexing</span></span>
<span><span class="co"># Extract first element</span></span>
<span><span class="va">a</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code>[1] 1</code></pre>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract elements 2:3</span></span>
<span><span class="va">a</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span></code></pre></div>
<pre><code>[1] 2.0 3.3</code></pre>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Logical indexing</span></span>
<span><span class="va">a</span><span class="op">[</span><span class="va">a</span> <span class="op">&gt;</span> <span class="fl">3</span><span class="op">]</span></span></code></pre></div>
<pre><code>[1] 3.3 6.7</code></pre>
<p>Consider a matrix <code>m</code>.</p>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">9</span>, nrow <span class="op">=</span> <span class="fl">3</span>, ncol <span class="op">=</span> <span class="fl">3</span>, byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'a'</span>, <span class="st">'b'</span>, <span class="st">'c'</span><span class="op">)</span></span>
<span><span class="va">m</span></span></code></pre></div>
<pre><code>     a b c
[1,] 1 2 3
[2,] 4 5 6
[3,] 7 8 9</code></pre>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract elements</span></span>
<span><span class="va">m</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="co"># 2nd column of matrix</span></span></code></pre></div>
<pre><code>[1] 2 5 8</code></pre>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span><span class="op">[</span><span class="fl">3</span>,<span class="op">]</span> <span class="co"># 3rd row of matrix</span></span></code></pre></div>
<pre><code>a b c 
7 8 9 </code></pre>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="co"># rows 2,3 of columns 1,2,3 </span></span></code></pre></div>
<pre><code>     a b c
[1,] 4 5 6
[2,] 7 8 9</code></pre>
<div class="sourceCode" id="cb133"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">]</span> <span class="co"># Element in 2nd column of 2nd row</span></span></code></pre></div>
<pre><code><span><span class="va">b</span> </span>
<span><span class="fl">5</span> </span></code></pre>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span><span class="op">[</span>, <span class="st">'b'</span><span class="op">]</span> <span class="co"># Column 'b'</span></span></code></pre></div>
<pre><code>[1] 2 5 8</code></pre>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">m</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'a'</span>, <span class="st">'c'</span><span class="op">)</span><span class="op">]</span> <span class="co"># Column 'a' and 'c'</span></span></code></pre></div>
<pre><code>     a c
[1,] 1 3
[2,] 4 6
[3,] 7 9</code></pre>
<p>Consider a list <code>w</code>.</p>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>vec <span class="op">=</span> <span class="va">a</span>, mat <span class="op">=</span> <span class="va">m</span>, data <span class="op">=</span> <span class="va">df</span>, alist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">b</span>, <span class="va">c</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Indexing by number</span></span>
<span><span class="va">w</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="co"># As list structure</span></span></code></pre></div>
<pre><code>$mat
     a b c
[1,] 1 2 3
[2,] 4 5 6
[3,] 7 8 9</code></pre>
<div class="sourceCode" id="cb141"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">w</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="co"># Without list structure</span></span></code></pre></div>
<pre><code>     a b c
[1,] 1 2 3
[2,] 4 5 6
[3,] 7 8 9</code></pre>
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Indexing by name</span></span>
<span><span class="va">w</span><span class="op">$</span><span class="va">vec</span></span></code></pre></div>
<pre><code>[1] 1.0 2.0 3.3 2.8 6.7</code></pre>
<div class="sourceCode" id="cb145"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">w</span><span class="op">$</span><span class="va">data</span></span></code></pre></div>
<pre><code>  L x y          z
1 A 1 1       This
2 B 1 2         is
3 C 1 3          a
4 D 1 4 data frame</code></pre>
<p>Consider a data frame <code>df</code>.</p>
<div class="sourceCode" id="cb147"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span></span></code></pre></div>
<pre><code>  L x y          z
1 A 1 1       This
2 B 1 2         is
3 C 1 3          a
4 D 1 4 data frame</code></pre>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Indexing by number</span></span>
<span><span class="va">df</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="co"># 2nd column of data frame</span></span></code></pre></div>
<pre><code>[1] 1 1 1 1</code></pre>
<div class="sourceCode" id="cb151"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="co"># 2nd column of data frame</span></span></code></pre></div>
<pre><code>  x
1 1
2 1
3 1
4 1</code></pre>
<div class="sourceCode" id="cb153"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span><span class="op">[</span><span class="fl">3</span>,<span class="op">]</span> <span class="co"># 3rd row of data frame</span></span></code></pre></div>
<pre><code>  L x y z
3 C 1 3 a</code></pre>
<div class="sourceCode" id="cb155"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="co"># rows 2,3 of columns 1,2,3 </span></span></code></pre></div>
<pre><code>  L x y
2 B 1 2
3 C 1 3</code></pre>
<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">]</span> <span class="co"># Element in 2nd column of 2nd row</span></span></code></pre></div>
<pre><code>[1] 1</code></pre>
<div class="sourceCode" id="cb159"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Indexing by name</span></span>
<span><span class="va">df</span><span class="op">$</span><span class="va">L</span></span></code></pre></div>
<pre><code>[1] "A" "B" "C" "D"</code></pre>
<div class="sourceCode" id="cb161"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span><span class="op">$</span><span class="va">z</span></span></code></pre></div>
<pre><code>[1] "This"       "is"         "a"          "data frame"</code></pre>
</div>
<div class="section level3">
<h3 id="help-documentation">4.6 Help Documentation<a class="anchor" aria-label="anchor" href="#help-documentation"></a>
</h3>
<p>The help documentation regarding any function can be viewed using the
<code>?</code> or <code><a href="https://rdrr.io/r/utils/help.html" class="external-link">help()</a></code> function. The help documentation
shows the default usage of the function including, the arguments that
are taken by the function and the type of output object returned
(‘Value’).</p>
<div class="sourceCode" id="cb163"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="va">ls</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">ls</span><span class="op">)</span></span>
<span></span>
<span><span class="op">?</span><span class="va">mean</span></span>
<span></span>
<span><span class="op">?</span><span class="va">setwd</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pack">4.7 Packages<a class="anchor" aria-label="anchor" href="#pack"></a>
</h3>
<p>Packages in <code>R</code> are collections of <code>R</code>
functions, data, and compiled code in a well-defined format. They are
add-ons which extend the functionality of <code>R</code> and at present,
there are <a href="https://cran.r-project.org/web/packages/available_packages_by_name.html" class="external-link">19501</a>
packages available for deployment and use at the official repository,
the Comprehensive R Archive Network (CRAN).</p>
<p>Valid packages from CRAN can be installed by using the
<code><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages()</a></code> command.</p>
<div class="sourceCode" id="cb164"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install the package 'readxl' for importing data from excel</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="va">readxl</span><span class="op">)</span></span></code></pre></div>
<p>Installed packages can be loaded using the function
<code><a href="https://rdrr.io/r/base/library.html" class="external-link">library()</a></code>.</p>
<div class="sourceCode" id="cb165"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install the package 'readxl' for importing data from excel</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readxl.tidyverse.org" class="external-link">readxl</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="impexp">4.8 Importing and Exporting Tabular Data<a class="anchor" aria-label="anchor" href="#impexp"></a>
</h3>
<p>Tabular data from a spreadsheet can be imported into <code>R</code>
in different ways. Consider some data such as in Table 1. Copy this data
in to a spreadsheet editor such as MS Excel and save it as
<code>augdata.csv</code>, a comma-separated-value file and
<code>augdata.xlsx</code>, an Excel file in the working directory
(<code><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd()</a></code>).</p>
<p style="text-align: left;">
<strong>Table 1</strong>: Example data from an experiment in augmented
RCBD design.
</p>
<table class="table">
<thead><tr class="header">
<th align="left">blk</th>
<th align="right">trt</th>
<th align="right">y1</th>
<th align="right">y2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">I</td>
<td align="right">1</td>
<td align="right">92</td>
<td align="right">258</td>
</tr>
<tr class="even">
<td align="left">I</td>
<td align="right">2</td>
<td align="right">79</td>
<td align="right">224</td>
</tr>
<tr class="odd">
<td align="left">I</td>
<td align="right">3</td>
<td align="right">87</td>
<td align="right">238</td>
</tr>
<tr class="even">
<td align="left">I</td>
<td align="right">4</td>
<td align="right">81</td>
<td align="right">278</td>
</tr>
<tr class="odd">
<td align="left">I</td>
<td align="right">7</td>
<td align="right">96</td>
<td align="right">347</td>
</tr>
<tr class="even">
<td align="left">I</td>
<td align="right">11</td>
<td align="right">89</td>
<td align="right">300</td>
</tr>
<tr class="odd">
<td align="left">I</td>
<td align="right">12</td>
<td align="right">82</td>
<td align="right">289</td>
</tr>
<tr class="even">
<td align="left">II</td>
<td align="right">1</td>
<td align="right">79</td>
<td align="right">260</td>
</tr>
<tr class="odd">
<td align="left">II</td>
<td align="right">2</td>
<td align="right">81</td>
<td align="right">220</td>
</tr>
<tr class="even">
<td align="left">II</td>
<td align="right">3</td>
<td align="right">81</td>
<td align="right">237</td>
</tr>
<tr class="odd">
<td align="left">II</td>
<td align="right">4</td>
<td align="right">91</td>
<td align="right">227</td>
</tr>
<tr class="even">
<td align="left">II</td>
<td align="right">5</td>
<td align="right">79</td>
<td align="right">281</td>
</tr>
<tr class="odd">
<td align="left">II</td>
<td align="right">9</td>
<td align="right">78</td>
<td align="right">311</td>
</tr>
<tr class="even">
<td align="left">III</td>
<td align="right">1</td>
<td align="right">83</td>
<td align="right">250</td>
</tr>
<tr class="odd">
<td align="left">III</td>
<td align="right">2</td>
<td align="right">77</td>
<td align="right">240</td>
</tr>
<tr class="even">
<td align="left">III</td>
<td align="right">3</td>
<td align="right">78</td>
<td align="right">268</td>
</tr>
<tr class="odd">
<td align="left">III</td>
<td align="right">4</td>
<td align="right">78</td>
<td align="right">287</td>
</tr>
<tr class="even">
<td align="left">III</td>
<td align="right">8</td>
<td align="right">70</td>
<td align="right">226</td>
</tr>
<tr class="odd">
<td align="left">III</td>
<td align="right">6</td>
<td align="right">75</td>
<td align="right">395</td>
</tr>
<tr class="even">
<td align="left">III</td>
<td align="right">10</td>
<td align="right">74</td>
<td align="right">450</td>
</tr>
</tbody>
</table>
<p>The <code>augdata.csv</code> file can be imported into <code>R</code>
using the <code><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv()</a></code> function or the <a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link"><code>read_csv()</code></a>
function in the <code>readr</code> package.</p>
<div class="sourceCode" id="cb166"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"augdata.csv"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span></code></pre></div>
<pre><code>'data.frame':   20 obs. of  4 variables:
 $ blk: Factor w/ 3 levels "I","II","III": 1 1 1 1 1 1 1 2 2 2 ...
 $ trt: num  1 2 3 4 7 11 12 1 2 3 ...
 $ y1 : num  92 79 87 81 96 89 82 79 81 81 ...
 $ y2 : num  258 224 238 278 347 300 289 260 220 237 ...</code></pre>
<p>The argument <code>stringsAsFactors = FALSE</code> reads the text
columns as of type <code>character</code> instead of the default
<code>factor</code>.</p>
<div class="sourceCode" id="cb168"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"augdata.csv"</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span></code></pre></div>
<pre><code>'data.frame':   20 obs. of  4 variables:
 $ blk: chr  "I" "I" "I" "I" ...
 $ trt: num  1 2 3 4 7 11 12 1 2 3 ...
 $ y1 : num  92 79 87 81 96 89 82 79 81 81 ...
 $ y2 : num  258 224 238 278 347 300 289 260 220 237 ...</code></pre>
<p>The <code>augdata.xlsx</code> file can be imported into
<code>R</code> using the <a href="https://readxl.tidyverse.org/reference/read_excel.html" class="external-link"><code>read_excel()</code></a>
function in the <code>readxl</code> package.</p>
<div class="sourceCode" id="cb170"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readxl.tidyverse.org" class="external-link">readxl</a></span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readxl.tidyverse.org/reference/read_excel.html" class="external-link">read_excel</a></span><span class="op">(</span>path <span class="op">=</span> <span class="st">"augdata.xlsx"</span><span class="op">)</span></span></code></pre></div>
<pre><code>'data.frame':   20 obs. of  4 variables:
 $ blk: chr  "I" "I" "I" "I" ...
 $ trt: num  1 2 3 4 7 11 12 1 2 3 ...
 $ y1 : num  92 79 87 81 96 89 82 79 81 81 ...
 $ y2 : num  258 224 238 278 347 300 289 260 220 237 ...</code></pre>
<p>The tabular data can be exported from <code>R</code> to a
<code>.csv</code> (comma-separated-value) file by the <a href="https://www.rdocumentation.org/packages/utils/versions/3.5.1/topics/write.table" class="external-link"><code>write.csv()</code></a>
function.</p>
<div class="sourceCode" id="cb172"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">data</span>, file <span class="op">=</span> <span class="st">"augdata.csv"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="additional-resources">4.9 Additional Resources<a class="anchor" aria-label="anchor" href="#additional-resources"></a>
</h3>
<p>To learn more about <code>R</code>, there are umpteen number of
online tutorials as well as free courses available. Queries about
various aspects can be put to the active and vibrant `R community
online.</p>
<ul>
<li>Online tutorials
<ul>
<li><a href="http://www.cran.r-project.org/other-docs.html" class="external-link uri">http://www.cran.r-project.org/other-docs.html</a></li>
<li><a href="https://bookdown.org/ndphillips/YaRrr/" class="external-link uri">https://bookdown.org/ndphillips/YaRrr/</a></li>
</ul>
</li>
<li>Free online courses
<ul>
<li><a href="http://tryr.codeschool.com/" class="external-link uri">http://tryr.codeschool.com/</a></li>
<li><a href="https://www.datacamp.com/courses/free-introduction-to-r" class="external-link uri">https://www.datacamp.com/courses/free-introduction-to-r</a></li>
</ul>
</li>
<li>
<code>R</code> community support
<ul>
<li><a href="http://stackoverflow.com/" class="external-link uri">http://stackoverflow.com/</a></li>
<li>
<code>R</code> help mailing lists : <a href="http://www.r-project.org/mail.html" class="external-link uri">http://www.r-project.org/mail.html</a>
</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="install">5 Installation of <code>augmentedRCBD</code><a class="anchor" aria-label="anchor" href="#install"></a>
</h2>
<p>The package <code>augmentedRCBD</code> can be installed using the
following functions.</p>
<div class="sourceCode" id="cb173"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from CRAN</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'augmentedRCBD'</span>, dependencies<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Install development version from Github</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">'devtools'</a></span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'devtools'</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"aravind-j/augmentedRCBD"</span><span class="op">)</span></span></code></pre></div>
<p>The stable release is hosted in <a href="https://CRAN.R-project.org/package=augmentedRCBD" class="external-link">CRAN</a> <a href="#pack">(see section 4.7)</a>, while the under-development version
is hosted as a <a href="https://github.com/aravind-j/augmentedRCBD" class="external-link">Github</a> repository.
To install from github, you need to use the <a href="https://devtools.r-lib.org/reference/reexports.html" class="external-link"><code>install_github()</code></a>
function from <a href="https://devtools.r-lib.org/" class="external-link">`devtools</a>
package.</p>
<p>Then the package can be loaded using the function</p>
<div class="sourceCode" id="cb174"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/aravind-j/augmentedRCBD" class="external-link">augmentedRCBD</a></span><span class="op">)</span></span></code></pre></div>
<pre><code>
--------------------------------------------------------------------------------
Welcome to augmentedRCBD version 0.1.5.9000


# To know how to use this package type:
  browseVignettes(package = 'augmentedRCBD')
  for the package vignette.

# To know whats new in this version type:
  news(package='augmentedRCBD')
  for the NEWS file.

# To cite the methods in the package type:
  citation(package='augmentedRCBD')

# To suppress this message use:
  suppressPackageStartupMessages(library(augmentedRCBD))
--------------------------------------------------------------------------------</code></pre>
<p>The current version of the package is 0.1.5. The previous versions
are as follows.</p>
<p><strong>Table 2.</strong> Version history of
<code>augmentedRCBD</code> <code>R</code> package.</p>
<table class="table">
<thead><tr class="header">
<th align="left">Version</th>
<th align="left">Date</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">0.1.0</td>
<td align="left">2018-07-10</td>
</tr>
<tr class="even">
<td align="left">0.1.1</td>
<td align="left">2019-07-21</td>
</tr>
<tr class="odd">
<td align="left">0.1.2</td>
<td align="left">2020-03-19</td>
</tr>
<tr class="even">
<td align="left">0.1.3</td>
<td align="left">2020-07-27</td>
</tr>
<tr class="odd">
<td align="left">0.1.4</td>
<td align="left">2021-02-17</td>
</tr>
</tbody>
</table>
<p>To know detailed history of changes use
<code>news(package='augmentedRCBD')</code>.</p>
</div>
<div class="section level2">
<h2 id="data-format">6 Data Format<a class="anchor" aria-label="anchor" href="#data-format"></a>
</h2>
<p>Certain details need to be considered for arranging experimental data
for analysis using the <code>augmentedRCBD</code> package.</p>
<p>The data should be in long/vertical form, where each row has the data
from one genotype per block. For example, consider the following data
(Table 3) recorded for a trait from an experiment laid out in an
augmented block design with 3 blocks and 12 genotypes(or treatment) with
6 to 7 genotypes/block. 8 genotypes (Test, G 5 to G 12) are not
replicated, while 4 genotypes (Check, G 1 to G 4) are replicated.</p>
<p style="text-align: left;">
<strong>Table 3</strong>: Data from an experiment in augmented RCBD
design.
</p>
<table class="table"><tbody>
<tr class="odd">
<td align="left"><strong>Block I</strong></td>
<td align="left">G12</td>
<td align="left"><strong>G4</strong></td>
<td align="left">G11</td>
<td align="left"><strong>G2</strong></td>
<td align="left"><strong>G1</strong></td>
<td align="left">G7</td>
<td align="left"><strong>G3</strong></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">82</td>
<td align="left">81</td>
<td align="left">89</td>
<td align="left">79</td>
<td align="left">92</td>
<td align="left">96</td>
<td align="left">87</td>
</tr>
<tr class="odd">
<td align="left"><strong>Block II</strong></td>
<td align="left">G5</td>
<td align="left">G9</td>
<td align="left">–</td>
<td align="left"><strong>G3</strong></td>
<td align="left"><strong>G1</strong></td>
<td align="left"><strong>G2</strong></td>
<td align="left"><strong>G4</strong></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">79</td>
<td align="left">78</td>
<td align="left">–</td>
<td align="left">81</td>
<td align="left">79</td>
<td align="left">81</td>
<td align="left">91</td>
</tr>
<tr class="odd">
<td align="left"><strong>Block III</strong></td>
<td align="left"><strong>G4</strong></td>
<td align="left"><strong>G2</strong></td>
<td align="left"><strong>G1</strong></td>
<td align="left">G6</td>
<td align="left">G10</td>
<td align="left"><strong>G3</strong></td>
<td align="left">G8</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">78</td>
<td align="left">77</td>
<td align="left">83</td>
<td align="left">75</td>
<td align="left">74</td>
<td align="left">78</td>
<td align="left">70</td>
</tr>
</tbody></table>
<p>This data needs to be arranged with columns showing block, genotype
(or treatment) and the data of the trait for each genotype per block
(Table 4).</p>
<p style="text-align: left;">
<strong>Table 4</strong>: Data from an experiment in augmented RCBD
design arranged in long-form.
</p>
<table class="table">
<thead><tr class="header">
<th align="left">Block</th>
<th align="left">Treatment</th>
<th align="right">Trait</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 1</td>
<td align="right">92</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 2</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 3</td>
<td align="right">87</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 4</td>
<td align="right">81</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 7</td>
<td align="right">96</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 11</td>
<td align="right">89</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 12</td>
<td align="right">82</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 1</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 2</td>
<td align="right">81</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 3</td>
<td align="right">81</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 4</td>
<td align="right">91</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 5</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 9</td>
<td align="right">78</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 1</td>
<td align="right">83</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 2</td>
<td align="right">77</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 3</td>
<td align="right">78</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 4</td>
<td align="right">78</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 8</td>
<td align="right">70</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 6</td>
<td align="right">75</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 10</td>
<td align="right">74</td>
</tr>
</tbody>
</table>
<p>The data for block and genotype (or treatment) can also be depicted
as numbers (Table 5).</p>
<p style="text-align: left;">
<strong>Table 5</strong>: Data from an experiment in augmented RCBD
design arranged in long-form (Block and Treatment as numbers).
</p>
<table class="table">
<thead><tr class="header">
<th align="right">Block</th>
<th align="right">Treatment</th>
<th align="right">Trait</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="right">1</td>
<td align="right">92</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">2</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">3</td>
<td align="right">87</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">4</td>
<td align="right">81</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">7</td>
<td align="right">96</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">11</td>
<td align="right">89</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">12</td>
<td align="right">82</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">1</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">2</td>
<td align="right">81</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">3</td>
<td align="right">81</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">4</td>
<td align="right">91</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">5</td>
<td align="right">79</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">9</td>
<td align="right">78</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">1</td>
<td align="right">83</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">2</td>
<td align="right">77</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">3</td>
<td align="right">78</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">4</td>
<td align="right">78</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">8</td>
<td align="right">70</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">6</td>
<td align="right">75</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">10</td>
<td align="right">74</td>
</tr>
</tbody>
</table>
<p>Multiple traits can be added as additional columns (Table 6).</p>
<p style="text-align: left;">
<strong>Table 6</strong>: Data from an experiment in augmented RCBD
design arranged in long-form (Multiple traits).
</p>
<table class="table">
<thead><tr class="header">
<th align="left">Block</th>
<th align="left">Treatment</th>
<th align="right">Trait1</th>
<th align="right">Trait2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 1</td>
<td align="right">92</td>
<td align="right">258</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 2</td>
<td align="right">79</td>
<td align="right">224</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 3</td>
<td align="right">87</td>
<td align="right">238</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 4</td>
<td align="right">81</td>
<td align="right">278</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 7</td>
<td align="right">96</td>
<td align="right">347</td>
</tr>
<tr class="even">
<td align="left">Block I</td>
<td align="left">G 11</td>
<td align="right">89</td>
<td align="right">300</td>
</tr>
<tr class="odd">
<td align="left">Block I</td>
<td align="left">G 12</td>
<td align="right">82</td>
<td align="right">289</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 1</td>
<td align="right">79</td>
<td align="right">260</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 2</td>
<td align="right">81</td>
<td align="right">220</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 3</td>
<td align="right">81</td>
<td align="right">237</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 4</td>
<td align="right">91</td>
<td align="right">227</td>
</tr>
<tr class="even">
<td align="left">Block II</td>
<td align="left">G 5</td>
<td align="right">79</td>
<td align="right">281</td>
</tr>
<tr class="odd">
<td align="left">Block II</td>
<td align="left">G 9</td>
<td align="right">78</td>
<td align="right">311</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 1</td>
<td align="right">83</td>
<td align="right">250</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 2</td>
<td align="right">77</td>
<td align="right">240</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 3</td>
<td align="right">78</td>
<td align="right">268</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 4</td>
<td align="right">78</td>
<td align="right">287</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 8</td>
<td align="right">70</td>
<td align="right">226</td>
</tr>
<tr class="odd">
<td align="left">Block III</td>
<td align="left">G 6</td>
<td align="right">75</td>
<td align="right">395</td>
</tr>
<tr class="even">
<td align="left">Block III</td>
<td align="left">G 10</td>
<td align="right">74</td>
<td align="right">450</td>
</tr>
</tbody>
</table>
<p>Data should preferably be balanced i.e. all the check genotypes
should be present in all the blocks. If not, a warning is issued. The
number of test genotypes can vary within a block. There should not be
any missing values. Rows of genotypes with missing values for one or
more traits should be removed.</p>
<p>Such a tabular data should be imported (<a href="#impexp">see section
7.8</a>) into <code>R</code> as a data frame object (<a href="#dataframe">see section 4.3.5</a>). The columns with the block and
treatment categorical data should of the type factor (<a href="#factor">see section 4.3.2</a>), while the column(s) with the
trait data should be of the type integer or numeric (<a href="#vector">see section 4.3.1</a>).</p>
</div>
<div class="section level2">
<h2 id="data-analysis-for-a-single-trait">7 Data Analysis for a Single Trait<a class="anchor" aria-label="anchor" href="#data-analysis-for-a-single-trait"></a>
</h2>
<p>Analysis of data for a single trait can be performed by using
<code>augmentedRCBD</code> function. It generates an object of class
<code>augmentedRCBD</code>. Such an object can then be taken as input by
the several functions to print the results to console
(<code>print.augmentedRCBD</code>), generate descriptive statistics from
adjusted means (<code>describe.augmentedRCBD</code>), plot frequency
distribution (<code>freqdist.augmentedRCBD</code>) and computed genetic
variability statistics (gva.augmentedRCBD). All these outputs can also
be exported as a MS Word report using the
<code>report.augmentedRCBD</code> function.</p>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-59-1.png" width="700"></p>
<p><strong>Fig. 4</strong>. Workflow for analysis of single traits with
<code>augmentedRCBD</code>.</p>
<div class="section level3">
<h3 id="augmentedrcbd">7.1 <code>augmentedRCBD()</code><a class="anchor" aria-label="anchor" href="#augmentedrcbd"></a>
</h3>
<p>Consider the data in <a href="#impexp">Table 1</a>. The data can be
imported into <code>R</code> as <a href="#vector">vectors</a> as
follows.</p>
<div class="sourceCode" id="cb176"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">11</span>, <span class="fl">12</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">9</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">8</span>, <span class="fl">6</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">y1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">92</span>, <span class="fl">79</span>, <span class="fl">87</span>, <span class="fl">81</span>, <span class="fl">96</span>, <span class="fl">89</span>, <span class="fl">82</span>, <span class="fl">79</span>, <span class="fl">81</span>, <span class="fl">81</span>, <span class="fl">91</span>, <span class="fl">79</span>, <span class="fl">78</span>, <span class="fl">83</span>, <span class="fl">77</span>, <span class="fl">78</span>, <span class="fl">78</span>,</span>
<span>        <span class="fl">70</span>, <span class="fl">75</span>, <span class="fl">74</span><span class="op">)</span></span>
<span><span class="va">y2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">258</span>, <span class="fl">224</span>, <span class="fl">238</span>, <span class="fl">278</span>, <span class="fl">347</span>, <span class="fl">300</span>, <span class="fl">289</span>, <span class="fl">260</span>, <span class="fl">220</span>, <span class="fl">237</span>, <span class="fl">227</span>, <span class="fl">281</span>, <span class="fl">311</span>, <span class="fl">250</span>,</span>
<span>        <span class="fl">240</span>, <span class="fl">268</span>, <span class="fl">287</span>, <span class="fl">226</span>, <span class="fl">395</span>, <span class="fl">450</span><span class="op">)</span></span></code></pre></div>
<p>The <code>blk</code> and <code>trt</code> vectors with the block and
treatment data need to be converted into factors as follows before
analysis.</p>
<div class="sourceCode" id="cb177"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert block and treatment to factors</span></span>
<span><span class="va">blk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">blk</span><span class="op">)</span></span>
<span><span class="va">trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">trt</span><span class="op">)</span></span></code></pre></div>
<p>With the data in appropriate format, the analysis can be performed as
follows for the trait <code>y1</code> as follows.</p>
<div class="sourceCode" id="cb178"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">blk</span>, <span class="va">trt</span>, <span class="va">y1</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,</span>
<span>                      alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"


ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value Pr(&gt;F)  
Block (ignoring Treatments)           2  360.1  180.04   6.675 0.0298 *
Treatment (eliminating Blocks)       11  285.1   25.92   0.961 0.5499  
  Treatment: Check                    3   52.9   17.64   0.654 0.6092  
  Treatment: Test and Test vs. Check  8  232.2   29.02   1.076 0.4779  
Residuals                             6  161.8   26.97                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value Pr(&gt;F)
Treatment (ignoring Blocks)    11  575.7   52.33   1.940  0.215
  Treatment: Check              3   52.9   17.64   0.654  0.609
  Treatment: Test               7  505.9   72.27   2.679  0.125
  Treatment: Test vs. Check     1   16.9   16.87   0.626  0.459
Block (eliminating Treatments)  2   69.5   34.75   1.288  0.342
Residuals                       6  161.8   26.97               

Coefficient of Variation
========================
6.372367

Overall Adjusted Mean
=====================
81.0625

Standard Errors
===============
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             4.240458 10.37603
Two Test Treatments (Same Block)                    7.344688 17.97180
Two Test Treatments (Different Blocks)              8.211611 20.09309
A Test Treatment and a Control Treatment            6.704752 16.40594

Treatment Means
===============
 Treatment Block Means   SE r   Min   Max Adjusted Means
         1       84.67 3.84 3 79.00 92.00          84.67
        10     3 74.00 &lt;NA&gt; 1 74.00 74.00          77.25
        11     1 89.00 &lt;NA&gt; 1 89.00 89.00          86.50
        12     1 82.00 &lt;NA&gt; 1 82.00 82.00          79.50
         2       79.00 1.15 3 77.00 81.00          79.00
         3       82.00 2.65 3 78.00 87.00          82.00
         4       83.33 3.93 3 78.00 91.00          83.33
         5     2 79.00 &lt;NA&gt; 1 79.00 79.00          78.25
         6     3 75.00 &lt;NA&gt; 1 75.00 75.00          78.25
         7     1 96.00 &lt;NA&gt; 1 96.00 96.00          93.50
         8     3 70.00 &lt;NA&gt; 1 70.00 70.00          73.25
         9     2 78.00 &lt;NA&gt; 1 78.00 78.00          77.25


Comparisons
===========

Method : lsd

                  contrast estimate   SE df t.ratio p.value sig
   treatment1 - treatment2     5.67 4.24  6   1.336   0.230    
   treatment1 - treatment3     2.67 4.24  6   0.629   0.553    
   treatment1 - treatment4     1.33 4.24  6   0.314   0.764    
   treatment1 - treatment5     6.42 6.36  6   1.009   0.352    
   treatment1 - treatment6     6.42 6.36  6   1.009   0.352    
   treatment1 - treatment7    -8.83 6.36  6  -1.389   0.214    
   treatment1 - treatment8    11.42 6.36  6   1.795   0.123    
   treatment1 - treatment9     7.42 6.36  6   1.166   0.288    
  treatment1 - treatment10     7.42 6.36  6   1.166   0.288    
  treatment1 - treatment11    -1.83 6.36  6  -0.288   0.783    
  treatment1 - treatment12     5.17 6.36  6   0.812   0.448    
   treatment2 - treatment3    -3.00 4.24  6  -0.707   0.506    
   treatment2 - treatment4    -4.33 4.24  6  -1.022   0.346    
   treatment2 - treatment5     0.75 6.36  6   0.118   0.910    
   treatment2 - treatment6     0.75 6.36  6   0.118   0.910    
   treatment2 - treatment7   -14.50 6.36  6  -2.280   0.063    
   treatment2 - treatment8     5.75 6.36  6   0.904   0.401    
   treatment2 - treatment9     1.75 6.36  6   0.275   0.792    
  treatment2 - treatment10     1.75 6.36  6   0.275   0.792    
  treatment2 - treatment11    -7.50 6.36  6  -1.179   0.283    
  treatment2 - treatment12    -0.50 6.36  6  -0.079   0.940    
   treatment3 - treatment4    -1.33 4.24  6  -0.314   0.764    
   treatment3 - treatment5     3.75 6.36  6   0.590   0.577    
   treatment3 - treatment6     3.75 6.36  6   0.590   0.577    
   treatment3 - treatment7   -11.50 6.36  6  -1.808   0.121    
   treatment3 - treatment8     8.75 6.36  6   1.376   0.218    
   treatment3 - treatment9     4.75 6.36  6   0.747   0.483    
  treatment3 - treatment10     4.75 6.36  6   0.747   0.483    
  treatment3 - treatment11    -4.50 6.36  6  -0.707   0.506    
  treatment3 - treatment12     2.50 6.36  6   0.393   0.708    
   treatment4 - treatment5     5.08 6.36  6   0.799   0.455    
   treatment4 - treatment6     5.08 6.36  6   0.799   0.455    
   treatment4 - treatment7   -10.17 6.36  6  -1.598   0.161    
   treatment4 - treatment8    10.08 6.36  6   1.585   0.164    
   treatment4 - treatment9     6.08 6.36  6   0.956   0.376    
  treatment4 - treatment10     6.08 6.36  6   0.956   0.376    
  treatment4 - treatment11    -3.17 6.36  6  -0.498   0.636    
  treatment4 - treatment12     3.83 6.36  6   0.603   0.569    
   treatment5 - treatment6     0.00 8.21  6   0.000   1.000    
   treatment5 - treatment7   -15.25 8.21  6  -1.857   0.113    
   treatment5 - treatment8     5.00 8.21  6   0.609   0.565    
   treatment5 - treatment9     1.00 7.34  6   0.136   0.896    
  treatment5 - treatment10     1.00 8.21  6   0.122   0.907    
  treatment5 - treatment11    -8.25 8.21  6  -1.005   0.354    
  treatment5 - treatment12    -1.25 8.21  6  -0.152   0.884    
   treatment6 - treatment7   -15.25 8.21  6  -1.857   0.113    
   treatment6 - treatment8     5.00 7.34  6   0.681   0.521    
   treatment6 - treatment9     1.00 8.21  6   0.122   0.907    
  treatment6 - treatment10     1.00 7.34  6   0.136   0.896    
  treatment6 - treatment11    -8.25 8.21  6  -1.005   0.354    
  treatment6 - treatment12    -1.25 8.21  6  -0.152   0.884    
   treatment7 - treatment8    20.25 8.21  6   2.466   0.049   *
   treatment7 - treatment9    16.25 8.21  6   1.979   0.095    
  treatment7 - treatment10    16.25 8.21  6   1.979   0.095    
  treatment7 - treatment11     7.00 7.34  6   0.953   0.377    
  treatment7 - treatment12    14.00 7.34  6   1.906   0.105    
   treatment8 - treatment9    -4.00 8.21  6  -0.487   0.643    
  treatment8 - treatment10    -4.00 7.34  6  -0.545   0.606    
  treatment8 - treatment11   -13.25 8.21  6  -1.614   0.158    
  treatment8 - treatment12    -6.25 8.21  6  -0.761   0.475    
  treatment9 - treatment10     0.00 8.21  6   0.000   1.000    
  treatment9 - treatment11    -9.25 8.21  6  -1.126   0.303    
  treatment9 - treatment12    -2.25 8.21  6  -0.274   0.793    
 treatment10 - treatment11    -9.25 8.21  6  -1.126   0.303    
 treatment10 - treatment12    -2.25 8.21  6  -0.274   0.793    
 treatment11 - treatment12     7.00 7.34  6   0.953   0.377    

Treatment Groups
================

Method : lsd

 Treatment Adjusted Means   SE df lower.CL upper.CL Group
         8          73.25 5.61  6    59.52    86.98    1 
         9          77.25 5.61  6    63.52    90.98    12
        10          77.25 5.61  6    63.52    90.98    12
         5          78.25 5.61  6    64.52    91.98    12
         6          78.25 5.61  6    64.52    91.98    12
         2          79.00 3.00  6    71.66    86.34    12
        12          79.50 5.61  6    65.77    93.23    12
         3          82.00 3.00  6    74.66    89.34    12
         4          83.33 3.00  6    76.00    90.67    12
         1          84.67 3.00  6    77.33    92.00    12
        11          86.50 5.61  6    72.77   100.23    12
         7          93.50 5.61  6    79.77   107.23     2</code></pre>
<div class="sourceCode" id="cb180"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">out1</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "augmentedRCBD"</code></pre>
<p>Similarly the analysis for the trait <code>y2</code> can be computed
as follows.</p>
<div class="sourceCode" id="cb182"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">blk</span>, <span class="va">trt</span>, <span class="va">y2</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,</span>
<span>                      alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"


ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Block (ignoring Treatments)           2   7019    3510  12.261 0.007597 ** 
Treatment (eliminating Blocks)       11  58965    5360  18.727 0.000920 ***
  Treatment: Check                    3   2150     717   2.504 0.156116    
  Treatment: Test and Test vs. Check  8  56815    7102  24.810 0.000473 ***
Residuals                             6   1718     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Treatment (ignoring Blocks)    11  64708    5883  20.550 0.000707 ***
  Treatment: Check              3   2150     717   2.504 0.156116    
  Treatment: Test               7  34863    4980  17.399 0.001366 ** 
  Treatment: Test vs. Check     1  27694   27694  96.749 6.36e-05 ***
Block (eliminating Treatments)  2   1277     639   2.231 0.188645    
Residuals                       6   1717     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Coefficient of Variation
========================
6.057617

Overall Adjusted Mean
=====================
298.4792

Standard Errors
===============
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             13.81424 33.80224
Two Test Treatments (Same Block)                    23.92697 58.54719
Two Test Treatments (Different Blocks)              26.75117 65.45775
A Test Treatment and a Control Treatment            21.84224 53.44603

Treatment Means
===============
 Treatment Block  Means    SE r    Min    Max Adjusted Means
         1       256.00  3.06 3 250.00 260.00         256.00
        10     3 450.00  &lt;NA&gt; 1 450.00 450.00         437.67
        11     1 300.00  &lt;NA&gt; 1 300.00 300.00         299.42
        12     1 289.00  &lt;NA&gt; 1 289.00 289.00         288.42
         2       228.00  6.11 3 220.00 240.00         228.00
         3       247.67 10.17 3 237.00 268.00         247.67
         4       264.00 18.68 3 227.00 287.00         264.00
         5     2 281.00  &lt;NA&gt; 1 281.00 281.00         293.92
         6     3 395.00  &lt;NA&gt; 1 395.00 395.00         382.67
         7     1 347.00  &lt;NA&gt; 1 347.00 347.00         346.42
         8     3 226.00  &lt;NA&gt; 1 226.00 226.00         213.67
         9     2 311.00  &lt;NA&gt; 1 311.00 311.00         323.92


Comparisons
===========

Method : lsd

                  contrast estimate    SE df t.ratio p.value sig
   treatment1 - treatment2    28.00 13.81  6   2.027   0.089    
   treatment1 - treatment3     8.33 13.81  6   0.603   0.568    
   treatment1 - treatment4    -8.00 13.81  6  -0.579   0.584    
   treatment1 - treatment5   -37.92 20.72  6  -1.830   0.117    
   treatment1 - treatment6  -126.67 20.72  6  -6.113   0.001 ***
   treatment1 - treatment7   -90.42 20.72  6  -4.363   0.005  **
   treatment1 - treatment8    42.33 20.72  6   2.043   0.087    
   treatment1 - treatment9   -67.92 20.72  6  -3.278   0.017   *
  treatment1 - treatment10  -181.67 20.72  6  -8.767   0.000 ***
  treatment1 - treatment11   -43.42 20.72  6  -2.095   0.081    
  treatment1 - treatment12   -32.42 20.72  6  -1.564   0.169    
   treatment2 - treatment3   -19.67 13.81  6  -1.424   0.204    
   treatment2 - treatment4   -36.00 13.81  6  -2.606   0.040   *
   treatment2 - treatment5   -65.92 20.72  6  -3.181   0.019   *
   treatment2 - treatment6  -154.67 20.72  6  -7.464   0.000 ***
   treatment2 - treatment7  -118.42 20.72  6  -5.715   0.001  **
   treatment2 - treatment8    14.33 20.72  6   0.692   0.515    
   treatment2 - treatment9   -95.92 20.72  6  -4.629   0.004  **
  treatment2 - treatment10  -209.67 20.72  6 -10.118   0.000 ***
  treatment2 - treatment11   -71.42 20.72  6  -3.447   0.014   *
  treatment2 - treatment12   -60.42 20.72  6  -2.916   0.027   *
   treatment3 - treatment4   -16.33 13.81  6  -1.182   0.282    
   treatment3 - treatment5   -46.25 20.72  6  -2.232   0.067    
   treatment3 - treatment6  -135.00 20.72  6  -6.515   0.001 ***
   treatment3 - treatment7   -98.75 20.72  6  -4.766   0.003  **
   treatment3 - treatment8    34.00 20.72  6   1.641   0.152    
   treatment3 - treatment9   -76.25 20.72  6  -3.680   0.010   *
  treatment3 - treatment10  -190.00 20.72  6  -9.169   0.000 ***
  treatment3 - treatment11   -51.75 20.72  6  -2.497   0.047   *
  treatment3 - treatment12   -40.75 20.72  6  -1.967   0.097    
   treatment4 - treatment5   -29.92 20.72  6  -1.444   0.199    
   treatment4 - treatment6  -118.67 20.72  6  -5.727   0.001  **
   treatment4 - treatment7   -82.42 20.72  6  -3.977   0.007  **
   treatment4 - treatment8    50.33 20.72  6   2.429   0.051    
   treatment4 - treatment9   -59.92 20.72  6  -2.892   0.028   *
  treatment4 - treatment10  -173.67 20.72  6  -8.381   0.000 ***
  treatment4 - treatment11   -35.42 20.72  6  -1.709   0.138    
  treatment4 - treatment12   -24.42 20.72  6  -1.178   0.283    
   treatment5 - treatment6   -88.75 26.75  6  -3.318   0.016   *
   treatment5 - treatment7   -52.50 26.75  6  -1.963   0.097    
   treatment5 - treatment8    80.25 26.75  6   3.000   0.024   *
   treatment5 - treatment9   -30.00 23.93  6  -1.254   0.257    
  treatment5 - treatment10  -143.75 26.75  6  -5.374   0.002  **
  treatment5 - treatment11    -5.50 26.75  6  -0.206   0.844    
  treatment5 - treatment12     5.50 26.75  6   0.206   0.844    
   treatment6 - treatment7    36.25 26.75  6   1.355   0.224    
   treatment6 - treatment8   169.00 23.93  6   7.063   0.000 ***
   treatment6 - treatment9    58.75 26.75  6   2.196   0.070    
  treatment6 - treatment10   -55.00 23.93  6  -2.299   0.061    
  treatment6 - treatment11    83.25 26.75  6   3.112   0.021   *
  treatment6 - treatment12    94.25 26.75  6   3.523   0.012   *
   treatment7 - treatment8   132.75 26.75  6   4.962   0.003  **
   treatment7 - treatment9    22.50 26.75  6   0.841   0.433    
  treatment7 - treatment10   -91.25 26.75  6  -3.411   0.014   *
  treatment7 - treatment11    47.00 23.93  6   1.964   0.097    
  treatment7 - treatment12    58.00 23.93  6   2.424   0.052    
   treatment8 - treatment9  -110.25 26.75  6  -4.121   0.006  **
  treatment8 - treatment10  -224.00 23.93  6  -9.362   0.000 ***
  treatment8 - treatment11   -85.75 26.75  6  -3.205   0.018   *
  treatment8 - treatment12   -74.75 26.75  6  -2.794   0.031   *
  treatment9 - treatment10  -113.75 26.75  6  -4.252   0.005  **
  treatment9 - treatment11    24.50 26.75  6   0.916   0.395    
  treatment9 - treatment12    35.50 26.75  6   1.327   0.233    
 treatment10 - treatment11   138.25 26.75  6   5.168   0.002  **
 treatment10 - treatment12   149.25 26.75  6   5.579   0.001  **
 treatment11 - treatment12    11.00 23.93  6   0.460   0.662    

Treatment Groups
================

Method : lsd

 Treatment Adjusted Means    SE df lower.CL upper.CL    Group
         8         213.67 18.27  6   168.95   258.38  12     
         2         228.00  9.77  6   204.10   251.90  1      
         3         247.67  9.77  6   223.76   271.57  123    
         1         256.00  9.77  6   232.10   279.90  1234   
         4         264.00  9.77  6   240.10   287.90   234   
        12         288.42 18.27  6   243.70   333.13    345  
         5         293.92 18.27  6   249.20   338.63    345  
        11         299.42 18.27  6   254.70   344.13     45  
         9         323.92 18.27  6   279.20   368.63      56 
         7         346.42 18.27  6   301.70   391.13      56 
         6         382.67 18.27  6   337.95   427.38       67
        10         437.67 18.27  6   392.95   482.38        7</code></pre>
<div class="sourceCode" id="cb184"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">out2</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "augmentedRCBD"</code></pre>
<p>The data can also be imported as a <a href="#dataframe">data
frame</a> and then used for analysis. Consider the data frame
<code>data</code> imported from <a href="#impexp">Table 1</a> according
to the instructions in <a href="#impexp">section 4.8</a>.</p>
<div class="sourceCode" id="cb186"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span></code></pre></div>
<pre><code>'data.frame':   20 obs. of  4 variables:
 $ blk: Factor w/ 3 levels "1","2","3": 1 1 1 1 1 1 1 2 2 2 ...
 $ trt: Factor w/ 12 levels "1","2","3","4",..: 1 2 3 4 7 11 12 1 2 3 ...
 $ y1 : num  92 79 87 81 96 89 82 79 81 81 ...
 $ y2 : num  258 224 238 278 347 300 289 260 220 237 ...</code></pre>
<div class="sourceCode" id="cb188"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert block and treatment to factors</span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">blk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span><span class="op">)</span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">trt</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb189"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Results for variable y1</span></span>
<span><span class="va">out1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span>, <span class="va">data</span><span class="op">$</span><span class="va">trt</span>, <span class="va">data</span><span class="op">$</span><span class="va">y1</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,</span>
<span>                      alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"


ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value Pr(&gt;F)  
Block (ignoring Treatments)           2  360.1  180.04   6.675 0.0298 *
Treatment (eliminating Blocks)       11  285.1   25.92   0.961 0.5499  
  Treatment: Check                    3   52.9   17.64   0.654 0.6092  
  Treatment: Test and Test vs. Check  8  232.2   29.02   1.076 0.4779  
Residuals                             6  161.8   26.97                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value Pr(&gt;F)
Treatment (ignoring Blocks)    11  575.7   52.33   1.940  0.215
  Treatment: Check              3   52.9   17.64   0.654  0.609
  Treatment: Test               7  505.9   72.27   2.679  0.125
  Treatment: Test vs. Check     1   16.9   16.87   0.626  0.459
Block (eliminating Treatments)  2   69.5   34.75   1.288  0.342
Residuals                       6  161.8   26.97               

Coefficient of Variation
========================
6.372367

Overall Adjusted Mean
=====================
81.0625

Standard Errors
===============
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             4.240458 10.37603
Two Test Treatments (Same Block)                    7.344688 17.97180
Two Test Treatments (Different Blocks)              8.211611 20.09309
A Test Treatment and a Control Treatment            6.704752 16.40594

Treatment Means
===============
 Treatment Block Means   SE r   Min   Max Adjusted Means
         1       84.67 3.84 3 79.00 92.00          84.67
        10     3 74.00 &lt;NA&gt; 1 74.00 74.00          77.25
        11     1 89.00 &lt;NA&gt; 1 89.00 89.00          86.50
        12     1 82.00 &lt;NA&gt; 1 82.00 82.00          79.50
         2       79.00 1.15 3 77.00 81.00          79.00
         3       82.00 2.65 3 78.00 87.00          82.00
         4       83.33 3.93 3 78.00 91.00          83.33
         5     2 79.00 &lt;NA&gt; 1 79.00 79.00          78.25
         6     3 75.00 &lt;NA&gt; 1 75.00 75.00          78.25
         7     1 96.00 &lt;NA&gt; 1 96.00 96.00          93.50
         8     3 70.00 &lt;NA&gt; 1 70.00 70.00          73.25
         9     2 78.00 &lt;NA&gt; 1 78.00 78.00          77.25


Comparisons
===========

Method : lsd

                  contrast estimate   SE df t.ratio p.value sig
   treatment1 - treatment2     5.67 4.24  6   1.336   0.230    
   treatment1 - treatment3     2.67 4.24  6   0.629   0.553    
   treatment1 - treatment4     1.33 4.24  6   0.314   0.764    
   treatment1 - treatment5     6.42 6.36  6   1.009   0.352    
   treatment1 - treatment6     6.42 6.36  6   1.009   0.352    
   treatment1 - treatment7    -8.83 6.36  6  -1.389   0.214    
   treatment1 - treatment8    11.42 6.36  6   1.795   0.123    
   treatment1 - treatment9     7.42 6.36  6   1.166   0.288    
  treatment1 - treatment10     7.42 6.36  6   1.166   0.288    
  treatment1 - treatment11    -1.83 6.36  6  -0.288   0.783    
  treatment1 - treatment12     5.17 6.36  6   0.812   0.448    
   treatment2 - treatment3    -3.00 4.24  6  -0.707   0.506    
   treatment2 - treatment4    -4.33 4.24  6  -1.022   0.346    
   treatment2 - treatment5     0.75 6.36  6   0.118   0.910    
   treatment2 - treatment6     0.75 6.36  6   0.118   0.910    
   treatment2 - treatment7   -14.50 6.36  6  -2.280   0.063    
   treatment2 - treatment8     5.75 6.36  6   0.904   0.401    
   treatment2 - treatment9     1.75 6.36  6   0.275   0.792    
  treatment2 - treatment10     1.75 6.36  6   0.275   0.792    
  treatment2 - treatment11    -7.50 6.36  6  -1.179   0.283    
  treatment2 - treatment12    -0.50 6.36  6  -0.079   0.940    
   treatment3 - treatment4    -1.33 4.24  6  -0.314   0.764    
   treatment3 - treatment5     3.75 6.36  6   0.590   0.577    
   treatment3 - treatment6     3.75 6.36  6   0.590   0.577    
   treatment3 - treatment7   -11.50 6.36  6  -1.808   0.121    
   treatment3 - treatment8     8.75 6.36  6   1.376   0.218    
   treatment3 - treatment9     4.75 6.36  6   0.747   0.483    
  treatment3 - treatment10     4.75 6.36  6   0.747   0.483    
  treatment3 - treatment11    -4.50 6.36  6  -0.707   0.506    
  treatment3 - treatment12     2.50 6.36  6   0.393   0.708    
   treatment4 - treatment5     5.08 6.36  6   0.799   0.455    
   treatment4 - treatment6     5.08 6.36  6   0.799   0.455    
   treatment4 - treatment7   -10.17 6.36  6  -1.598   0.161    
   treatment4 - treatment8    10.08 6.36  6   1.585   0.164    
   treatment4 - treatment9     6.08 6.36  6   0.956   0.376    
  treatment4 - treatment10     6.08 6.36  6   0.956   0.376    
  treatment4 - treatment11    -3.17 6.36  6  -0.498   0.636    
  treatment4 - treatment12     3.83 6.36  6   0.603   0.569    
   treatment5 - treatment6     0.00 8.21  6   0.000   1.000    
   treatment5 - treatment7   -15.25 8.21  6  -1.857   0.113    
   treatment5 - treatment8     5.00 8.21  6   0.609   0.565    
   treatment5 - treatment9     1.00 7.34  6   0.136   0.896    
  treatment5 - treatment10     1.00 8.21  6   0.122   0.907    
  treatment5 - treatment11    -8.25 8.21  6  -1.005   0.354    
  treatment5 - treatment12    -1.25 8.21  6  -0.152   0.884    
   treatment6 - treatment7   -15.25 8.21  6  -1.857   0.113    
   treatment6 - treatment8     5.00 7.34  6   0.681   0.521    
   treatment6 - treatment9     1.00 8.21  6   0.122   0.907    
  treatment6 - treatment10     1.00 7.34  6   0.136   0.896    
  treatment6 - treatment11    -8.25 8.21  6  -1.005   0.354    
  treatment6 - treatment12    -1.25 8.21  6  -0.152   0.884    
   treatment7 - treatment8    20.25 8.21  6   2.466   0.049   *
   treatment7 - treatment9    16.25 8.21  6   1.979   0.095    
  treatment7 - treatment10    16.25 8.21  6   1.979   0.095    
  treatment7 - treatment11     7.00 7.34  6   0.953   0.377    
  treatment7 - treatment12    14.00 7.34  6   1.906   0.105    
   treatment8 - treatment9    -4.00 8.21  6  -0.487   0.643    
  treatment8 - treatment10    -4.00 7.34  6  -0.545   0.606    
  treatment8 - treatment11   -13.25 8.21  6  -1.614   0.158    
  treatment8 - treatment12    -6.25 8.21  6  -0.761   0.475    
  treatment9 - treatment10     0.00 8.21  6   0.000   1.000    
  treatment9 - treatment11    -9.25 8.21  6  -1.126   0.303    
  treatment9 - treatment12    -2.25 8.21  6  -0.274   0.793    
 treatment10 - treatment11    -9.25 8.21  6  -1.126   0.303    
 treatment10 - treatment12    -2.25 8.21  6  -0.274   0.793    
 treatment11 - treatment12     7.00 7.34  6   0.953   0.377    

Treatment Groups
================

Method : lsd

 Treatment Adjusted Means   SE df lower.CL upper.CL Group
         8          73.25 5.61  6    59.52    86.98    1 
         9          77.25 5.61  6    63.52    90.98    12
        10          77.25 5.61  6    63.52    90.98    12
         5          78.25 5.61  6    64.52    91.98    12
         6          78.25 5.61  6    64.52    91.98    12
         2          79.00 3.00  6    71.66    86.34    12
        12          79.50 5.61  6    65.77    93.23    12
         3          82.00 3.00  6    74.66    89.34    12
         4          83.33 3.00  6    76.00    90.67    12
         1          84.67 3.00  6    77.33    92.00    12
        11          86.50 5.61  6    72.77   100.23    12
         7          93.50 5.61  6    79.77   107.23     2</code></pre>
<div class="sourceCode" id="cb191"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">out1</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "augmentedRCBD"</code></pre>
<div class="sourceCode" id="cb193"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Results for variable y2</span></span>
<span><span class="va">out2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span>, <span class="va">data</span><span class="op">$</span><span class="va">trt</span>, <span class="va">data</span><span class="op">$</span><span class="va">y2</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,</span>
<span>                     alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"


ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Block (ignoring Treatments)           2   7019    3510  12.261 0.007597 ** 
Treatment (eliminating Blocks)       11  58965    5360  18.727 0.000920 ***
  Treatment: Check                    3   2150     717   2.504 0.156116    
  Treatment: Test and Test vs. Check  8  56815    7102  24.810 0.000473 ***
Residuals                             6   1718     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Treatment (ignoring Blocks)    11  64708    5883  20.550 0.000707 ***
  Treatment: Check              3   2150     717   2.504 0.156116    
  Treatment: Test               7  34863    4980  17.399 0.001366 ** 
  Treatment: Test vs. Check     1  27694   27694  96.749 6.36e-05 ***
Block (eliminating Treatments)  2   1277     639   2.231 0.188645    
Residuals                       6   1717     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Coefficient of Variation
========================
6.057617

Overall Adjusted Mean
=====================
298.4792

Standard Errors
===============
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             13.81424 33.80224
Two Test Treatments (Same Block)                    23.92697 58.54719
Two Test Treatments (Different Blocks)              26.75117 65.45775
A Test Treatment and a Control Treatment            21.84224 53.44603

Treatment Means
===============
 Treatment Block  Means    SE r    Min    Max Adjusted Means
         1       256.00  3.06 3 250.00 260.00         256.00
        10     3 450.00  &lt;NA&gt; 1 450.00 450.00         437.67
        11     1 300.00  &lt;NA&gt; 1 300.00 300.00         299.42
        12     1 289.00  &lt;NA&gt; 1 289.00 289.00         288.42
         2       228.00  6.11 3 220.00 240.00         228.00
         3       247.67 10.17 3 237.00 268.00         247.67
         4       264.00 18.68 3 227.00 287.00         264.00
         5     2 281.00  &lt;NA&gt; 1 281.00 281.00         293.92
         6     3 395.00  &lt;NA&gt; 1 395.00 395.00         382.67
         7     1 347.00  &lt;NA&gt; 1 347.00 347.00         346.42
         8     3 226.00  &lt;NA&gt; 1 226.00 226.00         213.67
         9     2 311.00  &lt;NA&gt; 1 311.00 311.00         323.92


Comparisons
===========

Method : lsd

                  contrast estimate    SE df t.ratio p.value sig
   treatment1 - treatment2    28.00 13.81  6   2.027   0.089    
   treatment1 - treatment3     8.33 13.81  6   0.603   0.568    
   treatment1 - treatment4    -8.00 13.81  6  -0.579   0.584    
   treatment1 - treatment5   -37.92 20.72  6  -1.830   0.117    
   treatment1 - treatment6  -126.67 20.72  6  -6.113   0.001 ***
   treatment1 - treatment7   -90.42 20.72  6  -4.363   0.005  **
   treatment1 - treatment8    42.33 20.72  6   2.043   0.087    
   treatment1 - treatment9   -67.92 20.72  6  -3.278   0.017   *
  treatment1 - treatment10  -181.67 20.72  6  -8.767   0.000 ***
  treatment1 - treatment11   -43.42 20.72  6  -2.095   0.081    
  treatment1 - treatment12   -32.42 20.72  6  -1.564   0.169    
   treatment2 - treatment3   -19.67 13.81  6  -1.424   0.204    
   treatment2 - treatment4   -36.00 13.81  6  -2.606   0.040   *
   treatment2 - treatment5   -65.92 20.72  6  -3.181   0.019   *
   treatment2 - treatment6  -154.67 20.72  6  -7.464   0.000 ***
   treatment2 - treatment7  -118.42 20.72  6  -5.715   0.001  **
   treatment2 - treatment8    14.33 20.72  6   0.692   0.515    
   treatment2 - treatment9   -95.92 20.72  6  -4.629   0.004  **
  treatment2 - treatment10  -209.67 20.72  6 -10.118   0.000 ***
  treatment2 - treatment11   -71.42 20.72  6  -3.447   0.014   *
  treatment2 - treatment12   -60.42 20.72  6  -2.916   0.027   *
   treatment3 - treatment4   -16.33 13.81  6  -1.182   0.282    
   treatment3 - treatment5   -46.25 20.72  6  -2.232   0.067    
   treatment3 - treatment6  -135.00 20.72  6  -6.515   0.001 ***
   treatment3 - treatment7   -98.75 20.72  6  -4.766   0.003  **
   treatment3 - treatment8    34.00 20.72  6   1.641   0.152    
   treatment3 - treatment9   -76.25 20.72  6  -3.680   0.010   *
  treatment3 - treatment10  -190.00 20.72  6  -9.169   0.000 ***
  treatment3 - treatment11   -51.75 20.72  6  -2.497   0.047   *
  treatment3 - treatment12   -40.75 20.72  6  -1.967   0.097    
   treatment4 - treatment5   -29.92 20.72  6  -1.444   0.199    
   treatment4 - treatment6  -118.67 20.72  6  -5.727   0.001  **
   treatment4 - treatment7   -82.42 20.72  6  -3.977   0.007  **
   treatment4 - treatment8    50.33 20.72  6   2.429   0.051    
   treatment4 - treatment9   -59.92 20.72  6  -2.892   0.028   *
  treatment4 - treatment10  -173.67 20.72  6  -8.381   0.000 ***
  treatment4 - treatment11   -35.42 20.72  6  -1.709   0.138    
  treatment4 - treatment12   -24.42 20.72  6  -1.178   0.283    
   treatment5 - treatment6   -88.75 26.75  6  -3.318   0.016   *
   treatment5 - treatment7   -52.50 26.75  6  -1.963   0.097    
   treatment5 - treatment8    80.25 26.75  6   3.000   0.024   *
   treatment5 - treatment9   -30.00 23.93  6  -1.254   0.257    
  treatment5 - treatment10  -143.75 26.75  6  -5.374   0.002  **
  treatment5 - treatment11    -5.50 26.75  6  -0.206   0.844    
  treatment5 - treatment12     5.50 26.75  6   0.206   0.844    
   treatment6 - treatment7    36.25 26.75  6   1.355   0.224    
   treatment6 - treatment8   169.00 23.93  6   7.063   0.000 ***
   treatment6 - treatment9    58.75 26.75  6   2.196   0.070    
  treatment6 - treatment10   -55.00 23.93  6  -2.299   0.061    
  treatment6 - treatment11    83.25 26.75  6   3.112   0.021   *
  treatment6 - treatment12    94.25 26.75  6   3.523   0.012   *
   treatment7 - treatment8   132.75 26.75  6   4.962   0.003  **
   treatment7 - treatment9    22.50 26.75  6   0.841   0.433    
  treatment7 - treatment10   -91.25 26.75  6  -3.411   0.014   *
  treatment7 - treatment11    47.00 23.93  6   1.964   0.097    
  treatment7 - treatment12    58.00 23.93  6   2.424   0.052    
   treatment8 - treatment9  -110.25 26.75  6  -4.121   0.006  **
  treatment8 - treatment10  -224.00 23.93  6  -9.362   0.000 ***
  treatment8 - treatment11   -85.75 26.75  6  -3.205   0.018   *
  treatment8 - treatment12   -74.75 26.75  6  -2.794   0.031   *
  treatment9 - treatment10  -113.75 26.75  6  -4.252   0.005  **
  treatment9 - treatment11    24.50 26.75  6   0.916   0.395    
  treatment9 - treatment12    35.50 26.75  6   1.327   0.233    
 treatment10 - treatment11   138.25 26.75  6   5.168   0.002  **
 treatment10 - treatment12   149.25 26.75  6   5.579   0.001  **
 treatment11 - treatment12    11.00 23.93  6   0.460   0.662    

Treatment Groups
================

Method : lsd

 Treatment Adjusted Means    SE df lower.CL upper.CL    Group
         8         213.67 18.27  6   168.95   258.38  12     
         2         228.00  9.77  6   204.10   251.90  1      
         3         247.67  9.77  6   223.76   271.57  123    
         1         256.00  9.77  6   232.10   279.90  1234   
         4         264.00  9.77  6   240.10   287.90   234   
        12         288.42 18.27  6   243.70   333.13    345  
         5         293.92 18.27  6   249.20   338.63    345  
        11         299.42 18.27  6   254.70   344.13     45  
         9         323.92 18.27  6   279.20   368.63      56 
         7         346.42 18.27  6   301.70   391.13      56 
         6         382.67 18.27  6   337.95   427.38       67
        10         437.67 18.27  6   392.95   482.38        7</code></pre>
<div class="sourceCode" id="cb195"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">out2</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "augmentedRCBD"</code></pre>
<p>Check genotypes are inferred by default on the basis of number of
replications. However, if some test genotypes are also replicated, they
may also be falsely detected as checks. To avoid this, the checks can be
specified by the <code>checks</code> argument.</p>
<div class="sourceCode" id="cb197"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Results for variable y1 (checks specified)</span></span>
<span><span class="va">out1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span>, <span class="va">data</span><span class="op">$</span><span class="va">trt</span>, <span class="va">data</span><span class="op">$</span><span class="va">y1</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,</span>
<span>                      alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                      checks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"


ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value Pr(&gt;F)  
Block (ignoring Treatments)           2  360.1  180.04   6.675 0.0298 *
Treatment (eliminating Blocks)       11  285.1   25.92   0.961 0.5499  
  Treatment: Check                    3   52.9   17.64   0.654 0.6092  
  Treatment: Test and Test vs. Check  8  232.2   29.02   1.076 0.4779  
Residuals                             6  161.8   26.97                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value Pr(&gt;F)
Treatment (ignoring Blocks)    11  575.7   52.33   1.940  0.215
  Treatment: Check              3   52.9   17.64   0.654  0.609
  Treatment: Test               7  505.9   72.27   2.679  0.125
  Treatment: Test vs. Check     1   16.9   16.87   0.626  0.459
Block (eliminating Treatments)  2   69.5   34.75   1.288  0.342
Residuals                       6  161.8   26.97               

Coefficient of Variation
========================
6.372367

Overall Adjusted Mean
=====================
81.0625

Standard Errors
===============
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             4.240458 10.37603
Two Test Treatments (Same Block)                    7.344688 17.97180
Two Test Treatments (Different Blocks)              8.211611 20.09309
A Test Treatment and a Control Treatment            6.704752 16.40594

Treatment Means
===============
 Treatment Block Means   SE r   Min   Max Adjusted Means
         1       84.67 3.84 3 79.00 92.00          84.67
        10     3 74.00 &lt;NA&gt; 1 74.00 74.00          77.25
        11     1 89.00 &lt;NA&gt; 1 89.00 89.00          86.50
        12     1 82.00 &lt;NA&gt; 1 82.00 82.00          79.50
         2       79.00 1.15 3 77.00 81.00          79.00
         3       82.00 2.65 3 78.00 87.00          82.00
         4       83.33 3.93 3 78.00 91.00          83.33
         5     2 79.00 &lt;NA&gt; 1 79.00 79.00          78.25
         6     3 75.00 &lt;NA&gt; 1 75.00 75.00          78.25
         7     1 96.00 &lt;NA&gt; 1 96.00 96.00          93.50
         8     3 70.00 &lt;NA&gt; 1 70.00 70.00          73.25
         9     2 78.00 &lt;NA&gt; 1 78.00 78.00          77.25


Comparisons
===========

Method : lsd

                  contrast estimate   SE df t.ratio p.value sig
   treatment1 - treatment2     5.67 4.24  6   1.336   0.230    
   treatment1 - treatment3     2.67 4.24  6   0.629   0.553    
   treatment1 - treatment4     1.33 4.24  6   0.314   0.764    
   treatment1 - treatment5     6.42 6.36  6   1.009   0.352    
   treatment1 - treatment6     6.42 6.36  6   1.009   0.352    
   treatment1 - treatment7    -8.83 6.36  6  -1.389   0.214    
   treatment1 - treatment8    11.42 6.36  6   1.795   0.123    
   treatment1 - treatment9     7.42 6.36  6   1.166   0.288    
  treatment1 - treatment10     7.42 6.36  6   1.166   0.288    
  treatment1 - treatment11    -1.83 6.36  6  -0.288   0.783    
  treatment1 - treatment12     5.17 6.36  6   0.812   0.448    
   treatment2 - treatment3    -3.00 4.24  6  -0.707   0.506    
   treatment2 - treatment4    -4.33 4.24  6  -1.022   0.346    
   treatment2 - treatment5     0.75 6.36  6   0.118   0.910    
   treatment2 - treatment6     0.75 6.36  6   0.118   0.910    
   treatment2 - treatment7   -14.50 6.36  6  -2.280   0.063    
   treatment2 - treatment8     5.75 6.36  6   0.904   0.401    
   treatment2 - treatment9     1.75 6.36  6   0.275   0.792    
  treatment2 - treatment10     1.75 6.36  6   0.275   0.792    
  treatment2 - treatment11    -7.50 6.36  6  -1.179   0.283    
  treatment2 - treatment12    -0.50 6.36  6  -0.079   0.940    
   treatment3 - treatment4    -1.33 4.24  6  -0.314   0.764    
   treatment3 - treatment5     3.75 6.36  6   0.590   0.577    
   treatment3 - treatment6     3.75 6.36  6   0.590   0.577    
   treatment3 - treatment7   -11.50 6.36  6  -1.808   0.121    
   treatment3 - treatment8     8.75 6.36  6   1.376   0.218    
   treatment3 - treatment9     4.75 6.36  6   0.747   0.483    
  treatment3 - treatment10     4.75 6.36  6   0.747   0.483    
  treatment3 - treatment11    -4.50 6.36  6  -0.707   0.506    
  treatment3 - treatment12     2.50 6.36  6   0.393   0.708    
   treatment4 - treatment5     5.08 6.36  6   0.799   0.455    
   treatment4 - treatment6     5.08 6.36  6   0.799   0.455    
   treatment4 - treatment7   -10.17 6.36  6  -1.598   0.161    
   treatment4 - treatment8    10.08 6.36  6   1.585   0.164    
   treatment4 - treatment9     6.08 6.36  6   0.956   0.376    
  treatment4 - treatment10     6.08 6.36  6   0.956   0.376    
  treatment4 - treatment11    -3.17 6.36  6  -0.498   0.636    
  treatment4 - treatment12     3.83 6.36  6   0.603   0.569    
   treatment5 - treatment6     0.00 8.21  6   0.000   1.000    
   treatment5 - treatment7   -15.25 8.21  6  -1.857   0.113    
   treatment5 - treatment8     5.00 8.21  6   0.609   0.565    
   treatment5 - treatment9     1.00 7.34  6   0.136   0.896    
  treatment5 - treatment10     1.00 8.21  6   0.122   0.907    
  treatment5 - treatment11    -8.25 8.21  6  -1.005   0.354    
  treatment5 - treatment12    -1.25 8.21  6  -0.152   0.884    
   treatment6 - treatment7   -15.25 8.21  6  -1.857   0.113    
   treatment6 - treatment8     5.00 7.34  6   0.681   0.521    
   treatment6 - treatment9     1.00 8.21  6   0.122   0.907    
  treatment6 - treatment10     1.00 7.34  6   0.136   0.896    
  treatment6 - treatment11    -8.25 8.21  6  -1.005   0.354    
  treatment6 - treatment12    -1.25 8.21  6  -0.152   0.884    
   treatment7 - treatment8    20.25 8.21  6   2.466   0.049   *
   treatment7 - treatment9    16.25 8.21  6   1.979   0.095    
  treatment7 - treatment10    16.25 8.21  6   1.979   0.095    
  treatment7 - treatment11     7.00 7.34  6   0.953   0.377    
  treatment7 - treatment12    14.00 7.34  6   1.906   0.105    
   treatment8 - treatment9    -4.00 8.21  6  -0.487   0.643    
  treatment8 - treatment10    -4.00 7.34  6  -0.545   0.606    
  treatment8 - treatment11   -13.25 8.21  6  -1.614   0.158    
  treatment8 - treatment12    -6.25 8.21  6  -0.761   0.475    
  treatment9 - treatment10     0.00 8.21  6   0.000   1.000    
  treatment9 - treatment11    -9.25 8.21  6  -1.126   0.303    
  treatment9 - treatment12    -2.25 8.21  6  -0.274   0.793    
 treatment10 - treatment11    -9.25 8.21  6  -1.126   0.303    
 treatment10 - treatment12    -2.25 8.21  6  -0.274   0.793    
 treatment11 - treatment12     7.00 7.34  6   0.953   0.377    

Treatment Groups
================

Method : lsd

 Treatment Adjusted Means   SE df lower.CL upper.CL Group
         8          73.25 5.61  6    59.52    86.98    1 
         9          77.25 5.61  6    63.52    90.98    12
        10          77.25 5.61  6    63.52    90.98    12
         5          78.25 5.61  6    64.52    91.98    12
         6          78.25 5.61  6    64.52    91.98    12
         2          79.00 3.00  6    71.66    86.34    12
        12          79.50 5.61  6    65.77    93.23    12
         3          82.00 3.00  6    74.66    89.34    12
         4          83.33 3.00  6    76.00    90.67    12
         1          84.67 3.00  6    77.33    92.00    12
        11          86.50 5.61  6    72.77   100.23    12
         7          93.50 5.61  6    79.77   107.23     2</code></pre>
<div class="sourceCode" id="cb199"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Results for variable y2 (checks specified)</span></span>
<span><span class="va">out2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.html">augmentedRCBD</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span>, <span class="va">data</span><span class="op">$</span><span class="va">trt</span>, <span class="va">data</span><span class="op">$</span><span class="va">y2</span>, method.comp <span class="op">=</span> <span class="st">"lsd"</span>,</span>
<span>                      alpha <span class="op">=</span> <span class="fl">0.05</span>, group <span class="op">=</span> <span class="cn">TRUE</span>, console <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                      checks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"


ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Block (ignoring Treatments)           2   7019    3510  12.261 0.007597 ** 
Treatment (eliminating Blocks)       11  58965    5360  18.727 0.000920 ***
  Treatment: Check                    3   2150     717   2.504 0.156116    
  Treatment: Test and Test vs. Check  8  56815    7102  24.810 0.000473 ***
Residuals                             6   1718     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Treatment (ignoring Blocks)    11  64708    5883  20.550 0.000707 ***
  Treatment: Check              3   2150     717   2.504 0.156116    
  Treatment: Test               7  34863    4980  17.399 0.001366 ** 
  Treatment: Test vs. Check     1  27694   27694  96.749 6.36e-05 ***
Block (eliminating Treatments)  2   1277     639   2.231 0.188645    
Residuals                       6   1717     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Coefficient of Variation
========================
6.057617

Overall Adjusted Mean
=====================
298.4792

Standard Errors
===============
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             13.81424 33.80224
Two Test Treatments (Same Block)                    23.92697 58.54719
Two Test Treatments (Different Blocks)              26.75117 65.45775
A Test Treatment and a Control Treatment            21.84224 53.44603

Treatment Means
===============
 Treatment Block  Means    SE r    Min    Max Adjusted Means
         1       256.00  3.06 3 250.00 260.00         256.00
        10     3 450.00  &lt;NA&gt; 1 450.00 450.00         437.67
        11     1 300.00  &lt;NA&gt; 1 300.00 300.00         299.42
        12     1 289.00  &lt;NA&gt; 1 289.00 289.00         288.42
         2       228.00  6.11 3 220.00 240.00         228.00
         3       247.67 10.17 3 237.00 268.00         247.67
         4       264.00 18.68 3 227.00 287.00         264.00
         5     2 281.00  &lt;NA&gt; 1 281.00 281.00         293.92
         6     3 395.00  &lt;NA&gt; 1 395.00 395.00         382.67
         7     1 347.00  &lt;NA&gt; 1 347.00 347.00         346.42
         8     3 226.00  &lt;NA&gt; 1 226.00 226.00         213.67
         9     2 311.00  &lt;NA&gt; 1 311.00 311.00         323.92


Comparisons
===========

Method : lsd

                  contrast estimate    SE df t.ratio p.value sig
   treatment1 - treatment2    28.00 13.81  6   2.027   0.089    
   treatment1 - treatment3     8.33 13.81  6   0.603   0.568    
   treatment1 - treatment4    -8.00 13.81  6  -0.579   0.584    
   treatment1 - treatment5   -37.92 20.72  6  -1.830   0.117    
   treatment1 - treatment6  -126.67 20.72  6  -6.113   0.001 ***
   treatment1 - treatment7   -90.42 20.72  6  -4.363   0.005  **
   treatment1 - treatment8    42.33 20.72  6   2.043   0.087    
   treatment1 - treatment9   -67.92 20.72  6  -3.278   0.017   *
  treatment1 - treatment10  -181.67 20.72  6  -8.767   0.000 ***
  treatment1 - treatment11   -43.42 20.72  6  -2.095   0.081    
  treatment1 - treatment12   -32.42 20.72  6  -1.564   0.169    
   treatment2 - treatment3   -19.67 13.81  6  -1.424   0.204    
   treatment2 - treatment4   -36.00 13.81  6  -2.606   0.040   *
   treatment2 - treatment5   -65.92 20.72  6  -3.181   0.019   *
   treatment2 - treatment6  -154.67 20.72  6  -7.464   0.000 ***
   treatment2 - treatment7  -118.42 20.72  6  -5.715   0.001  **
   treatment2 - treatment8    14.33 20.72  6   0.692   0.515    
   treatment2 - treatment9   -95.92 20.72  6  -4.629   0.004  **
  treatment2 - treatment10  -209.67 20.72  6 -10.118   0.000 ***
  treatment2 - treatment11   -71.42 20.72  6  -3.447   0.014   *
  treatment2 - treatment12   -60.42 20.72  6  -2.916   0.027   *
   treatment3 - treatment4   -16.33 13.81  6  -1.182   0.282    
   treatment3 - treatment5   -46.25 20.72  6  -2.232   0.067    
   treatment3 - treatment6  -135.00 20.72  6  -6.515   0.001 ***
   treatment3 - treatment7   -98.75 20.72  6  -4.766   0.003  **
   treatment3 - treatment8    34.00 20.72  6   1.641   0.152    
   treatment3 - treatment9   -76.25 20.72  6  -3.680   0.010   *
  treatment3 - treatment10  -190.00 20.72  6  -9.169   0.000 ***
  treatment3 - treatment11   -51.75 20.72  6  -2.497   0.047   *
  treatment3 - treatment12   -40.75 20.72  6  -1.967   0.097    
   treatment4 - treatment5   -29.92 20.72  6  -1.444   0.199    
   treatment4 - treatment6  -118.67 20.72  6  -5.727   0.001  **
   treatment4 - treatment7   -82.42 20.72  6  -3.977   0.007  **
   treatment4 - treatment8    50.33 20.72  6   2.429   0.051    
   treatment4 - treatment9   -59.92 20.72  6  -2.892   0.028   *
  treatment4 - treatment10  -173.67 20.72  6  -8.381   0.000 ***
  treatment4 - treatment11   -35.42 20.72  6  -1.709   0.138    
  treatment4 - treatment12   -24.42 20.72  6  -1.178   0.283    
   treatment5 - treatment6   -88.75 26.75  6  -3.318   0.016   *
   treatment5 - treatment7   -52.50 26.75  6  -1.963   0.097    
   treatment5 - treatment8    80.25 26.75  6   3.000   0.024   *
   treatment5 - treatment9   -30.00 23.93  6  -1.254   0.257    
  treatment5 - treatment10  -143.75 26.75  6  -5.374   0.002  **
  treatment5 - treatment11    -5.50 26.75  6  -0.206   0.844    
  treatment5 - treatment12     5.50 26.75  6   0.206   0.844    
   treatment6 - treatment7    36.25 26.75  6   1.355   0.224    
   treatment6 - treatment8   169.00 23.93  6   7.063   0.000 ***
   treatment6 - treatment9    58.75 26.75  6   2.196   0.070    
  treatment6 - treatment10   -55.00 23.93  6  -2.299   0.061    
  treatment6 - treatment11    83.25 26.75  6   3.112   0.021   *
  treatment6 - treatment12    94.25 26.75  6   3.523   0.012   *
   treatment7 - treatment8   132.75 26.75  6   4.962   0.003  **
   treatment7 - treatment9    22.50 26.75  6   0.841   0.433    
  treatment7 - treatment10   -91.25 26.75  6  -3.411   0.014   *
  treatment7 - treatment11    47.00 23.93  6   1.964   0.097    
  treatment7 - treatment12    58.00 23.93  6   2.424   0.052    
   treatment8 - treatment9  -110.25 26.75  6  -4.121   0.006  **
  treatment8 - treatment10  -224.00 23.93  6  -9.362   0.000 ***
  treatment8 - treatment11   -85.75 26.75  6  -3.205   0.018   *
  treatment8 - treatment12   -74.75 26.75  6  -2.794   0.031   *
  treatment9 - treatment10  -113.75 26.75  6  -4.252   0.005  **
  treatment9 - treatment11    24.50 26.75  6   0.916   0.395    
  treatment9 - treatment12    35.50 26.75  6   1.327   0.233    
 treatment10 - treatment11   138.25 26.75  6   5.168   0.002  **
 treatment10 - treatment12   149.25 26.75  6   5.579   0.001  **
 treatment11 - treatment12    11.00 23.93  6   0.460   0.662    

Treatment Groups
================

Method : lsd

 Treatment Adjusted Means    SE df lower.CL upper.CL    Group
         8         213.67 18.27  6   168.95   258.38  12     
         2         228.00  9.77  6   204.10   251.90  1      
         3         247.67  9.77  6   223.76   271.57  123    
         1         256.00  9.77  6   232.10   279.90  1234   
         4         264.00  9.77  6   240.10   287.90   234   
        12         288.42 18.27  6   243.70   333.13    345  
         5         293.92 18.27  6   249.20   338.63    345  
        11         299.42 18.27  6   254.70   344.13     45  
         9         323.92 18.27  6   279.20   368.63      56 
         7         346.42 18.27  6   301.70   391.13      56 
         6         382.67 18.27  6   337.95   427.38       67
        10         437.67 18.27  6   392.95   482.38        7</code></pre>
<p>In case the large number of treatments or genotypes, it is advisable
to avoid treatment comparisons with the <code>group = FALSE</code>
argument as it will be memory and processor intensive. Further it is
advised to simplify output with <code>simplify = TRUE</code> in order to
reduce output object size.</p>
<p>If <code>truncate.means = TRUE</code>, then any negative adjusted
means will be truncated to zero with a warning.</p>
</div>
<div class="section level3">
<h3 id="print-augmentedrcbd">7.2 <code>print.augmentedRCBD()</code><a class="anchor" aria-label="anchor" href="#print-augmentedrcbd"></a>
</h3>
<p>The results of analysis in an object of class
<code>augmentedRCBD</code> can be printed to the console as follows.</p>
<div class="sourceCode" id="cb201"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Print results for variable y1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">out1</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"


ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value Pr(&gt;F)  
Block (ignoring Treatments)           2  360.1  180.04   6.675 0.0298 *
Treatment (eliminating Blocks)       11  285.1   25.92   0.961 0.5499  
  Treatment: Check                    3   52.9   17.64   0.654 0.6092  
  Treatment: Test and Test vs. Check  8  232.2   29.02   1.076 0.4779  
Residuals                             6  161.8   26.97                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value Pr(&gt;F)
Treatment (ignoring Blocks)    11  575.7   52.33   1.940  0.215
  Treatment: Check              3   52.9   17.64   0.654  0.609
  Treatment: Test               7  505.9   72.27   2.679  0.125
  Treatment: Test vs. Check     1   16.9   16.87   0.626  0.459
Block (eliminating Treatments)  2   69.5   34.75   1.288  0.342
Residuals                       6  161.8   26.97               

Coefficient of Variation
========================
6.372367

Overall Adjusted Mean
=====================
81.0625

Standard Errors
===============
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             4.240458 10.37603
Two Test Treatments (Same Block)                    7.344688 17.97180
Two Test Treatments (Different Blocks)              8.211611 20.09309
A Test Treatment and a Control Treatment            6.704752 16.40594

Treatment Means
===============
 Treatment Block Means   SE r   Min   Max Adjusted Means
         1       84.67 3.84 3 79.00 92.00          84.67
        10     3 74.00 &lt;NA&gt; 1 74.00 74.00          77.25
        11     1 89.00 &lt;NA&gt; 1 89.00 89.00          86.50
        12     1 82.00 &lt;NA&gt; 1 82.00 82.00          79.50
         2       79.00 1.15 3 77.00 81.00          79.00
         3       82.00 2.65 3 78.00 87.00          82.00
         4       83.33 3.93 3 78.00 91.00          83.33
         5     2 79.00 &lt;NA&gt; 1 79.00 79.00          78.25
         6     3 75.00 &lt;NA&gt; 1 75.00 75.00          78.25
         7     1 96.00 &lt;NA&gt; 1 96.00 96.00          93.50
         8     3 70.00 &lt;NA&gt; 1 70.00 70.00          73.25
         9     2 78.00 &lt;NA&gt; 1 78.00 78.00          77.25


Comparisons
===========

Method : lsd

                  contrast estimate   SE df t.ratio p.value sig
   treatment1 - treatment2     5.67 4.24  6   1.336   0.230    
   treatment1 - treatment3     2.67 4.24  6   0.629   0.553    
   treatment1 - treatment4     1.33 4.24  6   0.314   0.764    
   treatment1 - treatment5     6.42 6.36  6   1.009   0.352    
   treatment1 - treatment6     6.42 6.36  6   1.009   0.352    
   treatment1 - treatment7    -8.83 6.36  6  -1.389   0.214    
   treatment1 - treatment8    11.42 6.36  6   1.795   0.123    
   treatment1 - treatment9     7.42 6.36  6   1.166   0.288    
  treatment1 - treatment10     7.42 6.36  6   1.166   0.288    
  treatment1 - treatment11    -1.83 6.36  6  -0.288   0.783    
  treatment1 - treatment12     5.17 6.36  6   0.812   0.448    
   treatment2 - treatment3    -3.00 4.24  6  -0.707   0.506    
   treatment2 - treatment4    -4.33 4.24  6  -1.022   0.346    
   treatment2 - treatment5     0.75 6.36  6   0.118   0.910    
   treatment2 - treatment6     0.75 6.36  6   0.118   0.910    
   treatment2 - treatment7   -14.50 6.36  6  -2.280   0.063    
   treatment2 - treatment8     5.75 6.36  6   0.904   0.401    
   treatment2 - treatment9     1.75 6.36  6   0.275   0.792    
  treatment2 - treatment10     1.75 6.36  6   0.275   0.792    
  treatment2 - treatment11    -7.50 6.36  6  -1.179   0.283    
  treatment2 - treatment12    -0.50 6.36  6  -0.079   0.940    
   treatment3 - treatment4    -1.33 4.24  6  -0.314   0.764    
   treatment3 - treatment5     3.75 6.36  6   0.590   0.577    
   treatment3 - treatment6     3.75 6.36  6   0.590   0.577    
   treatment3 - treatment7   -11.50 6.36  6  -1.808   0.121    
   treatment3 - treatment8     8.75 6.36  6   1.376   0.218    
   treatment3 - treatment9     4.75 6.36  6   0.747   0.483    
  treatment3 - treatment10     4.75 6.36  6   0.747   0.483    
  treatment3 - treatment11    -4.50 6.36  6  -0.707   0.506    
  treatment3 - treatment12     2.50 6.36  6   0.393   0.708    
   treatment4 - treatment5     5.08 6.36  6   0.799   0.455    
   treatment4 - treatment6     5.08 6.36  6   0.799   0.455    
   treatment4 - treatment7   -10.17 6.36  6  -1.598   0.161    
   treatment4 - treatment8    10.08 6.36  6   1.585   0.164    
   treatment4 - treatment9     6.08 6.36  6   0.956   0.376    
  treatment4 - treatment10     6.08 6.36  6   0.956   0.376    
  treatment4 - treatment11    -3.17 6.36  6  -0.498   0.636    
  treatment4 - treatment12     3.83 6.36  6   0.603   0.569    
   treatment5 - treatment6     0.00 8.21  6   0.000   1.000    
   treatment5 - treatment7   -15.25 8.21  6  -1.857   0.113    
   treatment5 - treatment8     5.00 8.21  6   0.609   0.565    
   treatment5 - treatment9     1.00 7.34  6   0.136   0.896    
  treatment5 - treatment10     1.00 8.21  6   0.122   0.907    
  treatment5 - treatment11    -8.25 8.21  6  -1.005   0.354    
  treatment5 - treatment12    -1.25 8.21  6  -0.152   0.884    
   treatment6 - treatment7   -15.25 8.21  6  -1.857   0.113    
   treatment6 - treatment8     5.00 7.34  6   0.681   0.521    
   treatment6 - treatment9     1.00 8.21  6   0.122   0.907    
  treatment6 - treatment10     1.00 7.34  6   0.136   0.896    
  treatment6 - treatment11    -8.25 8.21  6  -1.005   0.354    
  treatment6 - treatment12    -1.25 8.21  6  -0.152   0.884    
   treatment7 - treatment8    20.25 8.21  6   2.466   0.049   *
   treatment7 - treatment9    16.25 8.21  6   1.979   0.095    
  treatment7 - treatment10    16.25 8.21  6   1.979   0.095    
  treatment7 - treatment11     7.00 7.34  6   0.953   0.377    
  treatment7 - treatment12    14.00 7.34  6   1.906   0.105    
   treatment8 - treatment9    -4.00 8.21  6  -0.487   0.643    
  treatment8 - treatment10    -4.00 7.34  6  -0.545   0.606    
  treatment8 - treatment11   -13.25 8.21  6  -1.614   0.158    
  treatment8 - treatment12    -6.25 8.21  6  -0.761   0.475    
  treatment9 - treatment10     0.00 8.21  6   0.000   1.000    
  treatment9 - treatment11    -9.25 8.21  6  -1.126   0.303    
  treatment9 - treatment12    -2.25 8.21  6  -0.274   0.793    
 treatment10 - treatment11    -9.25 8.21  6  -1.126   0.303    
 treatment10 - treatment12    -2.25 8.21  6  -0.274   0.793    
 treatment11 - treatment12     7.00 7.34  6   0.953   0.377    

Treatment Groups
================

Method : lsd

 Treatment Adjusted Means   SE df lower.CL upper.CL Group
         8          73.25 5.61  6    59.52    86.98    1 
         9          77.25 5.61  6    63.52    90.98    12
        10          77.25 5.61  6    63.52    90.98    12
         5          78.25 5.61  6    64.52    91.98    12
         6          78.25 5.61  6    64.52    91.98    12
         2          79.00 3.00  6    71.66    86.34    12
        12          79.50 5.61  6    65.77    93.23    12
         3          82.00 3.00  6    74.66    89.34    12
         4          83.33 3.00  6    76.00    90.67    12
         1          84.67 3.00  6    77.33    92.00    12
        11          86.50 5.61  6    72.77   100.23    12
         7          93.50 5.61  6    79.77   107.23     2</code></pre>
<div class="sourceCode" id="cb203"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Print results for variable y2</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">out2</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"


ANOVA, Treatment Adjusted
=========================
                                     Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Block (ignoring Treatments)           2   7019    3510  12.261 0.007597 ** 
Treatment (eliminating Blocks)       11  58965    5360  18.727 0.000920 ***
  Treatment: Check                    3   2150     717   2.504 0.156116    
  Treatment: Test and Test vs. Check  8  56815    7102  24.810 0.000473 ***
Residuals                             6   1718     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

ANOVA, Block Adjusted
=====================
                               Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Treatment (ignoring Blocks)    11  64708    5883  20.550 0.000707 ***
  Treatment: Check              3   2150     717   2.504 0.156116    
  Treatment: Test               7  34863    4980  17.399 0.001366 ** 
  Treatment: Test vs. Check     1  27694   27694  96.749 6.36e-05 ***
Block (eliminating Treatments)  2   1277     639   2.231 0.188645    
Residuals                       6   1717     286                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Coefficient of Variation
========================
6.057617

Overall Adjusted Mean
=====================
298.4792

Standard Errors
===============
                                         Std. Error of Diff.  CD (5%)
Control Treatment Means                             13.81424 33.80224
Two Test Treatments (Same Block)                    23.92697 58.54719
Two Test Treatments (Different Blocks)              26.75117 65.45775
A Test Treatment and a Control Treatment            21.84224 53.44603

Treatment Means
===============
 Treatment Block  Means    SE r    Min    Max Adjusted Means
         1       256.00  3.06 3 250.00 260.00         256.00
        10     3 450.00  &lt;NA&gt; 1 450.00 450.00         437.67
        11     1 300.00  &lt;NA&gt; 1 300.00 300.00         299.42
        12     1 289.00  &lt;NA&gt; 1 289.00 289.00         288.42
         2       228.00  6.11 3 220.00 240.00         228.00
         3       247.67 10.17 3 237.00 268.00         247.67
         4       264.00 18.68 3 227.00 287.00         264.00
         5     2 281.00  &lt;NA&gt; 1 281.00 281.00         293.92
         6     3 395.00  &lt;NA&gt; 1 395.00 395.00         382.67
         7     1 347.00  &lt;NA&gt; 1 347.00 347.00         346.42
         8     3 226.00  &lt;NA&gt; 1 226.00 226.00         213.67
         9     2 311.00  &lt;NA&gt; 1 311.00 311.00         323.92


Comparisons
===========

Method : lsd

                  contrast estimate    SE df t.ratio p.value sig
   treatment1 - treatment2    28.00 13.81  6   2.027   0.089    
   treatment1 - treatment3     8.33 13.81  6   0.603   0.568    
   treatment1 - treatment4    -8.00 13.81  6  -0.579   0.584    
   treatment1 - treatment5   -37.92 20.72  6  -1.830   0.117    
   treatment1 - treatment6  -126.67 20.72  6  -6.113   0.001 ***
   treatment1 - treatment7   -90.42 20.72  6  -4.363   0.005  **
   treatment1 - treatment8    42.33 20.72  6   2.043   0.087    
   treatment1 - treatment9   -67.92 20.72  6  -3.278   0.017   *
  treatment1 - treatment10  -181.67 20.72  6  -8.767   0.000 ***
  treatment1 - treatment11   -43.42 20.72  6  -2.095   0.081    
  treatment1 - treatment12   -32.42 20.72  6  -1.564   0.169    
   treatment2 - treatment3   -19.67 13.81  6  -1.424   0.204    
   treatment2 - treatment4   -36.00 13.81  6  -2.606   0.040   *
   treatment2 - treatment5   -65.92 20.72  6  -3.181   0.019   *
   treatment2 - treatment6  -154.67 20.72  6  -7.464   0.000 ***
   treatment2 - treatment7  -118.42 20.72  6  -5.715   0.001  **
   treatment2 - treatment8    14.33 20.72  6   0.692   0.515    
   treatment2 - treatment9   -95.92 20.72  6  -4.629   0.004  **
  treatment2 - treatment10  -209.67 20.72  6 -10.118   0.000 ***
  treatment2 - treatment11   -71.42 20.72  6  -3.447   0.014   *
  treatment2 - treatment12   -60.42 20.72  6  -2.916   0.027   *
   treatment3 - treatment4   -16.33 13.81  6  -1.182   0.282    
   treatment3 - treatment5   -46.25 20.72  6  -2.232   0.067    
   treatment3 - treatment6  -135.00 20.72  6  -6.515   0.001 ***
   treatment3 - treatment7   -98.75 20.72  6  -4.766   0.003  **
   treatment3 - treatment8    34.00 20.72  6   1.641   0.152    
   treatment3 - treatment9   -76.25 20.72  6  -3.680   0.010   *
  treatment3 - treatment10  -190.00 20.72  6  -9.169   0.000 ***
  treatment3 - treatment11   -51.75 20.72  6  -2.497   0.047   *
  treatment3 - treatment12   -40.75 20.72  6  -1.967   0.097    
   treatment4 - treatment5   -29.92 20.72  6  -1.444   0.199    
   treatment4 - treatment6  -118.67 20.72  6  -5.727   0.001  **
   treatment4 - treatment7   -82.42 20.72  6  -3.977   0.007  **
   treatment4 - treatment8    50.33 20.72  6   2.429   0.051    
   treatment4 - treatment9   -59.92 20.72  6  -2.892   0.028   *
  treatment4 - treatment10  -173.67 20.72  6  -8.381   0.000 ***
  treatment4 - treatment11   -35.42 20.72  6  -1.709   0.138    
  treatment4 - treatment12   -24.42 20.72  6  -1.178   0.283    
   treatment5 - treatment6   -88.75 26.75  6  -3.318   0.016   *
   treatment5 - treatment7   -52.50 26.75  6  -1.963   0.097    
   treatment5 - treatment8    80.25 26.75  6   3.000   0.024   *
   treatment5 - treatment9   -30.00 23.93  6  -1.254   0.257    
  treatment5 - treatment10  -143.75 26.75  6  -5.374   0.002  **
  treatment5 - treatment11    -5.50 26.75  6  -0.206   0.844    
  treatment5 - treatment12     5.50 26.75  6   0.206   0.844    
   treatment6 - treatment7    36.25 26.75  6   1.355   0.224    
   treatment6 - treatment8   169.00 23.93  6   7.063   0.000 ***
   treatment6 - treatment9    58.75 26.75  6   2.196   0.070    
  treatment6 - treatment10   -55.00 23.93  6  -2.299   0.061    
  treatment6 - treatment11    83.25 26.75  6   3.112   0.021   *
  treatment6 - treatment12    94.25 26.75  6   3.523   0.012   *
   treatment7 - treatment8   132.75 26.75  6   4.962   0.003  **
   treatment7 - treatment9    22.50 26.75  6   0.841   0.433    
  treatment7 - treatment10   -91.25 26.75  6  -3.411   0.014   *
  treatment7 - treatment11    47.00 23.93  6   1.964   0.097    
  treatment7 - treatment12    58.00 23.93  6   2.424   0.052    
   treatment8 - treatment9  -110.25 26.75  6  -4.121   0.006  **
  treatment8 - treatment10  -224.00 23.93  6  -9.362   0.000 ***
  treatment8 - treatment11   -85.75 26.75  6  -3.205   0.018   *
  treatment8 - treatment12   -74.75 26.75  6  -2.794   0.031   *
  treatment9 - treatment10  -113.75 26.75  6  -4.252   0.005  **
  treatment9 - treatment11    24.50 26.75  6   0.916   0.395    
  treatment9 - treatment12    35.50 26.75  6   1.327   0.233    
 treatment10 - treatment11   138.25 26.75  6   5.168   0.002  **
 treatment10 - treatment12   149.25 26.75  6   5.579   0.001  **
 treatment11 - treatment12    11.00 23.93  6   0.460   0.662    

Treatment Groups
================

Method : lsd

 Treatment Adjusted Means    SE df lower.CL upper.CL    Group
         8         213.67 18.27  6   168.95   258.38  12     
         2         228.00  9.77  6   204.10   251.90  1      
         3         247.67  9.77  6   223.76   271.57  123    
         1         256.00  9.77  6   232.10   279.90  1234   
         4         264.00  9.77  6   240.10   287.90   234   
        12         288.42 18.27  6   243.70   333.13    345  
         5         293.92 18.27  6   249.20   338.63    345  
        11         299.42 18.27  6   254.70   344.13     45  
         9         323.92 18.27  6   279.20   368.63      56 
         7         346.42 18.27  6   301.70   391.13      56 
         6         382.67 18.27  6   337.95   427.38       67
        10         437.67 18.27  6   392.95   482.38        7</code></pre>
</div>
<div class="section level3">
<h3 id="describe-augmentedrcbd">7.3 <code>describe.augmentedRCBD()</code><a class="anchor" aria-label="anchor" href="#describe-augmentedrcbd"></a>
</h3>
<p>The descriptive statistics such as count, mean, standard error,
minimum, maximum, skewness ( with p-value from D’Agostino test of
skewness (<span class="citation">D’Agostino (<a href="#ref-dagostino_transformation_1970" role="doc-biblioref">1970</a>)</span>)) and kurtosis (with p-value from
Anscombe-Glynn test of kurtosis (<span class="citation">Anscombe and
Glynn (<a href="#ref-anscombe_distribution_1983" role="doc-biblioref">1983</a>)</span>)) for the adjusted means from the
results in an object of class <code>augmentedRCBD</code> can be computed
as follows.</p>
<div class="sourceCode" id="cb205"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Descriptive statistics for variable y1</span></span>
<span><span class="fu"><a href="../reference/describe.augmentedRCBD.html">describe.augmentedRCBD</a></span><span class="op">(</span><span class="va">out1</span><span class="op">)</span></span></code></pre></div>
<pre><code>$Count
[1] 12

$Mean
[1] 81.0625

$Std.Error
[1] 1.547002

$Std.Deviation
[1] 5.358973

$Min
[1] 73.25

$Max
[1] 93.5

$`Skewness(statistic)`
     skew         z 
0.9250344 1.6745760 

$`Skewness(p.value)`
[1] 0.09401746

$`Kurtosis(statistic)`
    kurt        z 
3.522807 1.282305 

$`Kurtosis(p.value)`
[1] 0.1997357</code></pre>
<div class="sourceCode" id="cb207"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Descriptive statistics for variable y2</span></span>
<span><span class="fu"><a href="../reference/describe.augmentedRCBD.html">describe.augmentedRCBD</a></span><span class="op">(</span><span class="va">out2</span><span class="op">)</span></span></code></pre></div>
<pre><code>$Count
[1] 12

$Mean
[1] 298.4792

$Std.Error
[1] 18.92257

$Std.Deviation
[1] 65.5497

$Min
[1] 213.6667

$Max
[1] 437.6667

$`Skewness(statistic)`
     skew         z 
0.7449405 1.3680211 

$`Skewness(p.value)`
[1] 0.1713055

$`Kurtosis(statistic)`
    kurt        z 
2.787997 0.536812 

$`Kurtosis(p.value)`
[1] 0.5913975</code></pre>
</div>
<div class="section level3">
<h3 id="freqdist-augmentedrcbd">7.4 <code>freqdist.augmentedRCBD()</code><a class="anchor" aria-label="anchor" href="#freqdist-augmentedrcbd"></a>
</h3>
<p>The frequency distribution of the adjusted means from the results in
an object of class <code>augmentedRCBD</code> can be plotted as
follows.</p>
<div class="sourceCode" id="cb209"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Frequency distribution for variable y1</span></span>
<span><span class="va">freq1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out1</span>, xlab <span class="op">=</span> <span class="st">"Trait 1"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Warning:  [1m [22mRemoved 2 rows containing missing values (`geom_bar()`).</code></pre>
<div class="sourceCode" id="cb211"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">freq1</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-70-1.png" width="700"></p>
<div class="sourceCode" id="cb212"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Frequency distribution for variable y2</span></span>
<span><span class="va">freq2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out2</span>, xlab <span class="op">=</span> <span class="st">"Trait 2"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Warning:  [1m [22mRemoved 2 rows containing missing values (`geom_bar()`).</code></pre>
<div class="sourceCode" id="cb214"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">freq2</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-70-2.png" width="700"></p>
<p>The colours for the check values may be specified using the argument
<code>check.col</code>.</p>
<div class="sourceCode" id="cb215"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">colset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red3"</span>, <span class="st">"green4"</span>, <span class="st">"purple3"</span>, <span class="st">"darkorange3"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Frequency distribution for variable y1</span></span>
<span><span class="va">freq1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out1</span>, xlab <span class="op">=</span> <span class="st">"Trait 1"</span>, check.col <span class="op">=</span> <span class="va">colset</span><span class="op">)</span></span></code></pre></div>
<pre><code>Warning:  [1m [22mRemoved 2 rows containing missing values (`geom_bar()`).</code></pre>
<div class="sourceCode" id="cb217"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">freq1</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-71-1.png" width="700"></p>
<div class="sourceCode" id="cb218"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Frequency distribution for variable y2</span></span>
<span><span class="va">freq2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out2</span>, xlab <span class="op">=</span> <span class="st">"Trait 2"</span>, check.col <span class="op">=</span> <span class="va">colset</span><span class="op">)</span></span></code></pre></div>
<pre><code>Warning:  [1m [22mRemoved 2 rows containing missing values (`geom_bar()`).</code></pre>
<div class="sourceCode" id="cb220"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">freq2</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-71-2.png" width="700"></p>
<p>The default the check highlighting can be avoided using the argument
<code>highlight.check = FALSE</code>.</p>
<div class="sourceCode" id="cb221"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Frequency distribution for variable y1</span></span>
<span><span class="va">freq1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out1</span>, xlab <span class="op">=</span> <span class="st">"Trait 1"</span>,</span>
<span>                                highlight.check <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code>Warning:  [1m [22mRemoved 2 rows containing missing values (`geom_bar()`).</code></pre>
<div class="sourceCode" id="cb223"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">freq1</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-72-1.png" width="700"></p>
<div class="sourceCode" id="cb224"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Frequency distribution for variable y2</span></span>
<span><span class="va">freq2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/freqdist.augmentedRCBD.html">freqdist.augmentedRCBD</a></span><span class="op">(</span><span class="va">out2</span>, xlab <span class="op">=</span> <span class="st">"Trait 2"</span>,</span>
<span>                                highlight.check <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code>Warning:  [1m [22mRemoved 2 rows containing missing values (`geom_bar()`).</code></pre>
<div class="sourceCode" id="cb226"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">freq2</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-72-2.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="gva-augmentedrcbd">7.5 <code>gva.augmentedRCBD()</code><a class="anchor" aria-label="anchor" href="#gva-augmentedrcbd"></a>
</h3>
<p>The genetic variability statistics such as mean, phenotypic,
genotypic and environmental variation (<span class="citation">Federer
and Searle (<a href="#ref-federerModelConsiderationsVariance1976" role="doc-biblioref">1976</a>)</span>), phenotypic, genotypic and
environmental coefficient of variation (<span class="citation">Burton
(<a href="#ref-burton_quantitative_1951" role="doc-biblioref">1951</a>)</span>, <span class="citation">Burton (<a href="#ref-burton_qualitative_1952" role="doc-biblioref">1952</a>)</span>), category of phenotypic and
genotypic coefficient of variation according to <span class="citation">Sivasubramaniam and Madhavamenon (<a href="#ref-sivasubramaniam_genotypic_1973" role="doc-biblioref">1973</a>)</span>, broad-sense heritability
(<em>H<sup>2</sup></em>) (<span class="citation">Lush (<a href="#ref-lush_intra-sire_1940" role="doc-biblioref">1940</a>)</span>),
<em>H<sup>2</sup></em> category according to <span class="citation">Robinson (<a href="#ref-robinson_quantitative_1966" role="doc-biblioref">1966</a>)</span>, Genetic advance (GA), genetic
advance as per cent of mean (GAM) and GAM category according to <span class="citation">Johnson et al. (<a href="#ref-johnson_estimates_1955" role="doc-biblioref">1955</a>)</span> are computed from an object of
class <code>augmentedRCBD</code> as follows. Genetic variability
analysis needs to be performed only if the sum of squares of “Treatment:
Test” are significant.</p>
<div class="sourceCode" id="cb227"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Genetic variability statistics for variable y1</span></span>
<span><span class="fu"><a href="../reference/gva.augmentedRCBD.html">gva.augmentedRCBD</a></span><span class="op">(</span><span class="va">out1</span><span class="op">)</span></span></code></pre></div>
<pre><code>Warning in gva.augmentedRCBD(out1): P-value for "Treatment: Test" is &gt; 0.05.
Genetic variability analysis may not be appropriate for this trait.</code></pre>
<pre><code>$Mean
[1] 81.0625

$PV
[1] 72.26786

$GV
[1] 45.29563

$EV
[1] 26.97222

$GCV
[1] 8.302487

$`GCV category`
[1] "Low"

$PCV
[1] 10.48703

$`PCV category`
[1] "Medium"

$ECV
[1] 6.406759

$hBS
[1] 62.67743

$`hBS category`
[1] "High"

$GA
[1] 10.99216

$GAM
[1] 13.5601

$`GAM category`
[1] "Medium"</code></pre>
<div class="sourceCode" id="cb230"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Genetic variability statistics for variable y2</span></span>
<span><span class="fu"><a href="../reference/gva.augmentedRCBD.html">gva.augmentedRCBD</a></span><span class="op">(</span><span class="va">out2</span><span class="op">)</span></span></code></pre></div>
<pre><code>$Mean
[1] 298.4792

$PV
[1] 4980.411

$GV
[1] 4694.161

$EV
[1] 286.25

$GCV
[1] 22.95435

$`GCV category`
[1] "High"

$PCV
[1] 23.64387

$`PCV category`
[1] "High"

$ECV
[1] 5.668377

$hBS
[1] 94.25248

$`hBS category`
[1] "High"

$GA
[1] 137.2223

$GAM
[1] 45.97382

$`GAM category`
[1] "High"</code></pre>
<p>Negative estimates of variance components if computed are not
abnormal. For information on how to deal with these, refer <span class="citation">Robinson et al. (<a href="#ref-robinson_genetic_1955" role="doc-biblioref">1955</a>)</span> and <span class="citation">Dudley
and Moll (<a href="#ref-dudley_interpretation_1969" role="doc-biblioref">1969</a>)</span>.</p>
</div>
<div class="section level3">
<h3 id="report-augmentedrcbd">7.5 <code>report.augmentedRCBD()</code><a class="anchor" aria-label="anchor" href="#report-augmentedrcbd"></a>
</h3>
<p>The results generated by the analysis can be exported to a MS Word
file as follows.</p>
<div class="sourceCode" id="cb232"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># MS word report for variable y1</span></span>
<span><span class="fu"><a href="../reference/report.augmentedRCBD.html">report.augmentedRCBD</a></span><span class="op">(</span>aug <span class="op">=</span> <span class="va">out1</span>,</span>
<span>                     target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                                        <span class="st">"augmentedRCBD output.docx"</span><span class="op">)</span>,</span>
<span>                     file.type <span class="op">=</span> <span class="st">"word"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># MS word report for variable y2</span></span>
<span><span class="fu"><a href="../reference/report.augmentedRCBD.html">report.augmentedRCBD</a></span><span class="op">(</span>aug <span class="op">=</span> <span class="va">out2</span>,</span>
<span>                     target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                                        <span class="st">"augmentedRCBD output.docx"</span><span class="op">)</span>,</span>
<span>                     file.type <span class="op">=</span> <span class="st">"word"</span><span class="op">)</span></span></code></pre></div>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/augRCBDword.png" align="center" alt="The `R` download location."></p>
<p style="text-align: left;">
<strong>Fig. 6</strong>: MS Word report generated with
<code>report.agumentedRCBD</code> function.
</p>
<p>Alternatively, the analysis results can also be exported to a MS
Excel file as follows.</p>
<div class="sourceCode" id="cb233"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># MS excel report for variable y1</span></span>
<span><span class="fu"><a href="../reference/report.augmentedRCBD.html">report.augmentedRCBD</a></span><span class="op">(</span>aug <span class="op">=</span> <span class="va">out1</span>,</span>
<span>                     target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                                        <span class="st">"augmentedRCBD output.xlsx"</span><span class="op">)</span>,</span>
<span>                     file.type <span class="op">=</span> <span class="st">"excel"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># MS excel report for variable y2</span></span>
<span><span class="fu"><a href="../reference/report.augmentedRCBD.html">report.augmentedRCBD</a></span><span class="op">(</span>aug <span class="op">=</span> <span class="va">out2</span>,</span>
<span>                     target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                                        <span class="st">"augmentedRCBD output.xlsx"</span><span class="op">)</span>,</span>
<span>                     file.type <span class="op">=</span> <span class="st">"excel"</span><span class="op">)</span></span></code></pre></div>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/augRCBDexcel.png" align="center" alt="The `R` download location."></p>
<p style="text-align: left;">
<strong>Fig. 6</strong>: MS Excel report generated with
<code>report.agumentedRCBD</code> function.
</p>
</div>
</div>
<div class="section level2">
<h2 id="data-analysis-for-a-multiple-traits">8 Data Analysis for a Multiple Traits<a class="anchor" aria-label="anchor" href="#data-analysis-for-a-multiple-traits"></a>
</h2>
<p>Analysis of data for a multiple traits simultaneously can be
performed by using <code>augmentedRCBD.bulk</code> function. It
generates an object of class <code>augmentedRCBD.bulk</code>. Such an
object can then be taken as input by
<code>print.augmentedRCBD.bulk</code> to print the results to console.
The results can also be exported as a MS Word report using the
<code>report.augmentedRCBD.bulk</code> function.</p>
<p><img src="Data_Analysis_with_augmentedRCBD_files/figure-html/unnamed-chunk-80-1.png" width="700"></p>
<p><strong>Fig. 8</strong>. Workflow for analysis of multiple traits
with <code>augmentedRCBD</code>.</p>
<div class="section level3">
<h3 id="augmentedrcbd-bulk">8.1 <code>augmentedRCBD.bulk()</code><a class="anchor" aria-label="anchor" href="#augmentedrcbd-bulk"></a>
</h3>
<p>Consider the data frame <code>data</code> imported from <a href="#impexp">Table 1</a> according to the instructions in <a href="#impexp">section 4.8</a>.</p>
<div class="sourceCode" id="cb234"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span></code></pre></div>
<pre><code>'data.frame':   20 obs. of  4 variables:
 $ blk: Factor w/ 3 levels "1","2","3": 1 1 1 1 1 1 1 2 2 2 ...
 $ trt: Factor w/ 12 levels "1","2","3","4",..: 1 2 3 4 7 11 12 1 2 3 ...
 $ y1 : num  92 79 87 81 96 89 82 79 81 81 ...
 $ y2 : num  258 224 238 278 347 300 289 260 220 237 ...</code></pre>
<div class="sourceCode" id="cb236"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert block and treatment to factors</span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">blk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">blk</span><span class="op">)</span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">trt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">trt</span><span class="op">)</span></span></code></pre></div>
<p>Rather than performing the analysis individually for each
variable/trait separately using <code>augmentedRCBD</code>, the analysis
can be performed simultaneously for for both the traits using
<code>augmentedRCBD.bulk</code> function. It is a wrapper around the
<code>augmentedRCBD</code> core function and its associated helper
functions.</p>
<p>However in this case treatment comparisons/grouping by least
significant difference or Tukey’s honest significant difference method
is not computed. Also the output object size is reduced using the
<code>simplify = TRUE</code> argument in the <code>augmentedRCBD</code>
function.</p>
<p>The logical arguments <code>describe</code>, <code>freqdist</code>
and <code>gva</code> can be used to specify whether to generate the
descriptive statistics, frequency distribution plots and genetic
variability statistics respectively. If <code>gva = TRUE</code>, then
plots to compare phenotypic and genotypic coefficient of variation,
broad sense heritability and genetic advance over mean between traits
are also generated.</p>
<div class="sourceCode" id="cb237"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bout</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/augmentedRCBD.bulk.html">augmentedRCBD.bulk</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, block <span class="op">=</span> <span class="st">"blk"</span>,</span>
<span>                           treatment <span class="op">=</span> <span class="st">"trt"</span>, traits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"y1"</span>, <span class="st">"y2"</span><span class="op">)</span>,</span>
<span>                           checks <span class="op">=</span> <span class="cn">NULL</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, describe <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                           freqdist <span class="op">=</span> <span class="cn">TRUE</span>, gva <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                           check.col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"brown"</span>, <span class="st">"darkcyan"</span>,</span>
<span>                                         <span class="st">"forestgreen"</span>, <span class="st">"purple"</span><span class="op">)</span>,</span>
<span>                           console <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>
ANOVA for y1 computed (1/2)

ANOVA for y2 computed (2/2)

Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"
Number of traits           "2"         
Traits                     "y1, y2"    


ANOVA, Treatment Adjusted
=========================
                                          Mean.Sq
                             Source Df     y1         y2   
        Block (ignoring Treatments)  2 180.04  * 3509.67 **
     Treatment (eliminating Blocks) 11  25.92 ⁿˢ 5360.49 **
                   Treatment: Check  3  17.64 ⁿˢ  716.75 ⁿˢ
 Treatment: Test and Test vs. Check  8  29.02 ⁿˢ 7101.89 **
                          Residuals  6  26.97     286.25   
ⁿˢ P &gt; 0.05; * P &lt;= 0.05; ** P &lt;= 0.01

ANOVA, Block Adjusted
=====================
                                      Mean.Sq
                         Source Df    y1          y2   
    Treatment (ignoring Blocks) 11 52.33 ⁿˢ  5882.50 **
               Treatment: Check  3 17.64 ⁿˢ   716.75 ⁿˢ
      Treatment: Test vs. Check  1 16.87 ⁿˢ 27694.41 **
                Treatment: Test  7 72.27 ⁿˢ  4980.41 **
 Block (eliminating Treatments)  2 34.75 ⁿˢ   638.58 ⁿˢ
                      Residuals  6 26.97      286.25   
ⁿˢ P &gt; 0.05; * P &lt;= 0.05; ** P &lt;= 0.01

Coefficient of Variation
========================
 Trait   CV
    y1 6.37
    y2 6.06


Overall Adjusted Mean
=====================
 Trait Overall.adjusted.mean
    y1                 81.06
    y2                298.48


Standard Errors
===============
                               Comparison   y1    y2
 A Test Treatment and a Control Treatment 6.70 21.84
                  Control Treatment Means 4.24 13.81
   Two Test Treatments (Different Blocks) 8.21 26.75
         Two Test Treatments (Same Block) 7.34 23.93


Critical Difference
===================
alpha = 0.05 
                               Comparison    y1    y2
 A Test Treatment and a Control Treatment 16.41 53.45
                  Control Treatment Means 10.38 33.80
   Two Test Treatments (Different Blocks) 20.09 65.46
         Two Test Treatments (Same Block) 17.97 58.55


Descriptive Statistics
======================
 Trait Count   Mean Std.Error Std.Deviation    Min    Max Skewness Skewness_sig
    y1    12  81.06      1.55          5.36  73.25  93.50     0.93           ⁿˢ
    y2    12 298.48     18.92         65.55 213.67 437.67     0.74           ⁿˢ
 Kurtosis Kurtosis_sig
     3.52           ⁿˢ
     2.79           ⁿˢ
ⁿˢ P &gt; 0.05; * P &lt;= 0.05; ** P &lt;= 0.01


Genetic Variability Analysis
============================
k = 2.063 
 Trait   Mean      PV      GV     EV   GCV GCV.category   PCV PCV.category  ECV
  y1 †  81.06   72.27   45.30  26.97  8.30          Low 10.49       Medium 6.41
  y2   298.48 4980.41 4694.16 286.25 22.95         High 23.64         High 5.67
   hBS hBS.category     GA   GAM GAM.category
 62.68         High  10.99 13.56       Medium
 94.25         High 137.22 45.97         High</code></pre>
<pre><code>Warning: 
† P-value for "Treatment: Test" is &gt; 0.05. Genetic variability analysis may not be appropriate for this trait.</code></pre>
<pre><code>Warning: 
‡ Negative GV detected.
 GCV, GCV category, hBS, hBS category, GA, GAM and
 GAM category could not be computed.</code></pre>
<pre><code>

Warning Messages
================


[Frequency Distribution]
&lt;y1&gt;
Removed 2 rows containing missing values (`geom_bar()`).
&lt;y2&gt;
Removed 2 rows containing missing values (`geom_bar()`).


[GVA]
&lt;y1&gt;
P-value for "Treatment: Test" is &gt; 0.05. Genetic variability analysis may not be appropriate for this trait.

Treatment Means
===============
 Treatment Block    y1     y2
         1       84.67 256.00
        10     3 77.25 437.67
        11     1 86.50 299.42
        12     1 79.50 288.42
         2       79.00 228.00
         3       82.00 247.67
         4       83.33 264.00
         5     2 78.25 293.92
         6     3 78.25 382.67
         7     1 93.50 346.42
         8     3 73.25 213.67
         9     2 77.25 323.92</code></pre>
</div>
<div class="section level3">
<h3 id="print-augmentedrcbd-bulk">8.2 <code>print.augmentedRCBD.bulk()</code><a class="anchor" aria-label="anchor" href="#print-augmentedrcbd-bulk"></a>
</h3>
<p>The results of analysis in an object of class
<code>augmentedRCBD.bulk</code> can be printed to the console as
follows.</p>
<div class="sourceCode" id="cb242"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Print results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">bout</span><span class="op">)</span></span></code></pre></div>
<pre><code>
Augmented Design Details
========================
                                       
Number of blocks           "3"         
Number of treatments       "12"        
Number of check treatments "4"         
Number of test treatments  "8"         
Check treatments           "1, 2, 3, 4"
Number of traits           "2"         
Traits                     "y1, y2"    


ANOVA, Treatment Adjusted
=========================
                                          Mean.Sq
                             Source Df     y1         y2   
        Block (ignoring Treatments)  2 180.04  * 3509.67 **
     Treatment (eliminating Blocks) 11  25.92 ⁿˢ 5360.49 **
                   Treatment: Check  3  17.64 ⁿˢ  716.75 ⁿˢ
 Treatment: Test and Test vs. Check  8  29.02 ⁿˢ 7101.89 **
                          Residuals  6  26.97     286.25   
ⁿˢ P &gt; 0.05; * P &lt;= 0.05; ** P &lt;= 0.01

ANOVA, Block Adjusted
=====================
                                      Mean.Sq
                         Source Df    y1          y2   
    Treatment (ignoring Blocks) 11 52.33 ⁿˢ  5882.50 **
               Treatment: Check  3 17.64 ⁿˢ   716.75 ⁿˢ
      Treatment: Test vs. Check  1 16.87 ⁿˢ 27694.41 **
                Treatment: Test  7 72.27 ⁿˢ  4980.41 **
 Block (eliminating Treatments)  2 34.75 ⁿˢ   638.58 ⁿˢ
                      Residuals  6 26.97      286.25   
ⁿˢ P &gt; 0.05; * P &lt;= 0.05; ** P &lt;= 0.01

Coefficient of Variation
========================
 Trait   CV
    y1 6.37
    y2 6.06


Overall Adjusted Mean
=====================
 Trait Overall.adjusted.mean
    y1                 81.06
    y2                298.48


Standard Errors
===============
                               Comparison   y1    y2
 A Test Treatment and a Control Treatment 6.70 21.84
                  Control Treatment Means 4.24 13.81
   Two Test Treatments (Different Blocks) 8.21 26.75
         Two Test Treatments (Same Block) 7.34 23.93


Critical Difference
===================
alpha = 0.05 
                               Comparison    y1    y2
 A Test Treatment and a Control Treatment 16.41 53.45
                  Control Treatment Means 10.38 33.80
   Two Test Treatments (Different Blocks) 20.09 65.46
         Two Test Treatments (Same Block) 17.97 58.55


Descriptive Statistics
======================
 Trait Count   Mean Std.Error Std.Deviation    Min    Max Skewness Skewness_sig
    y1    12  81.06      1.55          5.36  73.25  93.50     0.93           ⁿˢ
    y2    12 298.48     18.92         65.55 213.67 437.67     0.74           ⁿˢ
 Kurtosis Kurtosis_sig
     3.52           ⁿˢ
     2.79           ⁿˢ
ⁿˢ P &gt; 0.05; * P &lt;= 0.05; ** P &lt;= 0.01


Genetic Variability Analysis
============================
k = 2.063 
 Trait   Mean      PV      GV     EV   GCV GCV.category   PCV PCV.category  ECV
  y1 †  81.06   72.27   45.30  26.97  8.30          Low 10.49       Medium 6.41
  y2   298.48 4980.41 4694.16 286.25 22.95         High 23.64         High 5.67
   hBS hBS.category     GA   GAM GAM.category
 62.68         High  10.99 13.56       Medium
 94.25         High 137.22 45.97         High</code></pre>
<pre><code>Warning: 
† P-value for "Treatment: Test" is &gt; 0.05. Genetic variability analysis may not be appropriate for this trait.</code></pre>
<pre><code>Warning: 
‡ Negative GV detected.
 GCV, GCV category, hBS, hBS category, GA, GAM and
 GAM category could not be computed.</code></pre>
<pre><code>

Warning Messages
================


[Frequency Distribution]
&lt;y1&gt;
Removed 2 rows containing missing values (`geom_bar()`).
&lt;y2&gt;
Removed 2 rows containing missing values (`geom_bar()`).


[GVA]
&lt;y1&gt;
P-value for "Treatment: Test" is &gt; 0.05. Genetic variability analysis may not be appropriate for this trait.

Treatment Means
===============
 Treatment Block    y1     y2
         1       84.67 256.00
        10     3 77.25 437.67
        11     1 86.50 299.42
        12     1 79.50 288.42
         2       79.00 228.00
         3       82.00 247.67
         4       83.33 264.00
         5     2 78.25 293.92
         6     3 78.25 382.67
         7     1 93.50 346.42
         8     3 73.25 213.67
         9     2 77.25 323.92</code></pre>
</div>
<div class="section level3">
<h3 id="report-augmentedrcbd-bulk">8.3 <code>report.augmentedRCBD.bulk()</code><a class="anchor" aria-label="anchor" href="#report-augmentedrcbd-bulk"></a>
</h3>
<p>The results generated by the analysis can be exported to a MS Word
file as follows.</p>
<div class="sourceCode" id="cb247"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># MS word report</span></span>
<span><span class="fu"><a href="../reference/report.augmentedRCBD.bulk.html">report.augmentedRCBD.bulk</a></span><span class="op">(</span>aug.bulk <span class="op">=</span> <span class="va">bout</span>,</span>
<span>                          target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                                             <span class="st">"augmentedRCBD bulk output.docx"</span><span class="op">)</span>,</span>
<span>                          file.type <span class="op">=</span> <span class="st">"word"</span><span class="op">)</span></span></code></pre></div>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/augRCBDbulkword.png" align="center" alt="The `R` download location."></p>
<p style="text-align: left;">
<strong>Fig. 9</strong>: MS Word report generated with
<code>report.agumentedRCBD.bulk</code> function.
</p>
<p>Alternatively, the analysis results can also be exported to a MS
Excel file as follows.</p>
<div class="sourceCode" id="cb248"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># MS excel report</span></span>
<span><span class="fu"><a href="../reference/report.augmentedRCBD.bulk.html">report.augmentedRCBD.bulk</a></span><span class="op">(</span>aug.bulk <span class="op">=</span> <span class="va">bout</span>,</span>
<span>                          target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                                             <span class="st">"augmentedRCBD bulk output.xlsx"</span><span class="op">)</span>,</span>
<span>                          file.type <span class="op">=</span> <span class="st">"excel"</span><span class="op">)</span></span></code></pre></div>
<p><img src="https://github.com/aravind-j/augmentedRCBD/raw/master/vignettes/augRCBDbulkexcel.png" align="center" alt="The `R` download location."></p>
<p style="text-align: left;">
<strong>Fig. 10</strong>: MS Excel report generated with
<code>report.agumentedRCBD.bulk</code> function.
</p>
</div>
</div>
<div class="section level2">
<h2 id="citing-augmentedrcbd">9 Citing <code>augmentedRCBD</code><a class="anchor" aria-label="anchor" href="#citing-augmentedrcbd"></a>
</h2>
<pre><code>To cite the R package 'augmentedRCBD' in publications use:

  Aravind, J., Mukesh Sankar, S., Wankhede, D. P., and Kaur, V. (2023).
  augmentedRCBD: Analysis of Augmented Randomised Complete Block
  Designs. R package version 0.1.5.9000,
  https://aravind-j.github.io/augmentedRCBD/https://cran.r-project.org/package=augmentedRCBD.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {augmentedRCBD: Analysis of Augmented Randomised Complete Block Designs},
    author = {J. Aravind and S. {Mukesh Sankar} and Dhammaprakash Pandhari Wankhede and Vikender Kaur},
    year = {2023},
    note = {R package version 0.1.5.9000},
    note = {https://aravind-j.github.io/augmentedRCBD/},
    note = {https://cran.r-project.org/package=augmentedRCBD},
  }

This free and open-source software implements academic research by the
authors and co-workers. If you use it, please support the project by
citing the package.</code></pre>
</div>
<div class="section level2">
<h2 id="session-info">10 Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb250"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.6.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: UTC
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] diagram_1.6.5            shape_1.4.6              augmentedRCBD_0.1.5.9000

loaded via a namespace (and not attached):
  [1] tidyselect_1.2.0        farver_2.1.1            dplyr_1.1.2            
  [4] bitops_1.0-7            RCurl_1.98-1.12         fastmap_1.1.1          
  [7] TH.data_1.1-2           fontquiver_0.2.1        mathjaxr_1.6-0         
 [10] promises_1.2.0.1        XML_3.99-0.14           digest_0.6.31          
 [13] estimability_1.4.1      mime_0.12               lifecycle_1.0.3        
 [16] multcompView_0.1-9      ellipsis_0.3.2          gfonts_0.2.0           
 [19] survival_3.5-5          magrittr_2.0.3          compiler_4.3.0         
 [22] rlang_1.1.1             sass_0.4.6              tools_4.3.0            
 [25] utf8_1.2.3              yaml_2.3.7              data.table_1.14.8      
 [28] knitr_1.42              labeling_0.4.2          askpass_1.1            
 [31] curl_5.0.0              plyr_1.8.8              xml2_1.3.4             
 [34] multcomp_1.4-23         numform_0.7.0           httpcode_0.3.0         
 [37] withr_2.5.0             purrr_1.0.1             desc_1.4.2             
 [40] grid_4.3.0              fansi_1.0.4             gdtools_0.3.3          
 [43] xtable_1.8-4            colorspace_2.1-0        ggplot2_3.4.2          
 [46] emmeans_1.8.6           scales_1.2.1            MASS_7.3-58.4          
 [49] crul_1.3                cli_3.6.1               mvtnorm_1.1-3          
 [52] rmarkdown_2.21          crayon_1.5.2            ragg_1.2.5             
 [55] generics_0.1.3          httr_1.4.6              reshape2_1.4.4         
 [58] cachem_1.0.8            stringr_1.5.0           splines_4.3.0          
 [61] vctrs_0.6.2             Matrix_1.5-4            sandwich_3.0-2         
 [64] jsonlite_1.8.4          fontBitstreamVera_0.1.1 systemfonts_1.0.4      
 [67] jquerylib_0.1.4         glue_1.6.2              pkgdown_2.0.7          
 [70] codetools_0.2-19        stringi_1.7.12          flextable_0.9.1        
 [73] gtable_0.3.3            later_1.3.1             munsell_0.5.0          
 [76] tibble_3.2.1            pillar_1.9.0            htmltools_0.5.5        
 [79] openssl_2.0.6           R6_2.5.1                textshaping_0.3.6      
 [82] Rdpack_2.4              rprojroot_2.0.3         evaluate_0.21          
 [85] shiny_1.7.4             lattice_0.21-8          highr_0.10             
 [88] rbibutils_2.2.13        moments_0.14.1          openxlsx_4.2.5.2       
 [91] memoise_2.0.1           fontLiberation_0.1.0    httpuv_1.6.11          
 [94] bslib_0.4.2             Rcpp_1.0.10             zip_2.3.0              
 [97] uuid_1.1-0              officer_0.6.2           xfun_0.39              
[100] fs_1.6.2                zoo_1.8-12              pkgconfig_2.0.3        </code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-anscombe_distribution_1983" class="csl-entry">
Anscombe, F. J., and Glynn, W. J. (1983). Distribution of the kurtosis
statistic <em>b</em><span class="math inline">\(_{{2}}\)</span> for
normal samples. <em>Biometrika</em> 70, 227–234. doi:<a href="https://doi.org/10.1093/biomet/70.1.227" class="external-link">10.1093/biomet/70.1.227</a>.
</div>
<div id="ref-burton_quantitative_1951" class="csl-entry">
Burton, G. W. (1951). Quantitative inheritance in pearl millet
(<em><span>Pennisetum</span> glaucum</em>). <em>Agronomy Journal</em>
43, 409–417. doi:<a href="https://doi.org/10.2134/agronj1951.00021962004300090001x" class="external-link">10.2134/agronj1951.00021962004300090001x</a>.
</div>
<div id="ref-burton_qualitative_1952" class="csl-entry">
Burton, G. W. (1952). Qualitative inheritance in grasses.
<span>Vol</span>. 1. in <em>Proceedings of the 6th
<span>International</span> <span>Grassland</span> <span>Congress</span>,
<span>Pennsylvania</span> <span>State</span> <span>College</span></em>,
17–23.
</div>
<div id="ref-dagostino_transformation_1970" class="csl-entry">
D’Agostino, R. B. (1970). Transformation to normality of the null
distribution of <em>g</em><span class="math inline">\(_{{1}}\)</span>.
<em>Biometrika</em> 57, 679–681. doi:<a href="https://doi.org/10.1093/biomet/57.3.679" class="external-link">10.1093/biomet/57.3.679</a>.
</div>
<div id="ref-dudley_interpretation_1969" class="csl-entry">
Dudley, J. W., and Moll, R. H. (1969). Interpretation and use of
estimates of heritability and genetic variances in plant breeding.
<em>Crop Science</em> 9, 257–262. doi:<a href="https://doi.org/10.2135/cropsci1969.0011183X000900030001x" class="external-link">10.2135/cropsci1969.0011183X000900030001x</a>.
</div>
<div id="ref-federer_augmented_1956-1" class="csl-entry">
Federer, W. T. (1956b). Augmented (or <span>Hoonuiaku</span>)
<span>Designs</span>. New York: Cornell University Available at: <a href="https://ecommons.cornell.edu/handle/1813/32841" class="external-link">https://ecommons.cornell.edu/handle/1813/32841</a>.
</div>
<div id="ref-federer_augmented_1956" class="csl-entry">
Federer, W. T. (1956a). Augmented (or <span>Hoonuiaku</span>) designs.
<em>The Hawaiian Planters’ Record</em> LV(2), 191–208.
</div>
<div id="ref-federer_augmented_1961" class="csl-entry">
Federer, W. T. (1961). Augmented designs with one-way elimination of
heterogeneity. <em>Biometrics</em> 17, 447–473. doi:<a href="https://doi.org/10.2307/2527837" class="external-link">10.2307/2527837</a>.
</div>
<div id="ref-federerModelConsiderationsVariance1976" class="csl-entry">
Federer, W. T., and Searle, S. R. (1976). Model
<span>Considerations</span> and <span>Variance</span>
<span>Component</span> <span>Estimation</span> in <span>Augmented</span>
<span>Completely</span> <span>Randomized</span> and
<span>Randomized</span> <span>Complete</span> <span>Blocks</span>
<span>Designs</span>-<span>Preliminary</span> <span>Version</span>. New
York: Cornell University Available at: <a href="https://hdl.handle.net/1813/32691" class="external-link">https://hdl.handle.net/1813/32691</a>.
</div>
<div id="ref-johnson_estimates_1955" class="csl-entry">
Johnson, H. W., Robinson, H. F., and Comstock, R. E. (1955). Estimates
of genetic and environmental variability in soybeans. <em>Agronomy
journal</em> 47, 314–318. doi:<a href="https://doi.org/10.2134/agronj1955.00021962004700070009x" class="external-link">10.2134/agronj1955.00021962004700070009x</a>.
</div>
<div id="ref-lush_intra-sire_1940" class="csl-entry">
Lush, J. L. (1940). Intra-sire correlations or regressions of offspring
on dam as a method of estimating heritability of characteristics.
<em>Proceedings of the American Society of Animal Nutrition</em> 1940,
293–301. doi:<a href="https://doi.org/10.2527/jas1940.19401293x" class="external-link">10.2527/jas1940.19401293x</a>.
</div>
<div id="ref-robinson_quantitative_1966" class="csl-entry">
Robinson, H. F. (1966). Quantitative genetics in relation to breeding on
centennial of <span>Mendelism</span>. <em>Indian Journal of Genetics and
Plant Breeding</em>, 171.
</div>
<div id="ref-robinson_genetic_1955" class="csl-entry">
Robinson, H. F., Comstock, R. E., and Harvey, P. H. (1955). Genetic
variances in open pollinated varieties of corn. <em>Genetics</em> 40,
45–60. doi:<a href="https://doi.org/10.1093/genetics/40.1.45" class="external-link">10.1093/genetics/40.1.45</a>.
</div>
<div id="ref-sivasubramaniam_genotypic_1973" class="csl-entry">
Sivasubramaniam, S., and Madhavamenon, P. (1973). Genotypic and
phenotypic variability in rice. <em>The Madras Agricultural Journal</em>
60, 1093–1096.
</div>
<div id="ref-tippmann_programming_2015" class="csl-entry">
Tippmann, S. (2015). Programming tools: <span>Adventures</span> with
<span>R</span>. <em>Nature News</em> 517, 109. doi:<a href="https://doi.org/10.1038/517109a" class="external-link">10.1038/517109a</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by J. Aravind, S. Mukesh Sankar, Dhammaprakash Pandhari Wankhede, Vikender Kaur.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: 'bdafb7ea4490644e670f470ee3ef537f',
    indexName: 'augmentedrcbd',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
